<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
   "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
  <title>LogOddsLogo - User's Manual</title>
  <meta name="author" content="Yi-Kuo Yu, Aleksandar Stojmirovic, Stephen Altschul" >
  <!-- Credit to the original authors -->
  <meta name="author" content="Gavin E. Crooks" >
  <link href="qmbp.css" type="text/css" rel="stylesheet" />
  <link href="logoddslogo.css" type="text/css" rel="stylesheet" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
</head>

<body style="align:center">

<div id="header">
  <div id="qmbplogo"><h1><a href="http://www.ncbi.nlm.nih.gov/CBBresearch/Yu/index.html">QMBP</a></h1></div>
  <div id="logoddslogobanner">  <h1><a href="index.html">LogOddsLogo</a></h1></div>
  <div id="ncbilogo"><h1><a href="http://www.ncbi.nlm.nih.gov">NCBI</a></h1></div>
  <div class="cclr"></div>
</div>
<div id="leftpanel">
  <div id="gtgl1a" class="ctgl expanded"><div class="cover"><a id="gtgl1" class="ca1" href="#"></a><a href="http://www.ncbi.nlm.nih.gov/CBBresearch/Yu/webresources.html" class="ctgl">Web resources</a></div>
  <div id="gtgl1b" class="ctgldiv cback opened">
    <div id="gtgl1_1_2" class="ctgl none"><div class="cover"><span class="ca3"></span><a href="http://www.ncbi.nlm.nih.gov/CBBresearch/Yu/raid_dbs/index.html" class="ctgl">RAId_DbS</a></div></div>
    <div id="gtgl1_1_3a" class="ctgl none"><div class="cover"><span class="ca3"></span><a href="http://www.ncbi.nlm.nih.gov/CBBresearch/Yu/raid_aps/index.html" class="ctgl">RAId_aPS</a></div></div>
    <div id="gtgl1_1_5" class="ctgl none"><div class="cover"><span class="ca3"></span><a href="http://www.ncbi.nlm.nih.gov/CBBresearch/Yu/midas/index.html" class="ctgl">MIDAs</a></div></div>
    <div id="gtgl1_2a" class="ctgl none"><div class="cover"><span id="gtgl1_2" class="ca3"></span><a href="http://www.ncbi.nlm.nih.gov/CBBresearch/Yu/mn/itm_probe/index.html" class="ctgl">ITM Probe</a></div></div>
    <div id="gtgl1_2c" class="ctgl none"><div class="cover"><span class="ca2"></span><a href="http://www.ncbi.nlm.nih.gov/CBBresearch/Yu/mn/enrich/index.html" class="ctgl">SaddleSum</a></div></div>
    <div id="gtgl1_2c" class="ctgl none"><div class="cover cacur"><span class="ca2"></span><a href="./index.html" class="ctgl">LogOddsLogo</a></div></div>
  </div>
  </div>
  <div id="gtgl2a" class="ctgl none"><div class="cover"><span class="ca3"></span><a href="http://www.ncbi.nlm.nih.gov/CBBresearch/Yu/downloads.html" class="ctgl">Downloads</a></div>

  </div>
  <div id="gtgl3a" class="ctgl none"><div class="cover"><span id="gtgl3" class="ca3"></span><a href="http://www.ncbi.nlm.nih.gov/CBBresearch/Yu/members.html" class="ctgl">Members</a></div>

  </div>
  <div id="gtgl4" class="ctgl none"><div class="cover"><span class="ca2"></span><a href="http://www.ncbi.nlm.nih.gov/CBBresearch/Yu/publications.html" class="ctgl">Publications</a></div></div>
</div>
<div id="rightpanel">
<p class="sidetitle">LogOddsLogo</p>
<h4>Web service</h4>
<ul>
  <li><a href="proteins.cgi">Create Protein Logo</a></li>
  <li><a href="nucleicacid.cgi">Create Nucleic Acid Logo</a></li>
</ul><br />
<h4>Download</h4>
<ul>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/pub/qmbp/logoddslogo/">FTP Site</a></li>
</ul><br />
<h4>Documentation</h4>
<ul>
<li><a href="./">About</a></li>
<li><a href="examples.html">Examples</a></li>
<li><a href="manual.html">Manual</a></li>
</ul>
</div>

<div id="middlepanel">
<!--  ========================== -->


<h1>LogOddsLogo - User's Manual</h1>

<h4>Contents</h4>

<ul>
<li><a href="#intro">Introduction</a>
</li><li><a href="#create">Creating Sequences Logos using the Web interface</a>
</li><li><a href="#download">Downloading and Installing LogOddsLogo</a>
</li><li><a href="#CLI"> Command Line Interface (CLI)</a>
</li><li><a href="#API"> Application Programmer Interface (API)</a>
</li><li><a href="#misc">Miscellanea</a>
</li>
</ul>

<a name="intro" ></a><h2>Introduction</h2>


<p>
<strong>LogOddsLogo</strong>
is a web based application designed to make the generation of
sequence logos from biological sequence alignments as easy and painless as possible.
 Building on the <a href="http://weblogo.threeplusone.com">WebLogo 3</a> source code, LogOddsLogo uses per-observation multiple-alignment log-odds scores as measures of information
content at each position of a sequence logo. It provides the methods for logo generation that have a proper statistical basis and are optimal for recognising functionally relevant alignment columns.
</p>


<p>
A <a href="http://www.lecb.ncifcrf.gov/~toms/sequencelogo.html">sequence&nbsp;logo</a>
is a graphical representation of an amino acid
or nucleic acid multiple sequence alignment.
Each logo consists of stacks of symbols, one stack for each position in the
sequence. The overall height of the stack indicates the sequence conservation
at that position, while the height of symbols within the stack indicates the
relative frequency of each amino or nucleic acid at that position. The width of the stack is proportional to the fraction of valid symbols in that position. (Positions with many gaps have thin stacks.)  In general, a sequence logo provides a richer and more precise description of, for example,a binding site, than would a consensus sequence.
</p>







<!-- ============================================================== -->

<h4>References</h4>

<p>
Yu YK, Capra JA, Stojmirovic A, Landsman D, Altschul, SF.
Log-odds sequence logos. <em>Submitted for publication</em>, (2014).
</p>

<p>
Altschul SF, Wootton JC, Zaslavsky E, Yu YK.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/20657661">The construction and use of log-odds substitution scores for multiple sequence alignment.</a>
<em>PLoS Comput Biol.</em> 6:e1000852, (2010).
</p>


<p>
Altschul SF, Gertz EM, Agarwala R, Sch√§ffer AA, Yu YK.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19088134">PSI-BLAST pseudocounts and the minimum description length principle.</a>
<em>Nucleic Acids Res.</em> 37:815-824, (2009).
</p>

<p>
Crooks GE, Hon G, Chandonia JM, Brenner SE.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/15173120"> WebLogo: A sequence logo
generator.</a> <em>Genome Research</em> 14:1188-1190, (2004).
</p>

<p>
Sunyaev, S. R. <em>et al.</em>
<a href="http://www.ncbi.nlm.nih.gov/pubmed/10360979">PSIC: Profile extraction from sequence alignments with position-specific counts of independent
observations. </a> <em> Protein Eng.</em> 12:387-394, (1999).
<p>
Robinson AB, Robinson LR.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/1924347">
Distribution of glutamine and asparagine residues and their near neighbors in peptides and proteins.</a> <em> Proc. Natl Acad. Sci. USA </em> 88:8888-8884, (1991).
</p>


<p>
Schneider TD, Stephens RM.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/2172928">Sequence logos: A new way to display consensus sequences.</a>
<em>Nucleic Acids Res.</em> 18:6097-6100, (1990).
</p>



<a name="create" ></a>
<h2>Creating Sequences Logos using the Web interface</h2>

<p>Unlike its predecessor WebLogo, LogOddsLogo offers two web interfaces for creating sequence logos, one for <a href="proteins.cgi">protein</a> sequences and another for <a href="nucleicacid.cgi">nucleic acid</a> (DNA or RNA) sequences. Each interface only offers the options relevant to the type of sequences it supports. Initally both interfaces offer the entries for only the <a href="#basic">most essential</a> parameters. The <a href="#advanced">advanced parameters</a> can be accessed by clicking on the <em>More parameters</em> link.

<a name="basic" ></a>
<h3>Basic Parameters</h3>

<h4>Sequence Data</h4>
Enter your multiple sequence alignment or position weight matrix file, or select a file to upload. Supported file formats include CLUSTALW, FASTA, plain flatfile, MSF, NBRF, PIR, NEXUS and PHYLIP for multiple sequence alignments, and transfac for position weight matrices.  All sequences must be the same length, else LogOddsLogo will return an error and report the first sequence that differed in length from previous sequences.


<h4>Output format</h4>

<ul>
<li> PNG : (600 DPI) Print resolution bitmap
</li><li> PNG :  (low res, 96 DPI) Screen resolution bitmap
</li><li> JPEG :Screen resolution bitmap
</li><li> EPS : Encapsulated postscript
</li><li> PDF : Portable Document Format
</li><li> SVG : Scalable Vector Graphics
</li>
</ul>
Generally speaking, vector formats (EPS and PDF) are better for printing, while bitmaps (JPEG and PNG) are more suitable for displaying on the screen or embedding into a web page.

<h4>Logo size</h4>
The physical dimensions of the generated logo.
Specifically, controls the size of individual symbols stacks.
<ul>
<li>     small : 5.4 points wide (same as 9pt Courier), aspect ratio 5:1
</li><li> medium : Double the width and height of small.
</li><li> large : Triple the width and height of small.
</li>
</ul>
The choices have been limited to promote inter-logo consistency. Small logos can fit 80 stacks across a printed page, or 40 across a half page column. The <a href="#CLI">command line interface</a> provides greater control, if so desired.

<h4>Sequence type (nucleic acids only)</h4>
The type of biological molecule.
<ul>
<li> DNA
</li><li> RNA
</li>
</ul>

<a name="advanced" ></a>
<h3>Advanced Parameters</h3>

<h4>Scoring method</h4>
<ul>
<li> BILD (Bayesian Integral Log-odds) score
</li><li> Normalized Maximum Likelihood score
</li><li> Schneider and Stephens entropy difference score - Corrected
</li><li> Schneider and Stephens entropy difference score - Uncorrected
</li>
</ul>

<h4>Dirichlet mixture (protein BILD score only)</h4>
A stored Dirichlet mixture to be used as a prior for BILD scores (the first five are taken from <a href="http://compbio.soe.ucsc.edu/dirichlets/index.html">UCSC Dirichlet mixtures pages</a> while the last four are from the ncbi in-house development).
<ul>
<li> recode3-20 (20 components)
</li><li> recode4-20 (20 components)
</li><li> recode5-20 (20 components)
</li><li> Fournier-20 (20 components)
</li><li> dist-20 (20 components)
</li><li> dist-ncbi-52 (52 components)
</li><li> dist-ncbi-72 (72 components)
</li><li> dist-ncbi-110 (110 components)
</li><li> dist-ncbi-134 (134 components).
</li></ul>

<h4>Dirichlet concentration parameter (nucleic acid BILD score only)</h4>
The nucleic acid logos take as a Dirichlet mixture prior a single Dirichlet distribution with determined by the concentration parameter <em>alpha</em>.

<h4>Weight counts (proteins only)</h4>
Weight amino acid counts for each column by estimating aggregate numbers of independent observations, using the method of Sunyaev et al (1999) as modified by Altschul et al (2009).

<h4>Draw an overline on the positive score region</h4>
Determine the boundaries for the logo using a one-dimensional Smith-Waterman algorithm and draw an overline above the positive score region.

<h4>Stacks per line</h4>
If the length of the sequences is greater than this maximum number of stacks per line, then the logo will be split across multiple lines.


<h4>Ignore lower case</h4>
Disregard lower case letters and only count
upper case letters in sequences?

<h4>Units</h4>
The units used for the y-axis.
<ul>
<li>      probability: Show residue probabilities, rather than information content.
</li><li> bits: Information content in bits
</li><li> nats: Natural units, 1 bit = ln 2 (0.69) nats
</li><li> kT : Thermal energy units in natural units (Numerically the same as nats)
</li><li> kJ/mol : Thermal energy (assuming T = 300 K)
</li><li> kcal/mol : Thermal energy (assuming T = 300 K)
</li>
</ul>


<h4>First position number</h4>
<a name="First position number"></a>
The numerical label of the first position in the sequence data in the input file. The label must be an integer. Residue labels for the logo will be relative to this number. (See also: <a href="#logo range">Logo range</a>.)
<h4>Logo range</h4>
<a name="logo range"></a>
By default, all sequence data from the input file is displayed in the Sequence Logo. With this option, you can instead show a subrange of the sequence data. The numbering of Start and End Positions is relative to the First Position Number. Thus, if the First Position Number is "2", Start is "5" and End is "10", then the 4th through 9th (inclusive) sequence positions of the input file will be displayed, and they will be numbered "5", "6", "7", "8", "9" and "10".

<a name="CA"></a>
<h4>Composition</h4>
<p>
The background composition of the genome from which the sequences have been drawn. The default option is to use equiprobable background.  However, you may also explicitly set the expected CG content for nucleic acid sequences, insist on equiprobable background distributions, or use any of the prescribed compositions in the dropdown box.
</p>
<p>
For proteins, the background composition, consisting of a set of emission probabilities of amino acids, is used to calculate the scores for the Bayesian Integral Log-odds (BILD) method, the normalized maximum likelihood (NML) method and the Schneider-corrected (SC) method. Currently, we use the Robinson-Robinson frequencies as the background amino acid frequencies for the latter two methods mentioned above. However, in the command line version of the program, the user may specify a different set of background probabilities. BILD score, on the other hand, always uses the implicit background probabilities of the selected Dirichlet mixture.
</p>
<!-- <p> -->
<!-- Secondly, the background composition is used in the small sample correction of information content. Briefly, if only a few sequences are available in the multiple sequence alignment, then sites typically appear more conserved than they really are. Small samples bias the relative entropy upwards. To compensate, we add pseudocounts to the actual counts, proportional to the expected background  composition. These pseudocounts smooth the data for small samples, but become irrelevant for large samples. The proportionality constant is set to 4 for nucleic acid sequences, and 20 for proteins (these numbers have been found to give reasonable results in practice). -->
<!-- </p> -->
<!-- <p> -->
<!-- Behind the scenes, things are more complex. We do a full Bayesian calculation, starting with explicit Dirichlet priors based on the background composition, to which we add the data and then calculate both the posterior mean relative entropy (the stack height) and Bayesian 95% confidence intervals for error bars. These interesting details will be explained elsewhere. -->
<!-- </p> -->

<h4>Scale stack width</h4>
Scale the visible stack width by the fraction of symbols in the column?  (i.e. columns with many gaps or unknown residues are narrow.)

<h4>Title</h4>
Give your logo a title.

<h4>Figure label</h4>
An optional figure label, added to the top left (e.g. '(a)').

<h4>X-axis</h4>
Add a label to the x-axis, or hide axis altogether.

<h4>Y-axis</h4>
The vertical axis indicates the information content of a sequence position. Use this option to toggle the y-axis and override the default axis label.

<h4>Y-axis scale</h4>
The height of the y-axis in designated units. The automatic option will pick reasonable defaults based on the sequence type and axis unit.

<h4>Y-axis tic spacing</h4>
The distance between major tic marks on the y-axis.

<h4>Sequence end labels</h4>
Choose this option to label the 5' &amp; 3' ends of nucleic acid or the N &amp; C termini of amino acid sequences.

<h4>Version fineprint</h4>
Toggle display of the LogOddsLogo version information in the lower right corner.

<h4>Color Scheme</h4>
<ul>
<li> monochrome : All symbols black
</li><li> Base Pairing (NA default) :
<table class="striped">
  <tr class="row1">
    <td class="col-desc">2 Watson-Crick hydrogen bonds</td>
    <td class="col-letters">TAU</td>
    <td><span style="color:darkorange">dark orange</span></td></tr>
  <tr class="row2">
    <td>3 Watson-Crick hydrogen bonds</td>
    <td>GC</td>
    <td><span style="color:blue">blue</span></td>
</tr>
</table>

</li><li> Classic (NA) : WebLogo (version 1 and 2) and makelogo default color scheme for nucleic acids: G, orange; T &amp; U, red; C, blue; and A, green.
<table class="striped">
  <tr class="row1">
    <td class="col-desc">G</td>
    <td class="col-letters">G</td>
    <td style="color:orange"> orange</td>
  </tr>
  <tr class="row2">
    <td>TU</td>
    <td>TU</td>
    <td style="color:red"> red</td>
  </tr>
  <tr class="row1">
    <td>C</td>
    <td>C</td>
    <td style="color:blue">blue</td>
  </tr>
  <tr class="row2">
    <td>A</td>
    <td>A</td>
    <td style="color:green">green</td>
  </tr>
</table>

</li><li> Hydrophobicity (AA default) :
<table class="striped">
  <tr class="row1">
    <td class="col-desc">Hydrophilic</td>
    <td class="col-letters">RKDENQ</td>
    <td style="color:blue">blue</td>
  </tr>
  <tr class="row2">
    <td>Neutral</td>
    <td>SGHTAP</td>
    <td style="color:green"> green</td>
  </tr>
  <tr class="row1">
    <td>Hydrophobic</td>
    <td>YVMCLFIW</td>
    <td style="color:black"> black</td>
  </tr>
</table>

</li><li> Chemistry (AA) :  Color amino acids according to chemical properties. WebLogo (version 1 and 2) and makelogo default color. (Note that the WebLogo 2 documentation erroneously lists Q and N under green.)
<table class="striped">
  <tr class="row1">
    <td class="col-desc">Polar</td>
    <td class="col-letters">GSTYC</td>
    <td style="color:green"> green</td>
  </tr>
  <tr class="row2">
    <td>Neutral</td>
    <td>QN</td>
    <td style="color:purple"> purple</td>
  </tr>
  <tr class="row1">
    <td>Basic</td>
    <td>KRH</td>
    <td style="color:blue"> blue</td>
  </tr>
  <tr class="row2">
    <td>Acidic</td>
    <td>DE</td>
    <td style="color:red">red</td>
  </tr>
  <tr class="row1">
    <td>Hydrophobic</td>
    <td>AVLIPWFM</td>
    <td style="color:black">black</td>
  </tr>
</table>

</li><li> Charge (AA) :
<table class="striped">
  <tr class="row1">
    <td class="col-desc">Positive</td>
    <td class="col-letters">KRH</td><td style="color:blue"> blue</td>
  </tr>
  <tr class="row2">
    <td>Negative</td>
    <td>DE</td><td style="color:red"> red</td>
  </tr>
</table>


</li><li> Custom : A custom color scheme can be specified in the input field below. Specify colors on the left and associated symbols on the right. Colors are entered using  <a href="http://www.w3.org/TR/REC-CSS2/syndata.html#color-units">CSS2 (Cascading Style Sheet)</a> syntax. (E.g. 'red', '#F00', '#FF0000',  'rgb(255, 0, 0)', 'rgb(100%, 0%, 0%)' or 'hsl(0, 100%, 50%)' for the color red.)
</li>
</ul>

<h4>More Options</h4>
The weblogo <a href="#CLI">command line client</a>, <code>logoddslogo</code>, provides many more options and greater control over the final logo appearance.






<!-- ========================================================= -->
<a name="download" ></a>
<h2>Installing LogOddsLogo</h2>


<h4>Dependencies</h4>
<p>
LogOddsLogo is written in Python. It is necessary to have <a href="http://www.python.org/download/">Python 2.7</a>  and the
extension package <a href="http://www.scipy.org/Download">numpy</a>
installed before LogOddsLogo will run. LogOddsLogo also requires a recent version of  <a href="http://www.cs.wisc.edu/~ghost/">ghostscript</a> to create PNG and PDF output, and <a href="http://www.cityinthesky.co.uk/opensource/pdf2svg">pdf2svg</a> to generate SVG output.
</p>


<h4> Download and Installation</h4>

<p>
The LogOddsLogo source code can be downloaded from
<a href="ftp://ftp.ncbi.nlm.nih.gov/pub/qmbp/logoddslogo/">ftp://ftp.ncbi.nlm.nih.gov/pub/qmbp/logoddslogo/</a>.
This code is distributed under various <a href="http://www.opensource.org/docs/definition">open source licenses</a>. Please consult the <code>LICENSE.txt</code> file in the source distribution for details.
</p>

<p>
After unpacking the LogOddsLogo tarfile, it should be possible to immediately create
logos using the command line client (provided that python, numpy and ghostscript have already been installed).
</p>
<pre>
./logoddslogo --format PNG &lt; cap.fa &gt; cap.png
</pre>
<p>
Please consult the file <code>build_examples.sh</code> for more examples.
</p><p>
To run LogOddsLogo as a stand alone web service, run the logo server command :
</p>
<pre>
./logoddslogo --serve
</pre>
<p>
It should now be possible to access LogOddsLogo at <a href="http://localhost:8080/">http://localhost:8080/</a>.
</p>

<p>
The command line client and LogOddsLogo libraries can be permanently installed using the supplied <code>setup.py</code> script.</p>
<pre>
sudo python setup.py install
</pre>
<p>
Run <code>python setup.py help</code> for more installation options. For example, to specifically install the logoddslogo script to /usr/local/bin </p>
<pre>
sudo python setup.py install_scripts --install-dir /usr/local/bin
</pre>

<p>
<strong>Note</strong> that LogOddsLogo source code fully contains the code for WebLogo, version 3.3. All the changes made at the NCBI are placed in separate directories and hence LogOddsLogo and WebLogo-3.3 can coexist on the same Python installation. The WebLogo executable is still present in LogOddsLogo distribution, but it is not installed by the setup script.
</p>



<h4>Web App</h4>

<p>
  To use LogOddsLogo as a web application, first install the its dependencies and libraries as above, then
 place (or link) the <code>logoddslogolib/htdocs</code> directory
  somewhere within the document root of your webserver. The webserver
  must be able to execute the CGI scripts <code>proteins.cgi</code> and <code>nucleicacid.cgi</code>. For Apache, you may have to add an <code>ExecCGI</code>
  option and add a cgi handler in the <code>httpd.conf</code> configuration file.
  Something like this:
</p>
<pre>
&lt;Directory "/home/httpd/htdocs/logoddslogo/">
    Options FollowSymLinks MultiViews ExecCGI
    AllowOverride None
    Order allow,deny
    Allow from all
&lt;/Directory>
...
# To use CGI scripts outside of ScriptAliased directories:
# (You will also need to add "ExecCGI" to the "Options" directive.)
#
AddHandler cgi-script .cgi
</pre>
It may also be necessary to set the <code>PATH</code> and <code>PYTHONPATH</code> environment variables.
<pre>
SetEnv PYTHONPATH /path/to/logoddslogo/libraries
</pre>
The cgi script also has to be able to find the '<code>gs</code>' ghostscript executable. If ghostscript is installed in a non-standard location add the following environment variable.
<pre>
SetEnv COREBIOPATH /path/to/gs
</pre>
The maximum bytes of uploaded sequence data can be controlled with the <code>WEBLOGO_MAX_FILE_SIZE</code> environment variable.
<pre>
SetEnv WEBLOGO_MAX_FILE_SIZE 1000000
</pre>




<!-- ================================================================== -->
<a name="CLI" ></a>
<h2><code>logoddslogo</code>, The LogOddsLogo Command Line Interface (CLI)</h2>
The  command line client has many options not available through the web interface. Please consult the bundled <code>build_examples.sh</code> script for inspiration.
<pre>
Usage: logoddslogo [options]  < sequence_data.fa > sequence_logo.eps

Create sequence logos from biological sequence alignments.

Options:
     --version                  show program's version number and exit
  -h --help                     show this help message and exit

  Input/Output Options:
    -f --fin FILENAME           Sequence input file (default: stdin)
    -D --datatype FORMAT        Type of multiple sequence alignment or
                                position weight matrix file: (clustal, fasta,
                                plain, msf, genbank, nbrf, nexus, phylip,
                                stockholm, intelligenetics, table, array,
                                transfac)
    -o --fout FILENAME          Output file (default: stdout)
    -F --format FORMAT          Format of output: eps (default), png,
                                png_print, pdf, jpeg, svg, logodata

  Logo Data Options:
    -M --score-method SCORE_METHOD
                                The method for scoring: 'BILD' or 'NML' or 'SC'
                                or 'SU':
                                BILD - BILD score
                                NML  - Normalized Maximum Likelihood score
                                SC   - Schneider score - Corrected
                                SU   - Schneider score - Uncorrected
    -d --dmnumber DMNUMBER      Dirichlet mixture parameter for BILD score.
                                For nucleotide sequences, it should be a
                                floating point value corresponding to
                                Dirichlet concentration alpha (DEFAULT 1.0).
                                For protein sequences, it should be an integer
                                (0 to 8) indicating a particular stored
                                Dirichlet mixture:
                                0 - recode3-20 (20 components)
                                1 - recode4-20 (20 components)
                                2 - recode5-20 (20 components)
                                3 - Fournier-20 (20 components)
                                4 - dist-20 (20 components)
                                5 - dist-ncbi-52 (52 components - DEFAULT)
                                6 - dist-ncbi-72 (72 components)
                                7 - dist-ncbi-110 (110 components)
                                8 - dist-ncbi-134 (134 components)
                                This parameter is ignored for any other
                                scoring method.
    -A --sequence-type TYPE     The type of sequence data: 'protein', 'rna' or
                                'dna'.
    -a --alphabet ALPHABET      The set of symbols to count, e.g. 'AGTC'. All
                                characters not in the alphabet are ignored. If
                                neither the alphabet nor sequence-type are
                                specified then logoddslogo will examine the
                                input data and make an educated guess. See
                                also --sequence-type, --ignore-lower-case
    -U --units NUMBER           A unit of entropy ('bits' (default), 'nats',
                                'digits'), or a unit of free energy ('kT',
                                'kJ/mol', 'kcal/mol'), or 'probability' for
                                probabilities
       --composition COMP.      The expected composition of the sequences:
                                'equiprobable (default)', a CG percentage,
                                a species name (e.g. 'E. coli','H. sapiens'),
                                or an explicit distribution
                                (e.g. "{'A':10, 'C':40, 'G':40, 'T':10}").
                                For proteins, NML and SC use the
                                Robinson-Robinson frequencies as the background,
                                although an explicit specification (same format as above)
                                is also allowed. BILD always uses the implicit background
                                frequencies of the selected Dirichlet mixture.
       --weight NUMBER          The weight of prior data.  Default depends on
                                alphabet length
       --no-weighcounts         For proteins only, do not use other columns to
                                estimate the number of independent
                                observations of an amino acid within the
                                column considered.  Default is True
       --ovline                 Draw an overline on the positive score region.
                                Default is False
    -i --first-index INDEX      Index of first position in sequence data
                                (default: 1)
    -l --lower INDEX            Lower bound of sequence to display
    -u --upper INDEX            Upper bound of sequence to display

  Transformations:
    Optional transformations of the sequence data.

       --ignore-lower-case      Disregard lower case letters and only count
                                upper case letters in sequences.
       --reverse                reverse sequences
       --complement             complement DNA sequences

  Logo Format Options:
    These options control the format and display of the logo.

    -s --size LOGOSIZE          Specify a standard logo size (small, medium
                                (default), large)
    -n --stacks-per-line COUNT  Maximum number of logo stacks per logo line.
                                (default: 40)
    -t --title TEXT             Logo title text.
       --label TEXT             A figure label, e.g. '2a'
    -X --show-xaxis YES/NO      Display sequence numbers along x-axis?
                                (default: True)
    -x --xlabel TEXT            X-axis label
       --annotate TEXT          A comma separated list of custom stack
                                annotations, e.g. '1,3,4,5,6,7'.  Annotation
                                list must be same length as sequences.
    -S --yaxis UNIT             Height of yaxis in units. (Default: Maximum
                                value with uninformative prior.)
    -Y --show-yaxis YES/NO      Display entropy scale along y-axis? (default:
                                True)
    -y --ylabel TEXT            Y-axis label (default depends on plot type and
                                units)
    -E --show-ends YES/NO       Label the ends of the sequence? (default:
                                False)
    -P --fineprint TEXT         The fine print (default: logoddslogo version)
       --ticmarks NUMBER        Distance between ticmarks (default: 1.0)
       --errorbars YES/NO       Display error bars? (default: False)
       --reverse-stacks YES/NO  Draw stacks with largest letters on top?
                                (default: False)

  Color Options:
    Colors can be specified using CSS2 syntax. e.g. 'red', '#FF0000', etc.

    -c --color-scheme SCHEME    Specify a standard color scheme (auto, base
                                pairing, charge, chemistry, classic,
                                hydrophobicity, monochrome)
    -C --color COLOR SYMBOLS DESCRIPTION
                                Specify symbol colors, e.g. --color black AG
                                'Purine' --color red TC 'Pyrimidine'
       --default-color COLOR    Symbol color if not otherwise specified.

  Advanced Format Options:
    These options provide fine control over the display of the logo.

    -W --stack-width POINTS     Width of a logo stack (default: 10.8)
       --aspect-ratio POINTS    Ratio of stack height to width (default: 5)
       --box YES/NO             Draw boxes around symbols? (default: no)
       --resolution DPI         Bitmap resolution in dots per inch (DPI).
                                (Default: 96 DPI, except png_print, 600 DPI)
                                Low resolution bitmaps (DPI<300) are
                                antialiased.
       --scale-width YES/NO     Scale the visible stack width by the fraction
                                of symbols in the column?  (I.e. columns with
                                many gaps of unknowns are narrow.)  (Default:
                                yes)
       --debug YES/NO           Output additional diagnostic information.
                                (Default: False)

  LogOddsLogo Server:
    Run a standalone webserver on a local port.

       --serve                  Start a standalone LogOddsLogo server for
                                creating sequence logos.
       --port PORT              Listen to this local port. (Default: 8080)
</pre>
<!-- ===================================================================== -->
<a name="API" ></a>
<h2>LogOddsLogo Application Programmer Interface (API)</h2>

The LogOddsLogo python libraries provide even greater flexibility than the command line client. The code is split between three packages, <code>logoddslogolib</code>, <code>weblogolib</code> and <code>corebio</code>. The package <code>logoddslogolib</code> contains all NCBI changes from WebLogo-3.3. The other two packages are copied without change from WebLogo-3.3: <code>weblogolib</code> contains specialized sequence logo generation code, while <code>corebio</code> contains code of more general utility. Please consult the <a href="apidocs/">API documentation</a> for all three packages generated using <a href="http://epydoc.sourceforge.net/">Epydoc</a>.

<!-- ================================================================== -->

<a name="misc" ></a>
<h2>Miscellanea</h2>

<h4> Release Notes and Known Bugs</h4>
The <a href="logoddslogo_changelog.txt">LogOddsLogo release notes</a> detail changes to LogOddsLogo and known issues with particular versions.


<h4>WebLogo</h4>

<p>The WebLogo 3 sever can be found <a href="http://weblogo.threeplusone.com/">here.</a></p>
<p>The legacy WebLogo 2 sever can be found <a href="http://weblogo.berkeley.edu/">here.</a></p>


<h4>Acknowledgments</h4>

<p>LogOddsLogo was written by Yi-Kuo Yu, Stephen Altschul and Aleksandar Stojmirovic, with input from David Landsman and John Capra. It was based on the <a href="http://weblogo.threeplusone.com">WebLogo-3.3</a> source code and the research of Yi-Kuo Yu and Stephen Altschul on information content of multiple sequence alignments.</p>

<p>
WebLogo was created by
<a href="http://threeplusone.com/">Gavin E. Crooks</a>,
<a href="http://compbio.berkeley.edu/">Liana Lareau</a>,
<a href="http://compbio.berkeley.edu/">Gary Hon</a>,
<a href="http://compbio.berkeley.edu/">John-Marc Chandonia</a> and
<a href="http://compbio.berkeley.edu/people/brenner/">Steven E. Brenner</a>.
<a href="logoddslogo_changelog.txt">Many others</a> have provided suggestions, bug fixes and moral support.
</p>

<p>
WebLogo was originally based upon the programs
<a href="http://www.lecb.ncifcrf.gov/~toms/delila/alpro.html">alpro</a> and
<a href="http://www.lecb.ncifcrf.gov/~toms/delila/makelogo.html">makelogo</a>,
both of which are part of Tom Schneider's
<a href="http://www.lecb.ncifcrf.gov/~toms/delila.html">delila</a> package. Many thanks
are due to him for making this software freely available and for encouraging its use.
</p>


<h4>Feedback</h4>
<p>
Please direct questions and feedback about LogOddsLogo to <a href="http://www.ncbi.nlm.nih.gov/CBBresearch/Yu/">Yi-Kuo Yu</a>.
</p>

<!--  ========================= -->
</div>

<hr id="footsep" />
<div id="footer">
  <ul>
    <li><a href="http://www.ncbi.nih.gov/About/disclaimer.html">Copyright</a></li>
    <li><a href="http://www.ncbi.nih.gov/About/disclaimer.html#disclaimer">Disclaimer</a></li>
    <li><a href="http://www.nlm.nih.gov/privacy.html">Privacy</a></li>
    <li><a href="http://www.ncbi.nih.gov/About/accessibility.html">Accessibility</a></li>
  </ul>
  <!-- <span id="codedby">Maintained by Yi-Kuo Yu. Last update: February 28, 2014</span> -->
</div>

</body>
</html>
