<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>corebio.seq_io._nexus.Trees</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="corebio-module.html">Package&nbsp;corebio</a> ::
        <a href="corebio.seq_io-module.html">Package&nbsp;seq_io</a> ::
        <a href="corebio.seq_io._nexus-module.html" onclick="show_private();">Package&nbsp;_nexus</a> ::
        Module&nbsp;Trees
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="corebio.seq_io._nexus.Trees-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="corebio.seq_io._nexus.Trees-module.html">Module corebio.seq_io._nexus.Trees</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># Trees.py </tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2005 by Frank Kauff &amp; Cymon J. Cox. All rights reserved.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment"># license. Please see the LICENSE file that should have been included</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-comment"># Tree class handles phylogenetic trees. Provides a set of methods to read and write newick-format tree</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-comment"># descriptions, get information about trees (monphyly of taxon sets, congruence between trees, common ancestors,...)</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-comment"># and to manipulate trees (reroot trees, split terminal nodes).</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-comment"># Bug reports welcome: fkauff@duke.edu</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt><tt class="py-op">,</tt> <tt class="py-name">random</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-0" class="py-name" targets="Module corebio.seq_io._nexus.Nodes=corebio.seq_io._nexus.Nodes-module.html"><a title="corebio.seq_io._nexus.Nodes" class="py-name" href="#" onclick="return doclink('link-0', 'Nodes', 'link-0');">Nodes</a></tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-comment"># Python 2.3 compatability</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line">    <tt id="link-1" class="py-name" targets="Method corebio.seq_io._nexus.StepMatrix.set()=corebio.seq_io._nexus.StepMatrix-class.html#set"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-1', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-keyword">except</tt> <tt class="py-name">NameError</tt><tt class="py-op">:</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-name">sets</tt> <tt class="py-keyword">import</tt> <tt class="py-name">Set</tt> <tt class="py-keyword">as</tt> <tt id="link-2" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-2', 'set', 'link-1');">set</a></tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt id="link-3" class="py-name" targets="Variable corebio.seq_io._nexus.Trees.PRECISION_BRANCHLENGTH=corebio.seq_io._nexus.Trees-module.html#PRECISION_BRANCHLENGTH"><a title="corebio.seq_io._nexus.Trees.PRECISION_BRANCHLENGTH" class="py-name" href="#" onclick="return doclink('link-3', 'PRECISION_BRANCHLENGTH', 'link-3');">PRECISION_BRANCHLENGTH</a></tt><tt class="py-op">=</tt><tt class="py-number">6</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt id="link-4" class="py-name" targets="Variable corebio.seq_io._nexus.Trees.PRECISION_SUPPORT=corebio.seq_io._nexus.Trees-module.html#PRECISION_SUPPORT"><a title="corebio.seq_io._nexus.Trees.PRECISION_SUPPORT" class="py-name" href="#" onclick="return doclink('link-4', 'PRECISION_SUPPORT', 'link-4');">PRECISION_SUPPORT</a></tt><tt class="py-op">=</tt><tt class="py-number">6</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"> </tt>
<a name="TreeError"></a><div id="TreeError-def"><a name="L30"></a><tt class="py-lineno"> 30</tt> <a class="py-toggle" href="#" id="TreeError-toggle" onclick="return toggle('TreeError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.TreeError-class.html">TreeError</a><tt class="py-op">(</tt><tt class="py-base-class">Exception</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-keyword">pass</tt> </tt>
</div><a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"> </tt>
<a name="NodeData"></a><div id="NodeData-def"><a name="L32"></a><tt class="py-lineno"> 32</tt> <a class="py-toggle" href="#" id="NodeData-toggle" onclick="return toggle('NodeData');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.NodeData-class.html">NodeData</a><tt class="py-op">:</tt> </tt>
</div><div id="NodeData-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="NodeData-expanded"><a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line">    <tt class="py-docstring">"""Stores tree-relevant data associated with nodes (e.g. branches or otus)."""</tt> </tt>
<a name="NodeData.__init__"></a><div id="NodeData.__init__-def"><a name="L34"></a><tt class="py-lineno"> 34</tt> <a class="py-toggle" href="#" id="NodeData.__init__-toggle" onclick="return toggle('NodeData.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.NodeData-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">taxon</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt><tt class="py-param">branchlength</tt><tt class="py-op">=</tt><tt class="py-number">0.0</tt><tt class="py-op">,</tt><tt class="py-param">support</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NodeData.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="NodeData.__init__-expanded"><a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">=</tt><tt class="py-name">taxon</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">=</tt><tt class="py-name">branchlength</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">=</tt><tt class="py-name">support</tt> </tt>
</div></div><a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"> </tt>
<a name="Tree"></a><div id="Tree-def"><a name="L39"></a><tt class="py-lineno"> 39</tt> <a class="py-toggle" href="#" id="Tree-toggle" onclick="return toggle('Tree');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html">Tree</a><tt class="py-op">(</tt><tt class="py-base-class">Nodes</tt><tt class="py-op">.</tt><tt class="py-base-class">Chain</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Tree-expanded"><a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">    <tt class="py-docstring">"""Represents a tree using a chain of nodes with on predecessor (=ancestor)</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">    and multiple successors (=subclades).</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt>  </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">    <tt class="py-comment"># A newick tree is parsed into nested list and then converted to a node list in two stages</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">    <tt class="py-comment"># mostly due to historical reasons. This could be done in one swoop). Note: parentheses ( ) and</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">    <tt class="py-comment"># colon : are not allowed in taxon names. This is against NEXUS standard, but makes life much</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">    <tt class="py-comment"># easier when parsing trees.</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">     </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">    <tt class="py-comment">## NOTE: Tree should store its data class in something like self.dataclass=data,</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">    <tt class="py-comment">## so that nodes that are generated have easy access to the data class</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">    <tt class="py-comment">## Some routines use automatically NodeData, this needs to be more concise</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"> </tt>
<a name="Tree.__init__"></a><div id="Tree.__init__-def"><a name="L52"></a><tt class="py-lineno"> 52</tt> <a class="py-toggle" href="#" id="Tree.__init__-toggle" onclick="return toggle('Tree.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">tree</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt><tt class="py-param">weight</tt><tt class="py-op">=</tt><tt class="py-number">1.0</tt><tt class="py-op">,</tt><tt class="py-param">rooted</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt><tt class="py-param">name</tt><tt class="py-op">=</tt><tt class="py-string">''</tt><tt class="py-op">,</tt><tt class="py-param">data</tt><tt class="py-op">=</tt><tt id="link-5" class="py-name" targets="Class corebio.seq_io._nexus.Trees.NodeData=corebio.seq_io._nexus.Trees.NodeData-class.html"><a title="corebio.seq_io._nexus.Trees.NodeData" class="py-name" href="#" onclick="return doclink('link-5', 'NodeData', 'link-5');">NodeData</a></tt><tt class="py-op">,</tt><tt class="py-param">values_are_support</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt><tt class="py-param">max_support</tt><tt class="py-op">=</tt><tt class="py-number">1.0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.__init__-expanded"><a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">        <tt class="py-docstring">"""Ntree(self,tree)."""</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">        <tt id="link-6" class="py-name"><a title="corebio.seq_io._nexus.Nodes" class="py-name" href="#" onclick="return doclink('link-6', 'Nodes', 'link-0');">Nodes</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Class corebio.seq_io._nexus.Nodes.Chain=corebio.seq_io._nexus.Nodes.Chain-class.html"><a title="corebio.seq_io._nexus.Nodes.Chain" class="py-name" href="#" onclick="return doclink('link-7', 'Chain', 'link-7');">Chain</a></tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Method corebio.db.DataSource.__init__()=corebio.db.DataSource-class.html#__init__,Method corebio.db.Dbxref.__init__()=corebio.db.Dbxref-class.html#__init__,Method corebio.db.astral.Raf.__init__()=corebio.db.astral.Raf-class.html#__init__,Method corebio.db.astral.RafSeqMap.__init__()=corebio.db.astral.RafSeqMap-class.html#__init__,Method corebio.db.astral.Res.__init__()=corebio.db.astral.Res-class.html#__init__,Method corebio.db.scop.ClaRecord.__init__()=corebio.db.scop.ClaRecord-class.html#__init__,Method corebio.db.scop.DesRecord.__init__()=corebio.db.scop.DesRecord-class.html#__init__,Method corebio.db.scop.DomRecord.__init__()=corebio.db.scop.DomRecord-class.html#__init__,Method corebio.db.scop.Domain.__init__()=corebio.db.scop.Domain-class.html#__init__,Method corebio.db.scop.HieRecord.__init__()=corebio.db.scop.HieRecord-class.html#__init__,Method corebio.db.scop.Node.__init__()=corebio.db.scop.Node-class.html#__init__,Method corebio.db.scop.Residues.__init__()=corebio.db.scop.Residues-class.html#__init__,Method corebio.db.scop.Scop.__init__()=corebio.db.scop.Scop-class.html#__init__,Method corebio.matrix.AlphabeticArray.__init__()=corebio.matrix.AlphabeticArray-class.html#__init__,Method corebio.matrix.Motif.__init__()=corebio.matrix.Motif-class.html#__init__,Method corebio.matrix.SubMatrix.__init__()=corebio.matrix.SubMatrix-class.html#__init__,Method corebio.secstruc.dssp.DsspRecord.__init__()=corebio.secstruc.dssp.DsspRecord-class.html#__init__,Method corebio.secstruc.dssp.DsspResidue.__init__()=corebio.secstruc.dssp.DsspResidue-class.html#__init__,Method corebio.secstruc.dssp.RunDssp.__init__()=corebio.secstruc.dssp.RunDssp-class.html#__init__,Method corebio.secstruc.stride.RunStride.__init__()=corebio.secstruc.stride.RunStride-class.html#__init__,Method corebio.secstruc.stride.StrideRecord.__init__()=corebio.secstruc.stride.StrideRecord-class.html#__init__,Method corebio.secstruc.stride.StrideResidue.__init__()=corebio.secstruc.stride.StrideResidue-class.html#__init__,Method corebio.seq.SeqList.__init__()=corebio.seq.SeqList-class.html#__init__,Method corebio.seq_io._nexus.Block.__init__()=corebio.seq_io._nexus.Block-class.html#__init__,Method corebio.seq_io._nexus.CharBuffer.__init__()=corebio.seq_io._nexus.CharBuffer-class.html#__init__,Method corebio.seq_io._nexus.Commandline.__init__()=corebio.seq_io._nexus.Commandline-class.html#__init__,Method corebio.seq_io._nexus.Nexus.__init__()=corebio.seq_io._nexus.Nexus-class.html#__init__,Method corebio.seq_io._nexus.Nodes.Chain.__init__()=corebio.seq_io._nexus.Nodes.Chain-class.html#__init__,Method corebio.seq_io._nexus.Nodes.Node.__init__()=corebio.seq_io._nexus.Nodes.Node-class.html#__init__,Method corebio.seq_io._nexus.StepMatrix.__init__()=corebio.seq_io._nexus.StepMatrix-class.html#__init__,Method corebio.seq_io._nexus.Trees.NodeData.__init__()=corebio.seq_io._nexus.Trees.NodeData-class.html#__init__,Method corebio.seq_io._nexus.Trees.Tree.__init__()=corebio.seq_io._nexus.Trees.Tree-class.html#__init__,Method corebio.transform.GeneticCode.__init__()=corebio.transform.GeneticCode-class.html#__init__,Method corebio.transform.Transform.__init__()=corebio.transform.Transform-class.html#__init__,Method corebio.utils.ArgumentError.__init__()=corebio.utils.ArgumentError-class.html#__init__,Method corebio.utils.FileIndex.__init__()=corebio.utils.FileIndex-class.html#__init__,Method corebio.utils.Reiterate.__init__()=corebio.utils.Reiterate-class.html#__init__,Method corebio.utils.Token.__init__()=corebio.utils.Token-class.html#__init__,Method corebio.utils.deoptparse.DeHelpFormatter.__init__()=corebio.utils.deoptparse.DeHelpFormatter-class.html#__init__,Method corebio.utils.deoptparse.DeOptionParser.__init__()=corebio.utils.deoptparse.DeOptionParser-class.html#__init__,Method corebio.utils.frozendict.__init__()=corebio.utils.frozendict-class.html#__init__,Method logoddslogolib.LogoData.__init__()=logoddslogolib.LogoData-class.html#__init__,Method logoddslogolib.LogoFormat.__init__()=logoddslogolib.LogoFormat-class.html#__init__,Method logoddslogolib.LogoOptions.__init__()=logoddslogolib.LogoOptions-class.html#__init__,Method weblogolib.GhostscriptAPI.__init__()=weblogolib.GhostscriptAPI-class.html#__init__,Method weblogolib.LogoData.__init__()=weblogolib.LogoData-class.html#__init__,Method weblogolib.LogoFormat.__init__()=weblogolib.LogoFormat-class.html#__init__,Method weblogolib.LogoOptions.__init__()=weblogolib.LogoOptions-class.html#__init__,Method weblogolib._cgi.Field.__init__()=weblogolib._cgi.Field-class.html#__init__,Method weblogolib.color.Color.__init__()=weblogolib.color.Color-class.html#__init__,Method weblogolib.colorscheme.ColorGroup.__init__()=weblogolib.colorscheme.ColorGroup-class.html#__init__,Method weblogolib.colorscheme.ColorScheme.__init__()=weblogolib.colorscheme.ColorScheme-class.html#__init__,Method weblogolib.logomath.Dirichlet.__init__()=weblogolib.logomath.Dirichlet-class.html#__init__,Method weblogolib.logomath.Gamma.__init__()=weblogolib.logomath.Gamma-class.html#__init__"><a title="corebio.db.DataSource.__init__
corebio.db.Dbxref.__init__
corebio.db.astral.Raf.__init__
corebio.db.astral.RafSeqMap.__init__
corebio.db.astral.Res.__init__
corebio.db.scop.ClaRecord.__init__
corebio.db.scop.DesRecord.__init__
corebio.db.scop.DomRecord.__init__
corebio.db.scop.Domain.__init__
corebio.db.scop.HieRecord.__init__
corebio.db.scop.Node.__init__
corebio.db.scop.Residues.__init__
corebio.db.scop.Scop.__init__
corebio.matrix.AlphabeticArray.__init__
corebio.matrix.Motif.__init__
corebio.matrix.SubMatrix.__init__
corebio.secstruc.dssp.DsspRecord.__init__
corebio.secstruc.dssp.DsspResidue.__init__
corebio.secstruc.dssp.RunDssp.__init__
corebio.secstruc.stride.RunStride.__init__
corebio.secstruc.stride.StrideRecord.__init__
corebio.secstruc.stride.StrideResidue.__init__
corebio.seq.SeqList.__init__
corebio.seq_io._nexus.Block.__init__
corebio.seq_io._nexus.CharBuffer.__init__
corebio.seq_io._nexus.Commandline.__init__
corebio.seq_io._nexus.Nexus.__init__
corebio.seq_io._nexus.Nodes.Chain.__init__
corebio.seq_io._nexus.Nodes.Node.__init__
corebio.seq_io._nexus.StepMatrix.__init__
corebio.seq_io._nexus.Trees.NodeData.__init__
corebio.seq_io._nexus.Trees.Tree.__init__
corebio.transform.GeneticCode.__init__
corebio.transform.Transform.__init__
corebio.utils.ArgumentError.__init__
corebio.utils.FileIndex.__init__
corebio.utils.Reiterate.__init__
corebio.utils.Token.__init__
corebio.utils.deoptparse.DeHelpFormatter.__init__
corebio.utils.deoptparse.DeOptionParser.__init__
corebio.utils.frozendict.__init__
logoddslogolib.LogoData.__init__
logoddslogolib.LogoFormat.__init__
logoddslogolib.LogoOptions.__init__
weblogolib.GhostscriptAPI.__init__
weblogolib.LogoData.__init__
weblogolib.LogoFormat.__init__
weblogolib.LogoOptions.__init__
weblogolib._cgi.Field.__init__
weblogolib.color.Color.__init__
weblogolib.colorscheme.ColorGroup.__init__
weblogolib.colorscheme.ColorScheme.__init__
weblogolib.logomath.Dirichlet.__init__
weblogolib.logomath.Gamma.__init__" class="py-name" href="#" onclick="return doclink('link-8', '__init__', 'link-8');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dataclass</tt><tt class="py-op">=</tt><tt id="link-9" class="py-name" targets="Module corebio.data=corebio.data-module.html,Variable corebio.utils.Token.data=corebio.utils.Token-class.html#data"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-9', 'data', 'link-9');">data</a></tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__values_are_support</tt><tt class="py-op">=</tt><tt class="py-name">values_are_support</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">max_support</tt><tt class="py-op">=</tt><tt class="py-name">max_support</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">weight</tt><tt class="py-op">=</tt><tt class="py-name">weight</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">rooted</tt><tt class="py-op">=</tt><tt class="py-name">rooted</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Variable corebio.db.DataSource.name=corebio.db.DataSource-class.html#name,Variable corebio.matrix.SubMatrix.name=corebio.matrix.SubMatrix-class.html#name,Variable corebio.seq.SeqList.name=corebio.seq.SeqList-class.html#name,Variable corebio.transform.Transform.name=corebio.transform.Transform-class.html#name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-10', 'name', 'link-10');">name</a></tt><tt class="py-op">=</tt><tt id="link-11" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-11', 'name', 'link-10');">name</a></tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">        <tt class="py-name">root</tt><tt class="py-op">=</tt><tt id="link-12" class="py-name"><a title="corebio.seq_io._nexus.Nodes" class="py-name" href="#" onclick="return doclink('link-12', 'Nodes', 'link-0');">Nodes</a></tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Class corebio.db.scop.Node=corebio.db.scop.Node-class.html,Class corebio.seq_io._nexus.Nodes.Node=corebio.seq_io._nexus.Nodes.Node-class.html"><a title="corebio.db.scop.Node
corebio.seq_io._nexus.Nodes.Node" class="py-name" href="#" onclick="return doclink('link-13', 'Node', 'link-13');">Node</a></tt><tt class="py-op">(</tt><tt id="link-14" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-14', 'data', 'link-9');">data</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Method corebio.seq_io._nexus.Nodes.Chain.add()=corebio.seq_io._nexus.Nodes.Chain-class.html#add,Method corebio.seq_io._nexus.StepMatrix.add()=corebio.seq_io._nexus.StepMatrix-class.html#add"><a title="corebio.seq_io._nexus.Nodes.Chain.add
corebio.seq_io._nexus.StepMatrix.add" class="py-name" href="#" onclick="return doclink('link-15', 'add', 'link-15');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">root</tt><tt class="py-op">)</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">=</tt><tt class="py-name">root</tt><tt class="py-op">.</tt><tt class="py-name">id</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">tree</tt><tt class="py-op">:</tt>    <tt class="py-comment"># use the tree we have</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">            <tt class="py-comment"># if Tree is called from outside Nexus parser, we need to get rid of linebreaks, etc</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">            <tt class="py-name">tree</tt><tt class="py-op">=</tt><tt class="py-name">tree</tt><tt class="py-op">.</tt><tt class="py-name">strip</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'\n'</tt><tt class="py-op">,</tt><tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'\r'</tt><tt class="py-op">,</tt><tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">            <tt class="py-comment"># there's discrepancy whether newick allows semicolons et the end</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">            <tt class="py-name">tree</tt><tt class="py-op">=</tt><tt class="py-name">tree</tt><tt class="py-op">.</tt><tt class="py-name">rstrip</tt><tt class="py-op">(</tt><tt class="py-string">';'</tt><tt class="py-op">)</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_add_subtree</tt><tt class="py-op">(</tt><tt class="py-name">parent_id</tt><tt class="py-op">=</tt><tt class="py-name">root</tt><tt class="py-op">.</tt><tt class="py-name">id</tt><tt class="py-op">,</tt><tt class="py-name">tree</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parse</tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">         </tt>
<a name="Tree._parse"></a><div id="Tree._parse-def"><a name="L71"></a><tt class="py-lineno"> 71</tt> <a class="py-toggle" href="#" id="Tree._parse-toggle" onclick="return toggle('Tree._parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#_parse">_parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">tree</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree._parse-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree._parse-expanded"><a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parses (a,b,c...)[[[xx]:]yy] into subcomponents and travels down recursively."""</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">         </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">tree</tt><tt class="py-op">.</tt><tt class="py-name">count</tt><tt class="py-op">(</tt><tt class="py-string">'('</tt><tt class="py-op">)</tt><tt class="py-op">!=</tt><tt class="py-name">tree</tt><tt class="py-op">.</tt><tt class="py-name">count</tt><tt class="py-op">(</tt><tt class="py-string">')'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-16" class="py-name" targets="Class corebio.seq_io._nexus.Trees.TreeError=corebio.seq_io._nexus.Trees.TreeError-class.html"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-16', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">,</tt> <tt class="py-string">'Parentheses do not match in (sub)tree: '</tt><tt class="py-op">+</tt><tt class="py-name">tree</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">tree</tt><tt class="py-op">.</tt><tt class="py-name">count</tt><tt class="py-op">(</tt><tt class="py-string">'('</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> <tt class="py-comment"># a leaf</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">            <tt class="py-name">colon</tt><tt class="py-op">=</tt><tt class="py-name">tree</tt><tt class="py-op">.</tt><tt class="py-name">rfind</tt><tt class="py-op">(</tt><tt class="py-string">':'</tt><tt class="py-op">)</tt>    </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">colon</tt><tt class="py-op">&gt;</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">tree</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">colon</tt><tt class="py-op">]</tt><tt class="py-op">,</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_get_values</tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">[</tt><tt class="py-name">colon</tt><tt class="py-op">+</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">tree</tt><tt class="py-op">,</tt><tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">            <tt class="py-name">closing</tt><tt class="py-op">=</tt><tt class="py-name">tree</tt><tt class="py-op">.</tt><tt class="py-name">rfind</tt><tt class="py-op">(</tt><tt class="py-string">')'</tt><tt class="py-op">)</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">            <tt class="py-name">val</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_get_values</tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">[</tt><tt class="py-name">closing</tt><tt class="py-op">+</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">val</tt><tt class="py-op">:</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">                <tt class="py-name">val</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">]</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">            <tt class="py-name">subtrees</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">            <tt class="py-name">plevel</tt><tt class="py-op">=</tt><tt class="py-number">0</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">            <tt class="py-name">prev</tt><tt class="py-op">=</tt><tt class="py-number">1</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">p</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-name">closing</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">tree</tt><tt class="py-op">[</tt><tt class="py-name">p</tt><tt class="py-op">]</tt><tt class="py-op">==</tt><tt class="py-string">'('</tt><tt class="py-op">:</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">                    <tt class="py-name">plevel</tt><tt class="py-op">+=</tt><tt class="py-number">1</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">tree</tt><tt class="py-op">[</tt><tt class="py-name">p</tt><tt class="py-op">]</tt><tt class="py-op">==</tt><tt class="py-string">')'</tt><tt class="py-op">:</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">                    <tt class="py-name">plevel</tt><tt class="py-op">-=</tt><tt class="py-number">1</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">tree</tt><tt class="py-op">[</tt><tt class="py-name">p</tt><tt class="py-op">]</tt><tt class="py-op">==</tt><tt class="py-string">','</tt> <tt class="py-keyword">and</tt> <tt class="py-name">plevel</tt><tt class="py-op">==</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">                    <tt class="py-name">subtrees</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Method corebio.db.astral.RafSeqMap.append()=corebio.db.astral.RafSeqMap-class.html#append"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-17', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">[</tt><tt class="py-name">prev</tt><tt class="py-op">:</tt><tt class="py-name">p</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">                    <tt class="py-name">prev</tt><tt class="py-op">=</tt><tt class="py-name">p</tt><tt class="py-op">+</tt><tt class="py-number">1</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">            <tt class="py-name">subtrees</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-18', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">[</tt><tt class="py-name">prev</tt><tt class="py-op">:</tt><tt class="py-name">closing</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">            <tt class="py-name">subclades</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parse</tt><tt class="py-op">(</tt><tt class="py-name">subtree</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">subtree</tt> <tt class="py-keyword">in</tt> <tt class="py-name">subtrees</tt><tt class="py-op">]</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">subclades</tt><tt class="py-op">,</tt><tt class="py-name">val</tt><tt class="py-op">]</tt> </tt>
</div><a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">     </tt>
<a name="Tree._add_subtree"></a><div id="Tree._add_subtree-def"><a name="L102"></a><tt class="py-lineno">102</tt> <a class="py-toggle" href="#" id="Tree._add_subtree-toggle" onclick="return toggle('Tree._add_subtree');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#_add_subtree">_add_subtree</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">parent_id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt><tt class="py-param">tree</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree._add_subtree-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree._add_subtree-expanded"><a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">        <tt class="py-docstring">"""Adds leaf or tree (in newick format) to a parent_id. (self,parent_id,tree)."""</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">         </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">parent_id</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-19" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-19', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Need node_id to connect to.'</tt><tt class="py-op">)</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">st</tt> <tt class="py-keyword">in</tt> <tt class="py-name">tree</tt><tt class="py-op">:</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">type</tt><tt class="py-op">(</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-name">list</tt><tt class="py-op">:</tt> <tt class="py-comment"># it's a subtree</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">                <tt class="py-name">nd</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dataclass</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">&gt;=</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-comment"># if there's two values, support comes first. Is that always so?</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">                    <tt class="py-name">nd</tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">=</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">                        <tt class="py-name">nd</tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">=</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> <tt class="py-comment"># otherwise it could be real branchlengths or support as branchlengths</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__values_are_support</tt><tt class="py-op">:</tt> <tt class="py-comment"># default</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">                            <tt class="py-name">nd</tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">=</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">                        <tt class="py-name">nd</tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">=</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">                <tt class="py-name">sn</tt><tt class="py-op">=</tt><tt id="link-20" class="py-name"><a title="corebio.seq_io._nexus.Nodes" class="py-name" href="#" onclick="return doclink('link-20', 'Nodes', 'link-0');">Nodes</a></tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="corebio.db.scop.Node
corebio.seq_io._nexus.Nodes.Node" class="py-name" href="#" onclick="return doclink('link-21', 'Node', 'link-13');">Node</a></tt><tt class="py-op">(</tt><tt class="py-name">nd</tt><tt class="py-op">)</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.add
corebio.seq_io._nexus.StepMatrix.add" class="py-name" href="#" onclick="return doclink('link-22', 'add', 'link-15');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">sn</tt><tt class="py-op">,</tt><tt class="py-name">parent_id</tt><tt class="py-op">)</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_add_subtree</tt><tt class="py-op">(</tt><tt class="py-name">sn</tt><tt class="py-op">.</tt><tt class="py-name">id</tt><tt class="py-op">,</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> <tt class="py-comment"># it's a leaf</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">                <tt class="py-name">nd</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dataclass</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">                <tt class="py-name">nd</tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">=</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">st</tt><tt class="py-op">)</tt><tt class="py-op">&gt;</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">&gt;=</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-comment"># if there's two values, support comes first. Is that always so?</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">                        <tt class="py-name">nd</tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">=</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">                            <tt class="py-name">nd</tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">=</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> <tt class="py-comment"># otherwise it could be real branchlengths or support as branchlengths</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__values_are_support</tt><tt class="py-op">:</tt> <tt class="py-comment"># default</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">                                <tt class="py-name">nd</tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">=</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">                            <tt class="py-name">nd</tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">=</tt><tt class="py-name">st</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">                <tt class="py-name">leaf</tt><tt class="py-op">=</tt><tt id="link-23" class="py-name"><a title="corebio.seq_io._nexus.Nodes" class="py-name" href="#" onclick="return doclink('link-23', 'Nodes', 'link-0');">Nodes</a></tt><tt class="py-op">.</tt><tt id="link-24" class="py-name"><a title="corebio.db.scop.Node
corebio.seq_io._nexus.Nodes.Node" class="py-name" href="#" onclick="return doclink('link-24', 'Node', 'link-13');">Node</a></tt><tt class="py-op">(</tt><tt class="py-name">nd</tt><tt class="py-op">)</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.add
corebio.seq_io._nexus.StepMatrix.add" class="py-name" href="#" onclick="return doclink('link-25', 'add', 'link-15');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">leaf</tt><tt class="py-op">,</tt><tt class="py-name">parent_id</tt><tt class="py-op">)</tt> </tt>
</div><a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">     </tt>
<a name="Tree._get_values"></a><div id="Tree._get_values-def"><a name="L140"></a><tt class="py-lineno">140</tt> <a class="py-toggle" href="#" id="Tree._get_values-toggle" onclick="return toggle('Tree._get_values');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#_get_values">_get_values</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree._get_values-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree._get_values-expanded"><a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">        <tt class="py-docstring">"""Extracts values (support/branchlength) from xx[:yyy], xx."""</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">        </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">text</tt><tt class="py-op">==</tt><tt class="py-string">''</tt><tt class="py-op">:</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">t</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">t</tt> <tt class="py-keyword">in</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.split()=corebio.seq_io._nexus.Trees.Tree-class.html#split"><a title="corebio.seq_io._nexus.Trees.Tree.split" class="py-name" href="#" onclick="return doclink('link-26', 'split', 'link-26');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">':'</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">t</tt><tt class="py-op">.</tt><tt class="py-name">strip</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt>  </tt>
</div><a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">    </tt>
<a name="Tree._walk"></a><div id="Tree._walk-def"><a name="L147"></a><tt class="py-lineno">147</tt> <a class="py-toggle" href="#" id="Tree._walk-toggle" onclick="return toggle('Tree._walk');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#_walk">_walk</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">node</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree._walk-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree._walk-expanded"><a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return all node_ids downwards from a node."""</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">         </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-27" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.node()=corebio.seq_io._nexus.Trees.Tree-class.html#node"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-27', 'node', 'link-27');">node</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">            <tt id="link-28" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-28', 'node', 'link-27');">node</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-29', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-30" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-30', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">:</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">n</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">sn</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_walk</tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">sn</tt> </tt>
</div><a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"> </tt>
<a name="Tree.node"></a><div id="Tree.node-def"><a name="L157"></a><tt class="py-lineno">157</tt> <a class="py-toggle" href="#" id="Tree.node-toggle" onclick="return toggle('Tree.node');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#node">node</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">node_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.node-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.node-expanded"><a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the instance of node_id.</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line"><tt class="py-docstring">        node = node(self,node_id)</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">node_id</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">:</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-31" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-31', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Unknown node_id: %d'</tt> <tt class="py-op">%</tt> <tt class="py-name">node_id</tt><tt class="py-op">)</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">[</tt><tt class="py-name">node_id</tt><tt class="py-op">]</tt> </tt>
</div><a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line"> </tt>
<a name="Tree.split"></a><div id="Tree.split-def"><a name="L166"></a><tt class="py-lineno">166</tt> <a class="py-toggle" href="#" id="Tree.split-toggle" onclick="return toggle('Tree.split');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#split">split</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">parent_id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt><tt class="py-param">n</tt><tt class="py-op">=</tt><tt class="py-number">2</tt><tt class="py-op">,</tt><tt class="py-param">branchlength</tt><tt class="py-op">=</tt><tt class="py-number">1.0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.split-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.split-expanded"><a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">        <tt class="py-docstring">"""Speciation: generates n (default two) descendants of a node.</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line"><tt class="py-docstring">        [new ids] = split(self,parent_id=None,n=2,branchlength=1.0):</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>  </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">parent_id</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-32" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-32', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Missing node_id.'</tt><tt class="py-op">)</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">        <tt class="py-name">ids</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">        <tt class="py-name">parent_data</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">[</tt><tt class="py-name">parent_id</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-33', 'data', 'link-9');">data</a></tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">i</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">            <tt id="link-34" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-34', 'node', 'link-27');">node</a></tt><tt class="py-op">=</tt><tt id="link-35" class="py-name"><a title="corebio.seq_io._nexus.Nodes" class="py-name" href="#" onclick="return doclink('link-35', 'Nodes', 'link-0');">Nodes</a></tt><tt class="py-op">.</tt><tt id="link-36" class="py-name"><a title="corebio.db.scop.Node
corebio.seq_io._nexus.Nodes.Node" class="py-name" href="#" onclick="return doclink('link-36', 'Node', 'link-13');">Node</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">parent_data</tt><tt class="py-op">:</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">                <tt id="link-37" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-37', 'node', 'link-27');">node</a></tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-38', 'data', 'link-9');">data</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dataclass</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">                <tt class="py-comment"># each node has taxon and branchlength attribute</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">parent_data</tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">:</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">                    <tt id="link-39" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-39', 'node', 'link-27');">node</a></tt><tt class="py-op">.</tt><tt id="link-40" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-40', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">=</tt><tt class="py-name">parent_data</tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">+</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">i</tt><tt class="py-op">)</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">                <tt id="link-41" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-41', 'node', 'link-27');">node</a></tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-42', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">=</tt><tt class="py-name">branchlength</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">            <tt class="py-name">ids</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-43', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.add
corebio.seq_io._nexus.StepMatrix.add" class="py-name" href="#" onclick="return doclink('link-44', 'add', 'link-15');">add</a></tt><tt class="py-op">(</tt><tt id="link-45" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-45', 'node', 'link-27');">node</a></tt><tt class="py-op">,</tt><tt class="py-name">parent_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">ids</tt> </tt>
</div><a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line"> </tt>
<a name="Tree.search_taxon"></a><div id="Tree.search_taxon-def"><a name="L186"></a><tt class="py-lineno">186</tt> <a class="py-toggle" href="#" id="Tree.search_taxon-toggle" onclick="return toggle('Tree.search_taxon');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#search_taxon">search_taxon</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">taxon</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.search_taxon-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.search_taxon-expanded"><a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns the first matching taxon in self.data.taxon. Not restricted to terminal nodes.</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line"><tt class="py-docstring">        node_id = search_taxon(self,taxon)</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">id</tt><tt class="py-op">,</tt><tt id="link-46" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-46', 'node', 'link-27');">node</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">.</tt><tt class="py-name">items</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-47" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-47', 'node', 'link-27');">node</a></tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-48', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">==</tt><tt class="py-name">taxon</tt><tt class="py-op">:</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">id</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">    </tt>
<a name="Tree.prune"></a><div id="Tree.prune-def"><a name="L196"></a><tt class="py-lineno">196</tt> <a class="py-toggle" href="#" id="Tree.prune-toggle" onclick="return toggle('Tree.prune');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#prune">prune</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">taxon</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.prune-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.prune-expanded"><a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">        <tt class="py-docstring">"""Prunes a terminal taxon from the tree.</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"><tt class="py-docstring">        id_of_previous_node = prune(self,taxon)</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line"><tt class="py-docstring">        If taxon is from a bifurcation, the connectiong node will be collapsed</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"><tt class="py-docstring">        and its branchlength added to remaining terminal node. This might be no</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"><tt class="py-docstring">        longer a meaningful value'</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">         </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">        <tt class="py-name">id</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.search_taxon()=corebio.seq_io._nexus.Trees.Tree-class.html#search_taxon"><a title="corebio.seq_io._nexus.Trees.Tree.search_taxon" class="py-name" href="#" onclick="return doclink('link-49', 'search_taxon', 'link-49');">search_taxon</a></tt><tt class="py-op">(</tt><tt class="py-name">taxon</tt><tt class="py-op">)</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">id</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-50" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-50', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Taxon not found: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">taxon</tt><tt class="py-op">)</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">id</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.get_terminals()=corebio.seq_io._nexus.Trees.Tree-class.html#get_terminals"><a title="corebio.seq_io._nexus.Trees.Tree.get_terminals" class="py-name" href="#" onclick="return doclink('link-51', 'get_terminals', 'link-51');">get_terminals</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-52" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-52', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Not a terminal taxon: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">taxon</tt><tt class="py-op">)</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">            <tt class="py-name">prev</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name" targets="Method corebio.seq_io._nexus.Nodes.Chain.unlink()=corebio.seq_io._nexus.Nodes.Chain-class.html#unlink"><a title="corebio.seq_io._nexus.Nodes.Chain.unlink" class="py-name" href="#" onclick="return doclink('link-53', 'unlink', 'link-53');">unlink</a></tt><tt class="py-op">(</tt><tt class="py-name">id</tt><tt class="py-op">)</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name" targets="Method corebio.seq_io._nexus.Nodes.Chain.kill()=corebio.seq_io._nexus.Nodes.Chain-class.html#kill"><a title="corebio.seq_io._nexus.Nodes.Chain.kill" class="py-name" href="#" onclick="return doclink('link-54', 'kill', 'link-54');">kill</a></tt><tt class="py-op">(</tt><tt class="py-name">id</tt><tt class="py-op">)</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">prev</tt><tt class="py-op">==</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-55', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">prev</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">                <tt class="py-name">succ</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-56', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">prev</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">                <tt class="py-name">new_bl</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-57', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">prev</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-58', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">+</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-59', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-60', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name" targets="Method corebio.seq_io._nexus.Nodes.Chain.collapse()=corebio.seq_io._nexus.Nodes.Chain-class.html#collapse"><a title="corebio.seq_io._nexus.Nodes.Chain.collapse" class="py-name" href="#" onclick="return doclink('link-61', 'collapse', 'link-61');">collapse</a></tt><tt class="py-op">(</tt><tt class="py-name">prev</tt><tt class="py-op">)</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-62', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-63', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">=</tt><tt class="py-name">new_bl</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">prev</tt> </tt>
</div><a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">         </tt>
<a name="Tree.get_taxa"></a><div id="Tree.get_taxa-def"><a name="L220"></a><tt class="py-lineno">220</tt> <a class="py-toggle" href="#" id="Tree.get_taxa-toggle" onclick="return toggle('Tree.get_taxa');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#get_taxa">get_taxa</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">node_id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.get_taxa-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.get_taxa-expanded"><a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list of all otus downwards from a node (self, node_id).</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">        nodes = get_taxa(self,node_id=None)</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">node_id</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">            <tt class="py-name">node_id</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">node_id</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">:</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-64" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-64', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Unknown node_id: %d.'</tt> <tt class="py-op">%</tt> <tt class="py-name">node_id</tt><tt class="py-op">)</tt>     </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">[</tt><tt class="py-name">node_id</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">==</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">[</tt><tt class="py-name">node_id</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-65', 'data', 'link-9');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">[</tt><tt class="py-name">node_id</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-66', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">]</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">            <tt class="py-name">list</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">succ</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">[</tt><tt class="py-name">node_id</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">:</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">                <tt class="py-name">list</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name" targets="Method corebio.db.astral.RafSeqMap.extend()=corebio.db.astral.RafSeqMap-class.html#extend"><a title="corebio.db.astral.RafSeqMap.extend" class="py-name" href="#" onclick="return doclink('link-67', 'extend', 'link-67');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.get_taxa()=corebio.seq_io._nexus.Trees.Tree-class.html#get_taxa"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-68', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">list</tt> </tt>
</div><a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"> </tt>
<a name="Tree.get_terminals"></a><div id="Tree.get_terminals-def"><a name="L241"></a><tt class="py-lineno">241</tt> <a class="py-toggle" href="#" id="Tree.get_terminals-toggle" onclick="return toggle('Tree.get_terminals');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#get_terminals">get_terminals</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.get_terminals-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.get_terminals-expanded"><a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list of all terminal nodes."""</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">i</tt> <tt class="py-keyword">for</tt> <tt class="py-name">i</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name" targets="Method corebio.seq_io._nexus.Nodes.Chain.all_ids()=corebio.seq_io._nexus.Nodes.Chain-class.html#all_ids"><a title="corebio.seq_io._nexus.Nodes.Chain.all_ids" class="py-name" href="#" onclick="return doclink('link-69', 'all_ids', 'link-69');">all_ids</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-70', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">i</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">==</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
</div><a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"> </tt>
<a name="Tree.sum_branchlength"></a><div id="Tree.sum_branchlength-def"><a name="L245"></a><tt class="py-lineno">245</tt> <a class="py-toggle" href="#" id="Tree.sum_branchlength-toggle" onclick="return toggle('Tree.sum_branchlength');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#sum_branchlength">sum_branchlength</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">root</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt><tt class="py-param">node</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.sum_branchlength-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.sum_branchlength-expanded"><a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">        <tt class="py-docstring">"""Adds up the branchlengths from root (default self.root) to node.</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line"><tt class="py-docstring">        sum = sum_branchlength(self,root=None,node=None)</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line"> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">root</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">            <tt class="py-name">root</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-71" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-71', 'node', 'link-27');">node</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-72" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-72', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Missing node id.'</tt><tt class="py-op">)</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">        <tt class="py-name">blen</tt><tt class="py-op">=</tt><tt class="py-number">0.0</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt id="link-73" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-73', 'node', 'link-27');">node</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt id="link-74" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-74', 'node', 'link-27');">node</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">root</tt><tt class="py-op">:</tt>  </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">            <tt class="py-name">blen</tt><tt class="py-op">+=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-75', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-76" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-76', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-77', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">            <tt id="link-78" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-78', 'node', 'link-27');">node</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-79', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-80" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-80', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">prev</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">blen</tt> </tt>
</div><a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line"> </tt>
<a name="Tree.set_subtree"></a><div id="Tree.set_subtree-def"><a name="L261"></a><tt class="py-lineno">261</tt> <a class="py-toggle" href="#" id="Tree.set_subtree-toggle" onclick="return toggle('Tree.set_subtree');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#set_subtree">set_subtree</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">node</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.set_subtree-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.set_subtree-expanded"><a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return subtree as a set of nested sets.</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line"><tt class="py-docstring">        sets = set_subtree(self,node)</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">         </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-81', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-82" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-82', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">==</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-83', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-84" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-84', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-85', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-86" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-86', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.set_subtree()=corebio.seq_io._nexus.Trees.Tree-class.html#set_subtree"><a title="corebio.seq_io._nexus.Trees.Tree.set_subtree" class="py-name" href="#" onclick="return doclink('link-87', 'set_subtree', 'link-87');">set_subtree</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-88', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-89" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-89', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">             </tt>
<a name="Tree.is_identical"></a><div id="Tree.is_identical-def"><a name="L272"></a><tt class="py-lineno">272</tt> <a class="py-toggle" href="#" id="Tree.is_identical-toggle" onclick="return toggle('Tree.is_identical');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#is_identical">is_identical</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">tree2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.is_identical-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.is_identical-expanded"><a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">        <tt class="py-docstring">"""Compare tree and tree2 for identity.</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line"><tt class="py-docstring">        result = is_identical(self,tree2)</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.set_subtree" class="py-name" href="#" onclick="return doclink('link-90', 'set_subtree', 'link-87');">set_subtree</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.set_subtree" class="py-name" href="#" onclick="return doclink('link-91', 'set_subtree', 'link-87');">set_subtree</a></tt><tt class="py-op">(</tt><tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">)</tt> </tt>
</div><a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line"> </tt>
<a name="Tree.is_compatible"></a><div id="Tree.is_compatible-def"><a name="L279"></a><tt class="py-lineno">279</tt> <a class="py-toggle" href="#" id="Tree.is_compatible-toggle" onclick="return toggle('Tree.is_compatible');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#is_compatible">is_compatible</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">tree2</tt><tt class="py-op">,</tt><tt class="py-param">threshold</tt><tt class="py-op">,</tt><tt class="py-param">strict</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.is_compatible-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.is_compatible-expanded"><a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">        <tt class="py-docstring">"""Compares branches with support&gt;threshold for compatibility.</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line"><tt class="py-docstring">        result = is_compatible(self,tree2,threshold)</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line"> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">        <tt class="py-comment"># check if both trees have the same set of taxa. strict=True enforces this.</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">        <tt class="py-name">missing2</tt><tt class="py-op">=</tt><tt id="link-92" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-92', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-93', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">-</tt><tt id="link-94" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-94', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-95', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt class="py-name">missing1</tt><tt class="py-op">=</tt><tt id="link-96" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-96', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-97', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">-</tt><tt id="link-98" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-98', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-99', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">strict</tt> <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt class="py-name">missing1</tt> <tt class="py-keyword">or</tt> <tt class="py-name">missing2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">missing1</tt><tt class="py-op">:</tt>  </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">                <tt class="py-keyword">print</tt> <tt class="py-string">'Taxon/taxa %s is/are missing in tree %s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name" targets="Method corebio.seq.Seq.join()=corebio.seq.Seq-class.html#join"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-100', 'join', 'link-100');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">missing1</tt><tt class="py-op">)</tt> <tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-101" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-101', 'name', 'link-10');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">missing2</tt><tt class="py-op">:</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">                <tt class="py-keyword">print</tt> <tt class="py-string">'Taxon/taxa %s is/are missing in tree %s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-102', 'join', 'link-100');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">missing2</tt><tt class="py-op">)</tt> <tt class="py-op">,</tt> <tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-103', 'name', 'link-10');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-104" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-104', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">,</tt> <tt class="py-string">'Can\'t compare trees with different taxon compositions.'</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">        <tt class="py-name">t1</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt id="link-105" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-105', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-106', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-107', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-108', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.all_ids" class="py-name" href="#" onclick="return doclink('link-109', 'all_ids', 'link-69');">all_ids</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> \ </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-110', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt> <tt class="py-keyword">and</tt>\ </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">            <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-111', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-112', 'data', 'link-9');">data</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-113', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-114', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt> <tt class="py-keyword">and</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-115" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-115', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-116', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">&gt;=</tt><tt class="py-name">threshold</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">        <tt class="py-name">t2</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt id="link-117" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-117', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-118', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-119', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-120', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.all_ids" class="py-name" href="#" onclick="return doclink('link-121', 'all_ids', 'link-69');">all_ids</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> \ </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">            <tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-122', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt> <tt class="py-keyword">and</tt>\ </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">            <tt class="py-op">(</tt><tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-123', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-124', 'data', 'link-9');">data</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-125', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-126', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt> <tt class="py-keyword">and</tt> <tt class="py-name">tree2</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-127', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-128', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">&gt;=</tt><tt class="py-name">threshold</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">        <tt class="py-name">conflict</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-op">(</tt><tt class="py-name">st1</tt><tt class="py-op">,</tt><tt class="py-name">sup1</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">t1</tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-op">(</tt><tt class="py-name">st2</tt><tt class="py-op">,</tt><tt class="py-name">sup2</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">t2</tt><tt class="py-op">:</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">st1</tt><tt class="py-op">.</tt><tt class="py-name">issubset</tt><tt class="py-op">(</tt><tt class="py-name">st2</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">st2</tt><tt class="py-op">.</tt><tt class="py-name">issubset</tt><tt class="py-op">(</tt><tt class="py-name">st1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>                     <tt class="py-comment"># don't hiccup on upstream nodes</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">                    <tt class="py-name">intersect</tt><tt class="py-op">,</tt><tt class="py-name">notin1</tt><tt class="py-op">,</tt><tt class="py-name">notin2</tt><tt class="py-op">=</tt><tt class="py-name">st1</tt> <tt class="py-op">&amp;</tt> <tt class="py-name">st2</tt><tt class="py-op">,</tt> <tt class="py-name">st2</tt><tt class="py-op">-</tt><tt class="py-name">st1</tt><tt class="py-op">,</tt> <tt class="py-name">st1</tt><tt class="py-op">-</tt><tt class="py-name">st2</tt>                 <tt class="py-comment"># all three are non-empty sets</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">                    <tt class="py-comment"># if notin1==missing1 or notin2==missing2  &lt;==&gt; st1.issubset(st2) or st2.issubset(st1) ???</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">intersect</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">notin1</tt><tt class="py-op">.</tt><tt class="py-name">issubset</tt><tt class="py-op">(</tt><tt class="py-name">missing1</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">notin2</tt><tt class="py-op">.</tt><tt class="py-name">issubset</tt><tt class="py-op">(</tt><tt class="py-name">missing2</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>         <tt class="py-comment"># omit conflicts due to missing taxa</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">                        <tt class="py-name">conflict</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-129', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">st1</tt><tt class="py-op">,</tt><tt class="py-name">sup1</tt><tt class="py-op">,</tt><tt class="py-name">st2</tt><tt class="py-op">,</tt><tt class="py-name">sup2</tt><tt class="py-op">,</tt><tt class="py-name">intersect</tt><tt class="py-op">,</tt><tt class="py-name">notin1</tt><tt class="py-op">,</tt><tt class="py-name">notin2</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">conflict</tt> </tt>
</div><a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">         </tt>
<a name="Tree.common_ancestor"></a><div id="Tree.common_ancestor-def"><a name="L310"></a><tt class="py-lineno">310</tt> <a class="py-toggle" href="#" id="Tree.common_ancestor-toggle" onclick="return toggle('Tree.common_ancestor');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#common_ancestor">common_ancestor</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">node1</tt><tt class="py-op">,</tt><tt class="py-param">node2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.common_ancestor-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.common_ancestor-expanded"><a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the common ancestor that connects to nodes.</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line"><tt class="py-docstring">        node_id = common_ancestor(self,node1,node2)</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">         </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">        <tt class="py-name">l1</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">]</tt><tt class="py-op">+</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name" targets="Method corebio.seq_io._nexus.Nodes.Chain.trace()=corebio.seq_io._nexus.Nodes.Chain-class.html#trace"><a title="corebio.seq_io._nexus.Nodes.Chain.trace" class="py-name" href="#" onclick="return doclink('link-130', 'trace', 'link-130');">trace</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">,</tt><tt class="py-name">node1</tt><tt class="py-op">)</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-name">l2</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">]</tt><tt class="py-op">+</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.trace" class="py-name" href="#" onclick="return doclink('link-131', 'trace', 'link-130');">trace</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">,</tt><tt class="py-name">node2</tt><tt class="py-op">)</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">n</tt> <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt class="py-name">l1</tt> <tt class="py-keyword">if</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt class="py-name">l2</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
</div><a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line"> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line"> </tt>
<a name="Tree.distance"></a><div id="Tree.distance-def"><a name="L321"></a><tt class="py-lineno">321</tt> <a class="py-toggle" href="#" id="Tree.distance-toggle" onclick="return toggle('Tree.distance');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#distance">distance</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">node1</tt><tt class="py-op">,</tt><tt class="py-param">node2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.distance-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.distance-expanded"><a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add and return the sum of the branchlengths between two nodes.</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line"><tt class="py-docstring">        dist = distance(self,node1,node2)</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">         </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">        <tt class="py-name">ca</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.common_ancestor()=corebio.seq_io._nexus.Trees.Tree-class.html#common_ancestor"><a title="corebio.seq_io._nexus.Trees.Tree.common_ancestor" class="py-name" href="#" onclick="return doclink('link-132', 'common_ancestor', 'link-132');">common_ancestor</a></tt><tt class="py-op">(</tt><tt class="py-name">node1</tt><tt class="py-op">,</tt><tt class="py-name">node2</tt><tt class="py-op">)</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.sum_branchlength()=corebio.seq_io._nexus.Trees.Tree-class.html#sum_branchlength"><a title="corebio.seq_io._nexus.Trees.Tree.sum_branchlength" class="py-name" href="#" onclick="return doclink('link-133', 'sum_branchlength', 'link-133');">sum_branchlength</a></tt><tt class="py-op">(</tt><tt class="py-name">ca</tt><tt class="py-op">,</tt><tt class="py-name">node1</tt><tt class="py-op">)</tt><tt class="py-op">+</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-134" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.sum_branchlength" class="py-name" href="#" onclick="return doclink('link-134', 'sum_branchlength', 'link-133');">sum_branchlength</a></tt><tt class="py-op">(</tt><tt class="py-name">ca</tt><tt class="py-op">,</tt><tt class="py-name">node2</tt><tt class="py-op">)</tt> </tt>
</div><a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line"> </tt>
<a name="Tree.is_monophyletic"></a><div id="Tree.is_monophyletic-def"><a name="L329"></a><tt class="py-lineno">329</tt> <a class="py-toggle" href="#" id="Tree.is_monophyletic-toggle" onclick="return toggle('Tree.is_monophyletic');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#is_monophyletic">is_monophyletic</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">taxon_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.is_monophyletic-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.is_monophyletic-expanded"><a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return node_id of common ancestor if taxon_list is monophyletic, -1 otherwise.</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line"><tt class="py-docstring">        result = is_monophyletic(self,taxon_list)</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">taxon_list</tt><tt class="py-op">,</tt><tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">            <tt class="py-name">taxon_set</tt><tt class="py-op">=</tt><tt id="link-135" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-135', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">taxon_list</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">            <tt class="py-name">taxon_set</tt><tt class="py-op">=</tt><tt id="link-136" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-136', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">taxon_list</tt><tt class="py-op">)</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">        <tt class="py-name">node_id</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">            <tt class="py-name">subclade_taxa</tt><tt class="py-op">=</tt><tt id="link-137" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-137', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-138', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-name">node_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">subclade_taxa</tt><tt class="py-op">==</tt><tt class="py-name">taxon_set</tt><tt class="py-op">:</tt>                                        <tt class="py-comment"># are we there?</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">node_id</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt>                                                               <tt class="py-comment"># check subnodes</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">subnode</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">[</tt><tt class="py-name">node_id</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">:</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-139" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-139', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-140', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-name">subnode</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">issuperset</tt><tt class="py-op">(</tt><tt class="py-name">taxon_set</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># taxon_set is downstream</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">                        <tt class="py-name">node_id</tt><tt class="py-op">=</tt><tt class="py-name">subnode</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">                        <tt class="py-keyword">break</tt>   <tt class="py-comment"># out of for loop</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt>   <tt class="py-comment"># taxon set was not with successors, for loop exhausted</tt> </tt>
</div><a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line"> </tt>
<a name="Tree.is_bifurcating"></a><div id="Tree.is_bifurcating-def"><a name="L351"></a><tt class="py-lineno">351</tt> <a class="py-toggle" href="#" id="Tree.is_bifurcating-toggle" onclick="return toggle('Tree.is_bifurcating');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#is_bifurcating">is_bifurcating</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">node</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.is_bifurcating-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.is_bifurcating-expanded"><a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return True if tree downstream of node is strictly bifurcating."""</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-141" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-141', 'node', 'link-27');">node</a></tt><tt class="py-op">:</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">            <tt id="link-142" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-142', 'node', 'link-27');">node</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-143" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-143', 'node', 'link-27');">node</a></tt><tt class="py-op">==</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-144', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-145" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-145', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-comment">#root can be trifurcating, because it has no ancestor</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-146" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.is_bifurcating()=corebio.seq_io._nexus.Trees.Tree-class.html#is_bifurcating"><a title="corebio.seq_io._nexus.Trees.Tree.is_bifurcating" class="py-name" href="#" onclick="return doclink('link-146', 'is_bifurcating', 'link-146');">is_bifurcating</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-147" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-147', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-148" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-148', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-149" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.is_bifurcating" class="py-name" href="#" onclick="return doclink('link-149', 'is_bifurcating', 'link-146');">is_bifurcating</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-150', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-151" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-151', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.is_bifurcating" class="py-name" href="#" onclick="return doclink('link-152', 'is_bifurcating', 'link-146');">is_bifurcating</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-153', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-154" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-154', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-155', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-156" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-156', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.is_bifurcating" class="py-name" href="#" onclick="return doclink('link-157', 'is_bifurcating', 'link-146');">is_bifurcating</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-158" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-158', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-159" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-159', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-160" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.is_bifurcating" class="py-name" href="#" onclick="return doclink('link-160', 'is_bifurcating', 'link-146');">is_bifurcating</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-161" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-161', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-162" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-162', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-163" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-163', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-164" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-164', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line"> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line"> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line"> </tt>
<a name="Tree.branchlength2support"></a><div id="Tree.branchlength2support-def"><a name="L368"></a><tt class="py-lineno">368</tt> <a class="py-toggle" href="#" id="Tree.branchlength2support-toggle" onclick="return toggle('Tree.branchlength2support');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#branchlength2support">branchlength2support</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.branchlength2support-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.branchlength2support-expanded"><a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt class="py-docstring">"""Move values stored in data.branchlength to data.support, and set branchlength to 0.0</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line"><tt class="py-docstring">        This is necessary when support has been stored as branchlength (e.g. paup), and has thus</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line"><tt class="py-docstring">        been read in as branchlength. </tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line"> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">.</tt><tt class="py-name">keys</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-165', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-166', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-167', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-168', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-169', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-170" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-170', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">=</tt><tt class="py-number">0.0</tt> </tt>
</div><a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line"> </tt>
<a name="Tree.convert_absolute_support"></a><div id="Tree.convert_absolute_support-def"><a name="L379"></a><tt class="py-lineno">379</tt> <a class="py-toggle" href="#" id="Tree.convert_absolute_support-toggle" onclick="return toggle('Tree.convert_absolute_support');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#convert_absolute_support">convert_absolute_support</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">nrep</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.convert_absolute_support-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.convert_absolute_support-expanded"><a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">        <tt class="py-docstring">"""Convert absolute support (clade-count) to rel. frequencies.</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line"><tt class="py-docstring">        Some software (e.g. PHYLIP consense) just calculate how often clades appear, instead of</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line"><tt class="py-docstring">        calculating relative frequencies."""</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line"> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_walk</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-171', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-172', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">:</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-173" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-173', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-174', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">/=</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">nrep</tt><tt class="py-op">)</tt> </tt>
</div><a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line"> </tt>
<a name="Tree.randomize"></a><div id="Tree.randomize-def"><a name="L389"></a><tt class="py-lineno">389</tt> <a class="py-toggle" href="#" id="Tree.randomize-toggle" onclick="return toggle('Tree.randomize');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#randomize">randomize</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">ntax</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt><tt class="py-param">taxon_list</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt><tt class="py-param">branchlength</tt><tt class="py-op">=</tt><tt class="py-number">1.0</tt><tt class="py-op">,</tt><tt class="py-param">branchlength_sd</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt><tt class="py-param">bifurcate</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.randomize-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.randomize-expanded"><a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">        <tt class="py-docstring">"""Generates a random tree with ntax taxa and/or taxa from taxlabels.</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line"><tt class="py-docstring">        new_tree = randomize(self,ntax=None,taxon_list=None,branchlength=1.0,branchlength_sd=None,bifurcate=True)</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line"><tt class="py-docstring">        Trees are bifurcating by default. (Polytomies not yet supported).</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line"> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">ntax</tt> <tt class="py-keyword">and</tt> <tt class="py-name">taxon_list</tt><tt class="py-op">:</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">            <tt class="py-name">ntax</tt><tt class="py-op">=</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">taxon_list</tt><tt class="py-op">)</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">taxon_list</tt> <tt class="py-keyword">and</tt> <tt class="py-name">ntax</tt><tt class="py-op">:</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">            <tt class="py-name">taxon_list</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-string">'taxon'</tt><tt class="py-op">+</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">i</tt><tt class="py-op">+</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">i</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-name">ntax</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">ntax</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">taxon_list</tt><tt class="py-op">:</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-175" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-175', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Either numer of taxa or list of taxa must be specified.'</tt><tt class="py-op">)</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">ntax</tt><tt class="py-op">!=</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">taxon_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-176" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-176', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Length of taxon list must correspond to ntax.'</tt><tt class="py-op">)</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">        <tt class="py-comment"># initiate self with empty root</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="corebio.db.DataSource.__init__
corebio.db.Dbxref.__init__
corebio.db.astral.Raf.__init__
corebio.db.astral.RafSeqMap.__init__
corebio.db.astral.Res.__init__
corebio.db.scop.ClaRecord.__init__
corebio.db.scop.DesRecord.__init__
corebio.db.scop.DomRecord.__init__
corebio.db.scop.Domain.__init__
corebio.db.scop.HieRecord.__init__
corebio.db.scop.Node.__init__
corebio.db.scop.Residues.__init__
corebio.db.scop.Scop.__init__
corebio.matrix.AlphabeticArray.__init__
corebio.matrix.Motif.__init__
corebio.matrix.SubMatrix.__init__
corebio.secstruc.dssp.DsspRecord.__init__
corebio.secstruc.dssp.DsspResidue.__init__
corebio.secstruc.dssp.RunDssp.__init__
corebio.secstruc.stride.RunStride.__init__
corebio.secstruc.stride.StrideRecord.__init__
corebio.secstruc.stride.StrideResidue.__init__
corebio.seq.SeqList.__init__
corebio.seq_io._nexus.Block.__init__
corebio.seq_io._nexus.CharBuffer.__init__
corebio.seq_io._nexus.Commandline.__init__
corebio.seq_io._nexus.Nexus.__init__
corebio.seq_io._nexus.Nodes.Chain.__init__
corebio.seq_io._nexus.Nodes.Node.__init__
corebio.seq_io._nexus.StepMatrix.__init__
corebio.seq_io._nexus.Trees.NodeData.__init__
corebio.seq_io._nexus.Trees.Tree.__init__
corebio.transform.GeneticCode.__init__
corebio.transform.Transform.__init__
corebio.utils.ArgumentError.__init__
corebio.utils.FileIndex.__init__
corebio.utils.Reiterate.__init__
corebio.utils.Token.__init__
corebio.utils.deoptparse.DeHelpFormatter.__init__
corebio.utils.deoptparse.DeOptionParser.__init__
corebio.utils.frozendict.__init__
logoddslogolib.LogoData.__init__
logoddslogolib.LogoFormat.__init__
logoddslogolib.LogoOptions.__init__
weblogolib.GhostscriptAPI.__init__
weblogolib.LogoData.__init__
weblogolib.LogoFormat.__init__
weblogolib.LogoOptions.__init__
weblogolib._cgi.Field.__init__
weblogolib.color.Color.__init__
weblogolib.colorscheme.ColorGroup.__init__
weblogolib.colorscheme.ColorScheme.__init__
weblogolib.logomath.Dirichlet.__init__
weblogolib.logomath.Gamma.__init__" class="py-name" href="#" onclick="return doclink('link-177', '__init__', 'link-8');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">        <tt class="py-name">terminals</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_terminals" class="py-name" href="#" onclick="return doclink('link-178', 'get_terminals', 'link-51');">get_terminals</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">        <tt class="py-comment"># bifurcate randomly at terminal nodes until ntax is reached</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">terminals</tt><tt class="py-op">)</tt><tt class="py-op">&lt;</tt><tt class="py-name">ntax</tt><tt class="py-op">:</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">            <tt class="py-name">newsplit</tt><tt class="py-op">=</tt><tt class="py-name">random</tt><tt class="py-op">.</tt><tt class="py-name">choice</tt><tt class="py-op">(</tt><tt class="py-name">terminals</tt><tt class="py-op">)</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">            <tt class="py-name">new_terminals</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.split" class="py-name" href="#" onclick="return doclink('link-179', 'split', 'link-26');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">parent_id</tt><tt class="py-op">=</tt><tt class="py-name">newsplit</tt><tt class="py-op">,</tt><tt class="py-name">branchlength</tt><tt class="py-op">=</tt><tt class="py-name">branchlength</tt><tt class="py-op">)</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">            <tt class="py-comment"># if desired, give some variation to the branch length</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">branchlength_sd</tt><tt class="py-op">:</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">nt</tt> <tt class="py-keyword">in</tt> <tt class="py-name">new_terminals</tt><tt class="py-op">:</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">                    <tt class="py-name">bl</tt><tt class="py-op">=</tt><tt class="py-name">random</tt><tt class="py-op">.</tt><tt class="py-name">gauss</tt><tt class="py-op">(</tt><tt class="py-name">branchlength</tt><tt class="py-op">,</tt><tt class="py-name">branchlength_sd</tt><tt class="py-op">)</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">bl</tt><tt class="py-op">&lt;</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">                        <tt class="py-name">bl</tt><tt class="py-op">=</tt><tt class="py-number">0</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-180', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">nt</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-181" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-181', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">=</tt><tt class="py-name">bl</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">            <tt class="py-name">terminals</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="corebio.db.astral.RafSeqMap.extend" class="py-name" href="#" onclick="return doclink('link-182', 'extend', 'link-67');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">new_terminals</tt><tt class="py-op">)</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">            <tt class="py-name">terminals</tt><tt class="py-op">.</tt><tt id="link-183" class="py-name" targets="Method corebio.seq.Seq.remove()=corebio.seq.Seq-class.html#remove"><a title="corebio.seq.Seq.remove" class="py-name" href="#" onclick="return doclink('link-183', 'remove', 'link-183');">remove</a></tt><tt class="py-op">(</tt><tt class="py-name">newsplit</tt><tt class="py-op">)</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">        <tt class="py-comment"># distribute taxon labels randomly</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">        <tt class="py-name">random</tt><tt class="py-op">.</tt><tt class="py-name">shuffle</tt><tt class="py-op">(</tt><tt class="py-name">taxon_list</tt><tt class="py-op">)</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-op">(</tt><tt id="link-184" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-184', 'node', 'link-27');">node</a></tt><tt class="py-op">,</tt><tt id="link-185" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-185', 'name', 'link-10');">name</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">terminals</tt><tt class="py-op">,</tt><tt class="py-name">taxon_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-186', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-187" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-187', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-188', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">=</tt><tt id="link-189" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-189', 'name', 'link-10');">name</a></tt> </tt>
</div><a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line"> </tt>
<a name="Tree.display"></a><div id="Tree.display-def"><a name="L425"></a><tt class="py-lineno">425</tt> <a class="py-toggle" href="#" id="Tree.display-toggle" onclick="return toggle('Tree.display');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#display">display</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.display-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.display-expanded"><a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">        <tt class="py-docstring">"""Quick and dirty lists of all nodes."""</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">        <tt id="link-190" class="py-name" targets="Variable corebio.transform.GeneticCode.table=corebio.transform.GeneticCode-class.html#table,Variable corebio.transform.Transform.table=corebio.transform.Transform-class.html#table"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-190', 'table', 'link-190');">table</a></tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-string">'#'</tt><tt class="py-op">,</tt><tt class="py-string">'taxon'</tt><tt class="py-op">,</tt><tt class="py-string">'prev'</tt><tt class="py-op">,</tt><tt class="py-string">'succ'</tt><tt class="py-op">,</tt><tt class="py-string">'brlen'</tt><tt class="py-op">,</tt><tt class="py-string">'blen (sum)'</tt><tt class="py-op">,</tt><tt class="py-string">'support'</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">i</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-191" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.all_ids" class="py-name" href="#" onclick="return doclink('link-191', 'all_ids', 'link-69');">all_ids</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">            <tt class="py-name">n</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-192" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-192', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">i</tt><tt class="py-op">)</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">n</tt><tt class="py-op">.</tt><tt id="link-193" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-193', 'data', 'link-9');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">                <tt id="link-194" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-194', 'table', 'link-190');">table</a></tt><tt class="py-op">.</tt><tt id="link-195" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-195', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">i</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">.</tt><tt class="py-name">prev</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt><tt class="py-string">'-'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">                <tt class="py-name">tx</tt><tt class="py-op">=</tt><tt class="py-name">n</tt><tt class="py-op">.</tt><tt id="link-196" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-196', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">tx</tt><tt class="py-op">:</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">                    <tt class="py-name">tx</tt><tt class="py-op">=</tt><tt class="py-string">'-'</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">                <tt class="py-name">blength</tt><tt class="py-op">=</tt><tt class="py-name">n</tt><tt class="py-op">.</tt><tt id="link-197" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-197', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">blength</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">                    <tt class="py-name">blength</tt><tt class="py-op">=</tt><tt class="py-string">'-'</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">                    <tt class="py-name">sum_blength</tt><tt class="py-op">=</tt><tt class="py-string">'-'</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">                    <tt class="py-name">sum_blength</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-198" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.sum_branchlength" class="py-name" href="#" onclick="return doclink('link-198', 'sum_branchlength', 'link-133');">sum_branchlength</a></tt><tt class="py-op">(</tt><tt id="link-199" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-199', 'node', 'link-27');">node</a></tt><tt class="py-op">=</tt><tt class="py-name">i</tt><tt class="py-op">)</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">                <tt class="py-name">support</tt><tt class="py-op">=</tt><tt class="py-name">n</tt><tt class="py-op">.</tt><tt id="link-200" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-200', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">support</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">                    <tt class="py-name">support</tt><tt class="py-op">=</tt><tt class="py-string">'-'</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">                <tt id="link-201" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-201', 'table', 'link-190');">table</a></tt><tt class="py-op">.</tt><tt id="link-202" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-202', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">i</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-name">tx</tt><tt class="py-op">,</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">.</tt><tt class="py-name">prev</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-name">blength</tt><tt class="py-op">,</tt><tt class="py-name">sum_blength</tt><tt class="py-op">,</tt><tt class="py-name">support</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt> <tt class="py-string">'\n'</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-203', 'join', 'link-100');">join</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">'%3s %32s %15s %15s %8s %10s %8s'</tt> <tt class="py-op">%</tt> <tt class="py-name">l</tt> <tt class="py-keyword">for</tt> <tt class="py-name">l</tt> <tt class="py-keyword">in</tt> <tt id="link-204" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-204', 'table', 'link-190');">table</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt> <tt class="py-string">'\nRoot: '</tt><tt class="py-op">,</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> </tt>
</div><a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line"> </tt>
<a name="Tree.to_string"></a><div id="Tree.to_string-def"><a name="L449"></a><tt class="py-lineno">449</tt> <a class="py-toggle" href="#" id="Tree.to_string-toggle" onclick="return toggle('Tree.to_string');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#to_string">to_string</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">support_as_branchlengths</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt><tt class="py-param">branchlengths_only</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt><tt class="py-param">plain</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">,</tt><tt class="py-param">plain_newick</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.to_string-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.to_string-expanded"><a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a paup compatible tree line.</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line"><tt class="py-docstring">       </tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line"><tt class="py-docstring">        to_string(self,support_as_branchlengths=False,branchlengths_only=False,plain=True)</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">        <tt class="py-comment"># if there's a conflict in the arguments, we override plain=True</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">support_as_branchlengths</tt> <tt class="py-keyword">or</tt> <tt class="py-name">branchlengths_only</tt><tt class="py-op">:</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">            <tt class="py-name">plain</tt><tt class="py-op">=</tt><tt class="py-name">False</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">support_as_branchlengths</tt><tt class="py-op">=</tt><tt class="py-name">support_as_branchlengths</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">branchlengths_only</tt><tt class="py-op">=</tt><tt class="py-name">branchlengths_only</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">plain</tt><tt class="py-op">=</tt><tt class="py-name">plain</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line"> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">make_info_string</tt><tt class="py-op">(</tt><tt class="py-param">data</tt><tt class="py-op">,</tt><tt class="py-param">terminal</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">            <tt class="py-docstring">"""Creates nicely formatted support/branchlengths."""</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">            <tt class="py-comment"># CHECK FORMATTING</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">plain</tt><tt class="py-op">:</tt> <tt class="py-comment"># plain tree only. That's easy.</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-string">''</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">support_as_branchlengths</tt><tt class="py-op">:</tt> <tt class="py-comment"># support as branchlengths (eg. PAUP), ignore actual branchlengths</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">terminal</tt><tt class="py-op">:</tt>    <tt class="py-comment"># terminal branches have 100% support</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt class="py-string">':%1.2f'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">max_support</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt class="py-string">':%1.2f'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-205" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-205', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">)</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">branchlengths_only</tt><tt class="py-op">:</tt> <tt class="py-comment"># write only branchlengths, ignore support</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-string">':%1.5f'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-206" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-206', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">)</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt>   <tt class="py-comment"># write suport and branchlengths (e.g. .con tree of mrbayes)</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">terminal</tt><tt class="py-op">:</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt class="py-string">':%1.5f'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-207" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-207', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">)</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-208" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-208', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt id="link-209" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-209', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt>  <tt class="py-comment"># we have blen and suppport</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">                        <tt class="py-keyword">return</tt> <tt class="py-string">'%1.2f:%1.5f'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-210" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-210', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">,</tt><tt id="link-211" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-211', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">)</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt id="link-212" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-212', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt>                             <tt class="py-comment"># we have only blen</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">                        <tt class="py-keyword">return</tt> <tt class="py-string">'0.00000:%1.5f'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-213" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-213', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">)</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt id="link-214" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-214', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt>                                  <tt class="py-comment"># we have only support</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">                        <tt class="py-keyword">return</tt> <tt class="py-string">'%1.2f:0.00000'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-215" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-215', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">)</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">                        <tt class="py-keyword">return</tt> <tt class="py-string">'0.00:0.00000'</tt> </tt>
</div><a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line"> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">newickize</tt><tt class="py-op">(</tt><tt class="py-param">node</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">            <tt class="py-docstring">"""Convert a node tree to a newick tree recursively."""</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line"> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-216" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-216', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-217" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-217', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">:</tt>    <tt class="py-comment">#terminal</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-218', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-219" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-219', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-220', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">+</tt><tt class="py-name">make_info_string</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-221', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-222" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-222', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-223" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-223', 'data', 'link-9');">data</a></tt><tt class="py-op">,</tt><tt class="py-name">terminal</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-string">'(%s)%s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-224', 'join', 'link-100');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">map</tt><tt class="py-op">(</tt><tt class="py-name">newickize</tt><tt class="py-op">,</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-225', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-226" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-226', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-name">make_info_string</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-227" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-227', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-228" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-228', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-229" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-229', 'data', 'link-9');">data</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">subtree</tt> </tt>
</div><a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line">                     </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line">        <tt class="py-name">treeline</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-string">'tree'</tt><tt class="py-op">]</tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-230" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-230', 'name', 'link-10');">name</a></tt><tt class="py-op">:</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">            <tt class="py-name">treeline</tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-231', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-232', 'name', 'link-10');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">            <tt class="py-name">treeline</tt><tt class="py-op">.</tt><tt id="link-233" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-233', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'a_tree'</tt><tt class="py-op">)</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">        <tt class="py-name">treeline</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-234', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'='</tt><tt class="py-op">)</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">weight</tt><tt class="py-op">!=</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">            <tt class="py-name">treeline</tt><tt class="py-op">.</tt><tt id="link-235" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-235', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'[&amp;W%s]'</tt> <tt class="py-op">%</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">round</tt><tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">weight</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-number">3</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">rooted</tt><tt class="py-op">:</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">            <tt class="py-name">treeline</tt><tt class="py-op">.</tt><tt id="link-236" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-236', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'[&amp;R]'</tt><tt class="py-op">)</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">        <tt class="py-name">treeline</tt><tt class="py-op">.</tt><tt id="link-237" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-237', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'(%s)'</tt> <tt class="py-op">%</tt> <tt class="py-string">','</tt><tt class="py-op">.</tt><tt id="link-238" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-238', 'join', 'link-100');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">map</tt><tt class="py-op">(</tt><tt class="py-name">newickize</tt><tt class="py-op">,</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-239', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">        <tt class="py-name">treeline</tt><tt class="py-op">.</tt><tt id="link-240" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-240', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">';'</tt><tt class="py-op">)</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">plain_newick</tt><tt class="py-op">:</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">treeline</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">' '</tt><tt class="py-op">.</tt><tt id="link-241" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-241', 'join', 'link-100');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">treeline</tt><tt class="py-op">)</tt> </tt>
</div><a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">         </tt>
<a name="Tree.__str__"></a><div id="Tree.__str__-def"><a name="L512"></a><tt class="py-lineno">512</tt> <a class="py-toggle" href="#" id="Tree.__str__-toggle" onclick="return toggle('Tree.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.__str__-expanded"><a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">        <tt class="py-docstring">"""Short version of to_string(), gives plain tree"""</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.to_string()=corebio.seq_io._nexus.Trees.Tree-class.html#to_string"><a title="corebio.seq_io._nexus.Trees.Tree.to_string" class="py-name" href="#" onclick="return doclink('link-242', 'to_string', 'link-242');">to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">plain</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
</div><a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">         </tt>
<a name="Tree.unroot"></a><div id="Tree.unroot-def"><a name="L516"></a><tt class="py-lineno">516</tt> <a class="py-toggle" href="#" id="Tree.unroot-toggle" onclick="return toggle('Tree.unroot');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#unroot">unroot</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.unroot-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.unroot-expanded"><a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">        <tt class="py-docstring">"""Defines a unrooted Tree structure, using data of a rooted Tree."""</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line"> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">        <tt class="py-comment"># travel down the rooted tree structure and save all branches and the nodes they connect</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line"> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">_get_branches</tt><tt class="py-op">(</tt><tt class="py-param">node</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">            <tt class="py-name">branches</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">b</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-243" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-243', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-244" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-244', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">:</tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line">                <tt class="py-name">branches</tt><tt class="py-op">.</tt><tt id="link-245" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-245', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-246" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-246', 'node', 'link-27');">node</a></tt><tt class="py-op">,</tt><tt class="py-name">b</tt><tt class="py-op">,</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-247', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">b</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-248" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-248', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">,</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-249', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">b</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-250" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-250', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">                <tt class="py-name">branches</tt><tt class="py-op">.</tt><tt id="link-251" class="py-name"><a title="corebio.db.astral.RafSeqMap.extend" class="py-name" href="#" onclick="return doclink('link-251', 'extend', 'link-67');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">_get_branches</tt><tt class="py-op">(</tt><tt class="py-name">b</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">branches</tt> </tt>
</div><a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">     </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">=</tt><tt class="py-name">_get_branches</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">)</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">        <tt class="py-comment"># if root is bifurcating, then it is eliminated</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-252" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-252', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">            <tt class="py-comment"># find the two branches that connect to root</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">            <tt class="py-name">rootbranches</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-name">b</tt> <tt class="py-keyword">for</tt> <tt class="py-name">b</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt> <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> <tt class="py-keyword">in</tt> <tt class="py-name">b</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">            <tt class="py-name">b1</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">.</tt><tt class="py-name">pop</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">.</tt><tt id="link-253" class="py-name" targets="Method corebio.db.astral.RafSeqMap.index()=corebio.db.astral.RafSeqMap-class.html#index,Method corebio.matrix.AlphabeticArray.index()=corebio.matrix.AlphabeticArray-class.html#index,Function corebio.seq_io.fasta_io.index()=corebio.seq_io.fasta_io-module.html#index,Method corebio.utils.Reiterate.index()=corebio.utils.Reiterate-class.html#index"><a title="corebio.db.astral.RafSeqMap.index
corebio.matrix.AlphabeticArray.index
corebio.seq_io.fasta_io.index
corebio.utils.Reiterate.index" class="py-name" href="#" onclick="return doclink('link-253', 'index', 'link-253');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">rootbranches</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line">            <tt class="py-name">b2</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">.</tt><tt class="py-name">pop</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">.</tt><tt id="link-254" class="py-name"><a title="corebio.db.astral.RafSeqMap.index
corebio.matrix.AlphabeticArray.index
corebio.seq_io.fasta_io.index
corebio.utils.Reiterate.index" class="py-name" href="#" onclick="return doclink('link-254', 'index', 'link-253');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">rootbranches</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">            <tt class="py-comment"># Connect them two each other. If both have support, it should be identical (or one set to None?).</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">            <tt class="py-comment"># If both have branchlengths, they will be added</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">            <tt class="py-name">newbranch</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-name">b1</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt><tt class="py-name">b2</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt><tt class="py-name">b1</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">+</tt><tt class="py-name">b2</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">b1</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line">                <tt class="py-name">newbranch</tt><tt class="py-op">.</tt><tt id="link-255" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-255', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">b2</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-comment"># either None (both rootbranches are unsupported) or some support</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">b2</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">                <tt class="py-name">newbranch</tt><tt class="py-op">.</tt><tt id="link-256" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-256', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">b1</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-comment"># dito</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">b1</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">==</tt><tt class="py-name">b2</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">:</tt>           </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">                <tt class="py-name">newbranch</tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-257', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">b1</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-comment"># identical support</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">b1</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">==</tt><tt class="py-number">0</tt> <tt class="py-keyword">or</tt> <tt class="py-name">b2</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">==</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">                <tt class="py-name">newbranch</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-258', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">b1</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">+</tt><tt class="py-name">b2</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-comment"># one is 0, take the other</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-259" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-259', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">,</tt> <tt class="py-string">'Support mismatch in bifurcating root: %f, %f'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">b1</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">b2</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-260', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">newbranch</tt><tt class="py-op">)</tt> </tt>
</div><a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line"> </tt>
<a name="Tree.root_with_outgroup"></a><div id="Tree.root_with_outgroup-def"><a name="L550"></a><tt class="py-lineno">550</tt> <a class="py-toggle" href="#" id="Tree.root_with_outgroup-toggle" onclick="return toggle('Tree.root_with_outgroup');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees.Tree-class.html#root_with_outgroup">root_with_outgroup</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt><tt class="py-param">outgroup</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.root_with_outgroup-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Tree.root_with_outgroup-expanded"><a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line">         </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">_connect_subtree</tt><tt class="py-op">(</tt><tt class="py-param">parent</tt><tt class="py-op">,</tt><tt class="py-param">child</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line">            <tt class="py-docstring">"""Hook subtree starting with node child to parent."""</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">i</tt><tt class="py-op">,</tt><tt class="py-name">branch</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">parent</tt> <tt class="py-keyword">in</tt> <tt class="py-name">branch</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-keyword">and</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt class="py-name">branch</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">                    <tt class="py-name">branch</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">.</tt><tt class="py-name">pop</tt><tt class="py-op">(</tt><tt class="py-name">i</tt><tt class="py-op">)</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt>  </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-261" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-261', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">,</tt> <tt class="py-string">'Unable to connect nodes for rooting: nodes %d and %d are not connected'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">,</tt><tt class="py-name">child</tt><tt class="py-op">)</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-262" class="py-name" targets="Method corebio.seq_io._nexus.Nodes.Chain.link()=corebio.seq_io._nexus.Nodes.Chain-class.html#link"><a title="corebio.seq_io._nexus.Nodes.Chain.link" class="py-name" href="#" onclick="return doclink('link-262', 'link', 'link-262');">link</a></tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">,</tt><tt class="py-name">child</tt><tt class="py-op">)</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-263" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-263', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-264" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-264', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">branchlength</tt><tt class="py-op">=</tt><tt class="py-name">branch</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-265" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-265', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-266', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">=</tt><tt class="py-name">branch</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">            <tt class="py-comment">#now check if there are more branches connected to the child, and if so, connect them</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">            <tt class="py-name">child_branches</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-name">b</tt> <tt class="py-keyword">for</tt> <tt class="py-name">b</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt> <tt class="py-keyword">if</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt class="py-name">b</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">b</tt> <tt class="py-keyword">in</tt> <tt class="py-name">child_branches</tt><tt class="py-op">:</tt> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">child</tt><tt class="py-op">==</tt><tt class="py-name">b</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">                    <tt class="py-name">succ</tt><tt class="py-op">=</tt><tt class="py-name">b</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">                    <tt class="py-name">succ</tt><tt class="py-op">=</tt><tt class="py-name">b</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">                <tt class="py-name">_connect_subtree</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">,</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt>  </tt>
</div><a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">             </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">        <tt class="py-comment"># check the outgroup we're supposed to root with</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">outgroup</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">        <tt class="py-name">outgroup_node</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.is_monophyletic()=corebio.seq_io._nexus.Trees.Tree-class.html#is_monophyletic"><a title="corebio.seq_io._nexus.Trees.Tree.is_monophyletic" class="py-name" href="#" onclick="return doclink('link-267', 'is_monophyletic', 'link-267');">is_monophyletic</a></tt><tt class="py-op">(</tt><tt class="py-name">outgroup</tt><tt class="py-op">)</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">outgroup_node</tt><tt class="py-op">==</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">        <tt class="py-comment"># if tree is already rooted with outgroup on a bifurcating root,</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line">        <tt class="py-comment"># or the outgroup includes all taxa on the tree, then we're fine</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-268', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-number">2</tt> <tt class="py-keyword">and</tt> <tt class="py-name">outgroup_node</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-269" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-269', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">outgroup_node</tt><tt class="py-op">==</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">:</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">         </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-270" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.unroot()=corebio.seq_io._nexus.Trees.Tree-class.html#unroot"><a title="corebio.seq_io._nexus.Trees.Tree.unroot" class="py-name" href="#" onclick="return doclink('link-270', 'unroot', 'link-270');">unroot</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">        <tt class="py-comment"># now we find the branch that connects outgroup and ingroup</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">        <tt class="py-comment">#print self.node(outgroup_node).prev</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">i</tt><tt class="py-op">,</tt><tt class="py-name">b</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">outgroup_node</tt> <tt class="py-keyword">in</tt> <tt class="py-name">b</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-keyword">and</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-271" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-271', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">outgroup_node</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">prev</tt> <tt class="py-keyword">in</tt> <tt class="py-name">b</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">                <tt class="py-name">root_branch</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">.</tt><tt class="py-name">pop</tt><tt class="py-op">(</tt><tt class="py-name">i</tt><tt class="py-op">)</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-272" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-272', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">,</tt> <tt class="py-string">'Unrooted and rooted Tree do not match'</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">outgroup_node</tt><tt class="py-op">==</tt><tt class="py-name">root_branch</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">            <tt class="py-name">ingroup_node</tt><tt class="py-op">=</tt><tt class="py-name">root_branch</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line">            <tt class="py-name">ingroup_node</tt><tt class="py-op">=</tt><tt class="py-name">root_branch</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line">        <tt class="py-comment"># now we destroy the old tree structure, but keep node data. Nodes will be reconnected according to new outgroup</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-273" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.all_ids" class="py-name" href="#" onclick="return doclink('link-273', 'all_ids', 'link-69');">all_ids</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-274" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-274', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">prev</tt><tt class="py-op">=</tt><tt class="py-name">None</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-275" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-275', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">succ</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line">        <tt class="py-comment"># now we just add both subtrees (outgroup and ingroup) branch for branch</tt> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">        <tt class="py-name">root</tt><tt class="py-op">=</tt><tt id="link-276" class="py-name"><a title="corebio.seq_io._nexus.Nodes" class="py-name" href="#" onclick="return doclink('link-276', 'Nodes', 'link-0');">Nodes</a></tt><tt class="py-op">.</tt><tt id="link-277" class="py-name"><a title="corebio.db.scop.Node
corebio.seq_io._nexus.Nodes.Node" class="py-name" href="#" onclick="return doclink('link-277', 'Node', 'link-13');">Node</a></tt><tt class="py-op">(</tt><tt id="link-278" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-278', 'data', 'link-9');">data</a></tt><tt class="py-op">=</tt><tt id="link-279" class="py-name"><a title="corebio.seq_io._nexus.Trees.NodeData" class="py-name" href="#" onclick="return doclink('link-279', 'NodeData', 'link-5');">NodeData</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>            <tt class="py-comment"># new root    </tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.add
corebio.seq_io._nexus.StepMatrix.add" class="py-name" href="#" onclick="return doclink('link-280', 'add', 'link-15');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">root</tt><tt class="py-op">)</tt>                              <tt class="py-comment"># add to tree description</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">=</tt><tt class="py-name">root</tt><tt class="py-op">.</tt><tt class="py-name">id</tt>                           <tt class="py-comment"># set as root</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">.</tt><tt id="link-281" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-281', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">root</tt><tt class="py-op">.</tt><tt class="py-name">id</tt><tt class="py-op">,</tt><tt class="py-name">ingroup_node</tt><tt class="py-op">,</tt><tt class="py-name">root_branch</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt><tt class="py-name">root_branch</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>  <tt class="py-comment"># add branch to ingroup to unrooted tree</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unrooted</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-282', 'append', 'link-17');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">root</tt><tt class="py-op">.</tt><tt class="py-name">id</tt><tt class="py-op">,</tt><tt class="py-name">outgroup_node</tt><tt class="py-op">,</tt><tt class="py-number">0.0</tt><tt class="py-op">,</tt><tt class="py-number">0.0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>   <tt class="py-comment"># add branch to outgroup to unrooted tree</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">        <tt class="py-name">_connect_subtree</tt><tt class="py-op">(</tt><tt class="py-name">root</tt><tt class="py-op">.</tt><tt class="py-name">id</tt><tt class="py-op">,</tt><tt class="py-name">ingroup_node</tt><tt class="py-op">)</tt>      <tt class="py-comment"># add ingroup</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line">        <tt class="py-name">_connect_subtree</tt><tt class="py-op">(</tt><tt class="py-name">root</tt><tt class="py-op">.</tt><tt class="py-name">id</tt><tt class="py-op">,</tt><tt class="py-name">outgroup_node</tt><tt class="py-op">)</tt>     <tt class="py-comment"># add outgroup</tt> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line">        <tt class="py-comment"># if theres still a lonely node in self.chain, then it's the old root, and we delete it</tt> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">        <tt class="py-name">oldroot</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-name">i</tt> <tt class="py-keyword">for</tt> <tt class="py-name">i</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.all_ids" class="py-name" href="#" onclick="return doclink('link-283', 'all_ids', 'link-69');">all_ids</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-284', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">i</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">prev</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">i</tt><tt class="py-op">!=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">]</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">oldroot</tt><tt class="py-op">)</tt><tt class="py-op">&gt;</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-285" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-285', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">,</tt> <tt class="py-string">'Isolated nodes in tree description: %s'</tt> <tt class="py-op">%</tt> <tt class="py-string">','</tt><tt class="py-op">.</tt><tt id="link-286" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-286', 'join', 'link-100');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">oldroot</tt><tt class="py-op">)</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">oldroot</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-287" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.kill" class="py-name" href="#" onclick="return doclink('link-287', 'kill', 'link-54');">kill</a></tt><tt class="py-op">(</tt><tt class="py-name">oldroot</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">root</tt> </tt>
</div></div><a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">         </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line">          </tt>
<a name="consensus"></a><div id="consensus-def"><a name="L617"></a><tt class="py-lineno">617</tt> <a class="py-toggle" href="#" id="consensus-toggle" onclick="return toggle('consensus');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io._nexus.Trees-module.html#consensus">consensus</a><tt class="py-op">(</tt><tt class="py-param">trees</tt><tt class="py-op">,</tt> <tt class="py-param">threshold</tt><tt class="py-op">=</tt><tt class="py-number">0.5</tt><tt class="py-op">,</tt><tt class="py-param">outgroup</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="consensus-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="consensus-expanded"><a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line">    <tt class="py-docstring">"""Compute a majority rule consensus tree of all clades with relative frequency&gt;=threshold from a list of trees."""</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line">     </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">    <tt class="py-name">total</tt><tt class="py-op">=</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">trees</tt><tt class="py-op">)</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">total</tt><tt class="py-op">==</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line">    <tt class="py-comment"># shouldn't we make sure that it's NodeData or subclass??</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line">    <tt class="py-name">dataclass</tt><tt class="py-op">=</tt><tt class="py-name">trees</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">dataclass</tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">    <tt class="py-name">max_support</tt><tt class="py-op">=</tt><tt class="py-name">trees</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">max_support</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">    <tt class="py-name">clades</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line">    <tt class="py-comment">#countclades={}</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">    <tt class="py-name">alltaxa</tt><tt class="py-op">=</tt><tt id="link-288" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-288', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">trees</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-289" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-289', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line">    <tt class="py-comment"># calculate calde frequencies</tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">    <tt class="py-name">c</tt><tt class="py-op">=</tt><tt class="py-number">0</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">t</tt> <tt class="py-keyword">in</tt> <tt class="py-name">trees</tt><tt class="py-op">:</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">        <tt class="py-name">c</tt><tt class="py-op">+=</tt><tt class="py-number">1</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">        <tt class="py-comment">#if c%50==0:</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">        <tt class="py-comment">#    print c</tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">alltaxa</tt><tt class="py-op">!=</tt><tt id="link-290" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-290', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">t</tt><tt class="py-op">.</tt><tt id="link-291" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-291', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-292" class="py-name"><a title="corebio.seq_io._nexus.Trees.TreeError" class="py-name" href="#" onclick="return doclink('link-292', 'TreeError', 'link-16');">TreeError</a></tt><tt class="py-op">,</tt> <tt class="py-string">'Trees for consensus must contain the same taxa'</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">        <tt class="py-name">t</tt><tt class="py-op">.</tt><tt id="link-293" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.root_with_outgroup()=corebio.seq_io._nexus.Trees.Tree-class.html#root_with_outgroup"><a title="corebio.seq_io._nexus.Trees.Tree.root_with_outgroup" class="py-name" href="#" onclick="return doclink('link-293', 'root_with_outgroup', 'link-293');">root_with_outgroup</a></tt><tt class="py-op">(</tt><tt class="py-name">outgroup</tt><tt class="py-op">=</tt><tt class="py-name">outgroup</tt><tt class="py-op">)</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">st_node</tt> <tt class="py-keyword">in</tt> <tt class="py-name">t</tt><tt class="py-op">.</tt><tt class="py-name">_walk</tt><tt class="py-op">(</tt><tt class="py-name">t</tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">            <tt class="py-name">subclade_taxa</tt><tt class="py-op">=</tt><tt class="py-name">t</tt><tt class="py-op">.</tt><tt id="link-294" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.get_taxa" class="py-name" href="#" onclick="return doclink('link-294', 'get_taxa', 'link-68');">get_taxa</a></tt><tt class="py-op">(</tt><tt class="py-name">st_node</tt><tt class="py-op">)</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line">            <tt class="py-name">subclade_taxa</tt><tt class="py-op">.</tt><tt class="py-name">sort</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line">            <tt class="py-name">subclade_taxa</tt><tt class="py-op">=</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">subclade_taxa</tt><tt class="py-op">)</tt> <tt class="py-comment"># lists are not hashable</tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">subclade_taxa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">clades</tt><tt class="py-op">:</tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line">                <tt class="py-name">clades</tt><tt class="py-op">[</tt><tt class="py-name">subclade_taxa</tt><tt class="py-op">]</tt><tt class="py-op">+=</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">t</tt><tt class="py-op">.</tt><tt class="py-name">weight</tt><tt class="py-op">)</tt><tt class="py-op">/</tt><tt class="py-name">total</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">                <tt class="py-name">clades</tt><tt class="py-op">[</tt><tt class="py-name">subclade_taxa</tt><tt class="py-op">]</tt><tt class="py-op">=</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">t</tt><tt class="py-op">.</tt><tt class="py-name">weight</tt><tt class="py-op">)</tt><tt class="py-op">/</tt><tt class="py-name">total</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line">            <tt class="py-comment">#if subclade_taxa in countclades:</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">            <tt class="py-comment">#    countclades[subclade_taxa]+=t.weight</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">            <tt class="py-comment">#else:</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line">            <tt class="py-comment">#    countclades[subclade_taxa]=t.weight</tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">    <tt class="py-comment"># weed out clades below threshold</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">,</tt><tt class="py-name">p</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">clades</tt><tt class="py-op">.</tt><tt class="py-name">items</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">p</tt><tt class="py-op">&lt;</tt><tt class="py-name">threshold</tt><tt class="py-op">:</tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">clades</tt><tt class="py-op">[</tt><tt class="py-name">c</tt><tt class="py-op">]</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">    <tt class="py-comment"># create a tree with a root node</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">    <tt id="link-295" class="py-name" targets="Function corebio.seq_io._nexus.Trees.consensus()=corebio.seq_io._nexus.Trees-module.html#consensus"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-295', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">=</tt><tt id="link-296" class="py-name" targets="Class corebio.seq_io._nexus.Trees.Tree=corebio.seq_io._nexus.Trees.Tree-class.html"><a title="corebio.seq_io._nexus.Trees.Tree" class="py-name" href="#" onclick="return doclink('link-296', 'Tree', 'link-296');">Tree</a></tt><tt class="py-op">(</tt><tt id="link-297" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-297', 'name', 'link-10');">name</a></tt><tt class="py-op">=</tt><tt class="py-string">'consensus_%2.1f'</tt> <tt class="py-op">%</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">threshold</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt id="link-298" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-298', 'data', 'link-9');">data</a></tt><tt class="py-op">=</tt><tt class="py-name">dataclass</tt><tt class="py-op">)</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line">    <tt class="py-comment"># each clade needs a node in the new tree, add them as isolated nodes</tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">,</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">clades</tt><tt class="py-op">.</tt><tt class="py-name">items</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">        <tt id="link-299" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-299', 'node', 'link-27');">node</a></tt><tt class="py-op">=</tt><tt id="link-300" class="py-name"><a title="corebio.seq_io._nexus.Nodes" class="py-name" href="#" onclick="return doclink('link-300', 'Nodes', 'link-0');">Nodes</a></tt><tt class="py-op">.</tt><tt id="link-301" class="py-name"><a title="corebio.db.scop.Node
corebio.seq_io._nexus.Nodes.Node" class="py-name" href="#" onclick="return doclink('link-301', 'Node', 'link-13');">Node</a></tt><tt class="py-op">(</tt><tt id="link-302" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-302', 'data', 'link-9');">data</a></tt><tt class="py-op">=</tt><tt class="py-name">dataclass</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">        <tt id="link-303" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-303', 'node', 'link-27');">node</a></tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-304', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">=</tt><tt class="py-name">s</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">        <tt id="link-305" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-305', 'node', 'link-27');">node</a></tt><tt class="py-op">.</tt><tt id="link-306" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-306', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">=</tt><tt id="link-307" class="py-name"><a title="corebio.seq_io._nexus.StepMatrix.set" class="py-name" href="#" onclick="return doclink('link-307', 'set', 'link-1');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">eval</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line">        <tt id="link-308" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-308', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-309" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.add
corebio.seq_io._nexus.StepMatrix.add" class="py-name" href="#" onclick="return doclink('link-309', 'add', 'link-15');">add</a></tt><tt class="py-op">(</tt><tt id="link-310" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-310', 'node', 'link-27');">node</a></tt><tt class="py-op">)</tt> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line">    <tt class="py-comment"># set root node data</tt> </tt>
<a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">    <tt id="link-311" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-311', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-312" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-312', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-313" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-313', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-314" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-314', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">support</tt><tt class="py-op">=</tt><tt class="py-name">None</tt> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">    <tt id="link-315" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-315', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-316" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-316', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt id="link-317" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-317', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt class="py-name">root</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-318" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-318', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">=</tt><tt class="py-name">alltaxa</tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">    <tt class="py-comment"># we sort the nodes by no. of taxa in the clade, so root will be the last</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line">    <tt class="py-name">consensus_ids</tt><tt class="py-op">=</tt><tt id="link-319" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-319', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-320" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.all_ids" class="py-name" href="#" onclick="return doclink('link-320', 'all_ids', 'link-69');">all_ids</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">    <tt class="py-name">consensus_ids</tt><tt class="py-op">.</tt><tt class="py-name">sort</tt><tt class="py-op">(</tt><tt class="py-keyword">lambda</tt> <tt class="py-name">x</tt><tt class="py-op">,</tt><tt class="py-name">y</tt><tt class="py-op">:</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-321" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-321', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-322" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-322', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">x</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-323" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-323', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">)</tt><tt class="py-op">-</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-324" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-324', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-325" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-325', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">y</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-326', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line">    <tt class="py-comment"># now we just have to hook each node to the next smallest node that includes all taxa of the current </tt> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">i</tt><tt class="py-op">,</tt><tt class="py-name">current</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">consensus_ids</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-comment"># skip the last one which is the root</tt> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">        <tt class="py-comment">#print '----'</tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line">        <tt class="py-comment">#print 'current: ',consensus.node(current).data.taxon</tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">        <tt class="py-comment"># search remaining nodes</tt> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">parent</tt> <tt class="py-keyword">in</tt> <tt class="py-name">consensus_ids</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">+</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line">            <tt class="py-comment">#print 'parent: ',consensus.node(parent).data.taxon</tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-327" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-327', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-328" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-328', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-329" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-329', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">.</tt><tt class="py-name">issuperset</tt><tt class="py-op">(</tt><tt id="link-330" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-330', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-331" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-331', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">current</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-332" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-332', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">            <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">exit</tt><tt class="py-op">(</tt><tt class="py-string">'corrupt tree structure?'</tt><tt class="py-op">)</tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">        <tt class="py-comment"># internal nodes don't have taxa</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-333" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-333', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-334" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-334', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">current</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-335" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-335', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">            <tt id="link-336" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-336', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-337" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-337', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">current</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-338" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-338', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">=</tt><tt id="link-339" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-339', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-340" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-340', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">current</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-341', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">.</tt><tt class="py-name">pop</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">            <tt class="py-comment"># reset the support for terminal nodes to maximum</tt> </tt>
<a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line">            <tt class="py-comment">#consensus.node(current).data.support=max_support</tt> </tt>
<a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">            <tt id="link-342" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-342', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-343" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-343', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">current</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-344" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-344', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">=</tt><tt class="py-name">None</tt> </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line">        <tt id="link-345" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-345', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-346" class="py-name"><a title="corebio.seq_io._nexus.Nodes.Chain.link" class="py-name" href="#" onclick="return doclink('link-346', 'link', 'link-262');">link</a></tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">,</tt><tt class="py-name">current</tt><tt class="py-op">)</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">    <tt class="py-comment"># eliminate root taxon name</tt> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line">    <tt id="link-347" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-347', 'consensus', 'link-295');">consensus</a></tt><tt class="py-op">.</tt><tt id="link-348" class="py-name"><a title="corebio.seq_io._nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-348', 'node', 'link-27');">node</a></tt><tt class="py-op">(</tt><tt class="py-name">consensus_ids</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-349" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-349', 'data', 'link-9');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">taxon</tt><tt class="py-op">=</tt><tt class="py-name">None</tt>  </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-350" class="py-name"><a title="corebio.seq_io._nexus.Trees.consensus" class="py-name" href="#" onclick="return doclink('link-350', 'consensus', 'link-295');">consensus</a></tt> </tt>
</div><a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Mar 12 21:34:14 2014
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
