<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>corebio.seq_io</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="corebio-module.html">Package&nbsp;corebio</a> ::
        Package&nbsp;seq_io
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="corebio.seq_io-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== PACKAGE DESCRIPTION ==================== -->
<h1 class="epydoc">Package seq_io</h1><p class="nomargin-top"><span class="codelink"><a href="corebio.seq_io-pysrc.html">source&nbsp;code</a></span></p>
<pre class="literalblock">
Sequence file reading and writing.

Biological sequence data is stored and transmitted using a wide variety of
different file formats. This package provides convenient methods to read and
write several of these file fomats.

CoreBio is often capable of guessing the correct file type, either from the
file extension or the structure of the file:
&gt;&gt;&gt; import corebio.seq_io
&gt;&gt;&gt; afile = open(&quot;test_corebio/data/cap.fa&quot;)
&gt;&gt;&gt; seqs = corebio.seq_io.read(afile)

Alternatively, each sequence file type has a separate module named FILETYPE_io
(e.g. fasta_io, clustal_io).
&gt;&gt;&gt; import corebio.seq_io.fasta_io
&gt;&gt;&gt; afile = open(&quot;test_corebio/data/cap.fa&quot;)
&gt;&gt;&gt; seqs = corebio.seq_io.fasta_io.read( afile )

Sequence data can also be written back to files:
&gt;&gt;&gt; fout = open(&quot;out.fa&quot;, &quot;w&quot;)
&gt;&gt;&gt; corebio.seq_io.fasta_io.write( fout, seqs )


Supported File Formats
----------------------

Module              Name            Extension  read write features   
---------------------------------------------------------------------------
array_io            array, flatfile             yes  yes    none
clustal_io          clustalw        aln         yes  yes
fasta_io            fasta, Pearson  fa          yes  yes    none
genbank_io          genbank         gb          yes         
intelligenetics_io  intelligenetics ig          yes  yes
msf_io              msf             msf         yes
nbrf_io             nbrf, pir       pir         yes
nexus_io            nexus           nexus       yes
phylip_io           phylip          phy         yes
plain_io            plain, raw                  yes  yes    none
table_io            table           tbl         yes  yes    none

Each IO module defines one or more of the following functions and variables:

read(afile, alphabet=None) 
    Read a file of sequence data and return a SeqList, a collection
    of Seq's (Alphabetic strings) and features.

read_seq(afile, alphabet=None)
    Read a single sequence from a file.

iter_seq(afile, alphabet =None) 
    Iterate over the sequences in a file. 
    
index(afile, alphabet = None)
    Instead of loading all of the sequences into memory, scan the file and
    return an index map that will load sequences on demand. Typically not
    implemented for formats with interleaved sequences.

write(afile, seqlist)
    Write a collection of sequences to the specifed file.

write_seq(afile, seq)
    Write one sequence to the file. Only implemented for non-interleaved, 
    headerless formats, such as fasta and plain.

example
    A string containing a short example of the file format

names
    A list of synonyms for the file format. E.g. for fasta_io, ( 'fasta',    
    'pearson', 'fa'). The first entry is the preferred format name.

extensions
    A list of file name extensions used for this file format. e.g. 
    fasta_io.extensions is ('fa', 'fasta', 'fast', 'seq', 'fsa', 'fst', 'nt',
    'aa','fna','mpfa').  The preferred or standard extension is first in the 
    list.


Attributes :
- formats -- Available seq_io format parsers
- format_names -- A map between format names and format parsers.
- format_extensions -- A map between filename extensions and parsers.

</pre>

<!-- ==================== SUBMODULES ==================== -->
<a name="section-Submodules"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td align="left" colspan="2" class="table-header">
    <span class="table-header">Submodules</span></td>
</tr>
  <tr><td class="summary">
  <ul class="nomargin">
    <li class="private"> <strong class="uidlink"><a href="corebio.seq_io._nexus-module.html" onclick="show_private();">corebio.seq_io._nexus</a></strong>: <em class="summary">Parse the contents of a nexus file.</em>
    <ul class="private">
    <li> <strong class="uidlink"><a href="corebio.seq_io._nexus.Nodes-module.html">corebio.seq_io._nexus.Nodes</a></strong>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io._nexus.Trees-module.html">corebio.seq_io._nexus.Trees</a></strong>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.array_io-module.html">corebio.seq_io.array_io</a></strong>: <em class="summary">Read and write a rectangular array of sequence data.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.clustal_io-module.html">corebio.seq_io.clustal_io</a></strong>: <em class="summary">Read and write the CLUSTAL sequence file format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.fasta_io-module.html">corebio.seq_io.fasta_io</a></strong>: <em class="summary">Read and write sequence information in FASTA format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.genbank_io-module.html">corebio.seq_io.genbank_io</a></strong>: <em class="summary">Read GenBank flat files.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.intelligenetics_io-module.html">corebio.seq_io.intelligenetics_io</a></strong>: <em class="summary">Read and write sequence information in IntelliGenetics format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.msf_io-module.html">corebio.seq_io.msf_io</a></strong>: <em class="summary">Read sequence information in MSF format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.nbrf_io-module.html">corebio.seq_io.nbrf_io</a></strong>: <em class="summary">Sequence IO for NBRF/PIR format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.nexus_io-module.html">corebio.seq_io.nexus_io</a></strong>: <em class="summary">Read the sequence data from a nexus file.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.null_io-module.html">corebio.seq_io.null_io</a></strong>: <em class="summary">Null sequence IO.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.phylip_io-module.html">corebio.seq_io.phylip_io</a></strong>: <em class="summary">Reads Sequences in interleaved Phylip format (not sequential) and returns a
list of sequences.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.plain_io-module.html">corebio.seq_io.plain_io</a></strong>: <em class="summary">Read and write raw, unformatted sequence data.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.stockholm_io-module.html">corebio.seq_io.stockholm_io</a></strong>: <em class="summary">Read a multiple sequence alignment in STOCKHOLM format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.table_io-module.html">corebio.seq_io.table_io</a></strong>: <em class="summary">Read and write sequence information in tab-delimited format.</em>    </li>
  </ul></td></tr>
</table>

<br />
<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td align="left" colspan="2" class="table-header">
    <span class="table-header">Functions</span></td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="format_names"></a><span class="summary-sig-name">format_names</span>()</span><br />
      Return a map between format names and format modules</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="corebio.seq_io-pysrc.html#format_names">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="format_extensions"></a><span class="summary-sig-name">format_extensions</span>()</span><br />
      Return a map between filename extensions and sequence file types</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="corebio.seq_io-pysrc.html#format_extensions">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="corebio.seq_io-module.html#read" class="summary-sig-name">read</a>(<span class="summary-sig-arg">fin</span>,
        <span class="summary-sig-arg">alphabet</span>=<span class="summary-sig-default">None</span>)</span><br />
      Read a sequence file and attempt to guess its format.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="corebio.seq_io-pysrc.html#read">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td align="left" colspan="2" class="table-header">
    <span class="table-header">Variables</span></td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="corebio.seq_io-module.html#formats" class="summary-name">formats</a> = <code title="(&lt;module 'corebio.seq_io.clustal_io' from '/home/aleksand/Work/code/Lo\
gOddsLogo/corebio/seq_io/clustal_io.pyc'&gt;,
 &lt;module 'corebio.seq_io.fasta_io' from '/home/aleksand/Work/code/LogO\
ddsLogo/corebio/seq_io/fasta_io.pyc'&gt;,
 &lt;module 'corebio.seq_io.plain_io' from '/home/aleksand/Work/code/LogO\
ddsLogo/corebio/seq_io/plain_io.pyc'&gt;,
 &lt;module 'corebio.seq_io.msf_io' from '/home/aleksand/Work/code/LogOdd\
sLogo/corebio/seq_io/msf_io.pyc'&gt;,
..."><code class="variable-group">(</code>&lt;module 'corebio.seq_io.clustal_io' from '/home/ale<code class="variable-ellipsis">...</code></code><br />
      Available seq_io formats
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td align="left" colspan="2" class="table-header">
    <span class="table-header">Function Details</span></td>
</tr>
</table>
<a name="read"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">read</span>(<span class="sig-arg">fin</span>,
        <span class="sig-arg">alphabet</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="corebio.seq_io-pysrc.html#read">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <pre class="literalblock">
Read a sequence file and attempt to guess its format. 
First the filename extension (if available) is used to infer the format.
If that fails, then we attempt to parse the file using several common   
formats.

Note, fin cannot be unseekable stream such as sys.stdin

returns :
    SeqList
raises :
    ValueError - If the file cannot be parsed.
    ValueError - Sequence do not conform to the alphabet.

</pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== VARIABLES DETAILS ==================== -->
<a name="section-VariablesDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td align="left" colspan="2" class="table-header">
    <span class="table-header">Variables Details</span></td>
</tr>
</table>
<a name="formats"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">formats</h3>
  <p>Available seq_io formats</p>
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
<code class="variable-group">(</code>&lt;module 'corebio.seq_io.clustal_io' from '/home/aleksand/Work/code/Lo<span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
gOddsLogo/corebio/seq_io/clustal_io.pyc'&gt;<code class="variable-op">,</code>
 &lt;module 'corebio.seq_io.fasta_io' from '/home/aleksand/Work/code/LogO<span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
ddsLogo/corebio/seq_io/fasta_io.pyc'&gt;<code class="variable-op">,</code>
 &lt;module 'corebio.seq_io.plain_io' from '/home/aleksand/Work/code/LogO<span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
ddsLogo/corebio/seq_io/plain_io.pyc'&gt;<code class="variable-op">,</code>
 &lt;module 'corebio.seq_io.msf_io' from '/home/aleksand/Work/code/LogOdd<span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
sLogo/corebio/seq_io/msf_io.pyc'&gt;<code class="variable-op">,</code>
<code class="variable-ellipsis">...</code>
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Mar 12 21:34:13 2014
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
