<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>corebio.seq_io.clustal_io</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="corebio-module.html">Package&nbsp;corebio</a> ::
        <a href="corebio.seq_io-module.html">Package&nbsp;seq_io</a> ::
        Module&nbsp;clustal_io
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="corebio.seq_io.clustal_io-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="corebio.seq_io.clustal_io-module.html">Module corebio.seq_io.clustal_io</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line">  </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment">#  Copyright (c) 2005 Gavin E. Crooks &lt;gec@threeplusone.com&gt;</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment">#  This software is distributed under the MIT Open Source License.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment">#  &lt;http://www.opensource.org/licenses/mit-license.html&gt;</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment">#  Permission is hereby granted, free of charge, to any person obtaining a </tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment">#  copy of this software and associated documentation files (the "Software"),</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-comment">#  to deal in the Software without restriction, including without limitation</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-comment">#  the rights to use, copy, modify, merge, publish, distribute, sublicense,</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-comment">#  and/or sell copies of the Software, and to permit persons to whom the</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-comment">#  Software is furnished to do so, subject to the following conditions:</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-comment">#  The above copyright notice and this permission notice shall be included</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-comment">#  in all copies or substantial portions of the Software.</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-comment">#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR </tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-comment">#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, </tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-comment">#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-comment">#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER </tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-comment">#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-comment">#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN </tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-comment">#  THE SOFTWARE.</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring">""" Read and write the CLUSTAL sequence file format.</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring"> </tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">See :</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">- http://www.cmpharm.ucsf.edu/~goh/Treecorr/sampleAlignment.html</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">- http://www.bioperl.org/wiki/ClustalW_multiple_alignment_format</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring">  </tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">Ref :</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">-   Higgins D., Thompson J., Gibson T., Thompson J.D., Higgins D.G., Gibson </tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">    T.J. (1994). CLUSTAL W: improving the sensitivity of progressive multiple</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    sequence alignment through sequence weighting, position-specific gap</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">    penalties and weight matrix choice. Nucleic Acids Res. 22:4673-4680.</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-comment"># TODO: What happens if CLUSTAL is not the first line of the file?</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package corebio=corebio-module.html"><a title="corebio" class="py-name" href="#" onclick="return doclink('link-0', 'corebio', 'link-0');">corebio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package corebio.utils=corebio.utils-module.html"><a title="corebio.utils" class="py-name" href="#" onclick="return doclink('link-1', 'utils', 'link-1');">utils</a></tt> <tt class="py-keyword">import</tt> <tt class="py-op">*</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="corebio" class="py-name" href="#" onclick="return doclink('link-2', 'corebio', 'link-0');">corebio</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Module corebio.seq=corebio.seq-module.html"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-3', 'seq', 'link-3');">seq</a></tt> <tt class="py-keyword">import</tt> <tt class="py-op">*</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="corebio" class="py-name" href="#" onclick="return doclink('link-4', 'corebio', 'link-0');">corebio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Package corebio.seq_io=corebio.seq_io-module.html"><a title="corebio.seq_io" class="py-name" href="#" onclick="return doclink('link-5', 'seq_io', 'link-5');">seq_io</a></tt> <tt class="py-keyword">import</tt> <tt class="py-op">*</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-name">__all__</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-string">'example'</tt><tt class="py-op">,</tt> <tt class="py-string">'names'</tt><tt class="py-op">,</tt> <tt class="py-string">'extensions'</tt><tt class="py-op">,</tt> <tt class="py-string">'read'</tt><tt class="py-op">)</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt id="link-6" class="py-name" targets="Variable corebio.seq_io.array_io.example=corebio.seq_io.array_io-module.html#example,Variable corebio.seq_io.clustal_io.example=corebio.seq_io.clustal_io-module.html#example,Variable corebio.seq_io.fasta_io.example=corebio.seq_io.fasta_io-module.html#example,Variable corebio.seq_io.intelligenetics_io.example=corebio.seq_io.intelligenetics_io-module.html#example,Variable corebio.seq_io.msf_io.example=corebio.seq_io.msf_io-module.html#example,Variable corebio.seq_io.plain_io.example=corebio.seq_io.plain_io-module.html#example,Variable corebio.seq_io.stockholm_io.example=corebio.seq_io.stockholm_io-module.html#example,Variable corebio.seq_io.table_io.example=corebio.seq_io.table_io-module.html#example"><a title="corebio.seq_io.array_io.example
corebio.seq_io.clustal_io.example
corebio.seq_io.fasta_io.example
corebio.seq_io.intelligenetics_io.example
corebio.seq_io.msf_io.example
corebio.seq_io.plain_io.example
corebio.seq_io.stockholm_io.example
corebio.seq_io.table_io.example" class="py-name" href="#" onclick="return doclink('link-6', 'example', 'link-6');">example</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"""</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-string">CLUSTAL W (1.81) multiple sequence alignment</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-string"></tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-string"></tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-string">CXCR3_MOUSE       --------------------------LENSTSPYDYGENESD-------FSDSPPCPQDF</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-string">BLR_HUMAN         --------------------------LENLEDLF-WELDRLD------NYNDTSLVENH-</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-string">CXCR1_HUMAN       --------------------------MSNITDPQMWDFDDLN-------FTGMPPADEDY</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-string">CXCR4_MURINE      -----------------------------------YTSDN---------YSGSGDYDSNK</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-string">                                                     :  :          :..     ..</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-string"> </tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-string">CXCR3_MOUSE       -SL-------NFDRTFLPALYSLLFLLGLLGNGAVAAVLLSQRTALSSTDTFLLHLAVAD</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-string">BLR_HUMAN         --LC-PATMASFKAVFVPVAYSLIFLLGVIGNVLVLVILERHRQTRSSTETFLFHLAVAD</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-string">CXCR1_HUMAN       -SPC-MLETETLNKYVVIIAYALVFLLSLLGNSLVMLVILYSRVGRSVTDVYLLNLALAD</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-string">CXCR4_MURINE      -EPC-RDENVHFNRIFLPTIYFIIFLTGIVGNGLVILVMGYQKKLRSMTDKYRLHLSVAD</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-string">                             :.  .:   * ::** .::**  *  ::   :   * *: : ::*::**</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-string"></tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-string">CXCR3_MOUSE       VLLVLTLPLWAVDAA-VQWVFGPGLCKVAGALFNINFYAGAFLLACISFDRYLSIVHATQ</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-string">BLR_HUMAN         LLLVFILPFAVAEGS-VGWVLGTFLCKTVIALHKVNFYCSSLLLACIAVDRYLAIVHAVH</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-string">CXCR1_HUMAN       LLFALTLPIWAASKV-NGWIFGTFLCKVVSLLKEVNFYSGILLLACISVDRYLAIVHATR</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-string">CXCR4_MURINE      LLFVITLPFWAVDAM-ADWYFGKFLCKAVHIIYTVNLYSSVLILAFISLDRYLAIVHATN</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-string">                  :*:.: **: ...     * :*  ***..  :  :*:*.. ::** *:.****:****..</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"><tt class="py-string">"""</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt id="link-7" class="py-name" targets="Variable corebio.seq_io.array_io.names=corebio.seq_io.array_io-module.html#names,Variable corebio.seq_io.clustal_io.names=corebio.seq_io.clustal_io-module.html#names,Variable corebio.seq_io.fasta_io.names=corebio.seq_io.fasta_io-module.html#names,Variable corebio.seq_io.genbank_io.names=corebio.seq_io.genbank_io-module.html#names,Variable corebio.seq_io.intelligenetics_io.names=corebio.seq_io.intelligenetics_io-module.html#names,Variable corebio.seq_io.msf_io.names=corebio.seq_io.msf_io-module.html#names,Variable corebio.seq_io.nbrf_io.names=corebio.seq_io.nbrf_io-module.html#names,Variable corebio.seq_io.nexus_io.names=corebio.seq_io.nexus_io-module.html#names,Variable corebio.seq_io.null_io.names=corebio.seq_io.null_io-module.html#names,Variable corebio.seq_io.phylip_io.names=corebio.seq_io.phylip_io-module.html#names,Variable corebio.seq_io.plain_io.names=corebio.seq_io.plain_io-module.html#names,Variable corebio.seq_io.stockholm_io.names=corebio.seq_io.stockholm_io-module.html#names,Variable corebio.seq_io.table_io.names=corebio.seq_io.table_io-module.html#names,Static Method weblogolib.color.Color.names()=weblogolib.color.Color-class.html#names"><a title="corebio.seq_io.array_io.names
corebio.seq_io.clustal_io.names
corebio.seq_io.fasta_io.names
corebio.seq_io.genbank_io.names
corebio.seq_io.intelligenetics_io.names
corebio.seq_io.msf_io.names
corebio.seq_io.nbrf_io.names
corebio.seq_io.nexus_io.names
corebio.seq_io.null_io.names
corebio.seq_io.phylip_io.names
corebio.seq_io.plain_io.names
corebio.seq_io.stockholm_io.names
corebio.seq_io.table_io.names
weblogolib.color.Color.names" class="py-name" href="#" onclick="return doclink('link-7', 'names', 'link-7');">names</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-string">"clustal"</tt><tt class="py-op">,</tt> <tt class="py-string">"clustalw"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt id="link-8" class="py-name" targets="Variable corebio.seq_io.array_io.extensions=corebio.seq_io.array_io-module.html#extensions,Variable corebio.seq_io.clustal_io.extensions=corebio.seq_io.clustal_io-module.html#extensions,Variable corebio.seq_io.fasta_io.extensions=corebio.seq_io.fasta_io-module.html#extensions,Variable corebio.seq_io.genbank_io.extensions=corebio.seq_io.genbank_io-module.html#extensions,Variable corebio.seq_io.intelligenetics_io.extensions=corebio.seq_io.intelligenetics_io-module.html#extensions,Variable corebio.seq_io.msf_io.extensions=corebio.seq_io.msf_io-module.html#extensions,Variable corebio.seq_io.nbrf_io.extensions=corebio.seq_io.nbrf_io-module.html#extensions,Variable corebio.seq_io.nexus_io.extensions=corebio.seq_io.nexus_io-module.html#extensions,Variable corebio.seq_io.null_io.extensions=corebio.seq_io.null_io-module.html#extensions,Variable corebio.seq_io.phylip_io.extensions=corebio.seq_io.phylip_io-module.html#extensions,Variable corebio.seq_io.plain_io.extensions=corebio.seq_io.plain_io-module.html#extensions,Variable corebio.seq_io.stockholm_io.extensions=corebio.seq_io.stockholm_io-module.html#extensions,Variable corebio.seq_io.table_io.extensions=corebio.seq_io.table_io-module.html#extensions"><a title="corebio.seq_io.array_io.extensions
corebio.seq_io.clustal_io.extensions
corebio.seq_io.fasta_io.extensions
corebio.seq_io.genbank_io.extensions
corebio.seq_io.intelligenetics_io.extensions
corebio.seq_io.msf_io.extensions
corebio.seq_io.nbrf_io.extensions
corebio.seq_io.nexus_io.extensions
corebio.seq_io.null_io.extensions
corebio.seq_io.phylip_io.extensions
corebio.seq_io.plain_io.extensions
corebio.seq_io.stockholm_io.extensions
corebio.seq_io.table_io.extensions" class="py-name" href="#" onclick="return doclink('link-8', 'extensions', 'link-8');">extensions</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-string">'aln'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt id="link-9" class="py-name" targets="Variable corebio.seq_io.clustal_io.header_line=corebio.seq_io.clustal_io-module.html#header_line,Variable corebio.seq_io.stockholm_io.header_line=corebio.seq_io.stockholm_io-module.html#header_line"><a title="corebio.seq_io.clustal_io.header_line
corebio.seq_io.stockholm_io.header_line" class="py-name" href="#" onclick="return doclink('link-9', 'header_line', 'link-9');">header_line</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'(CLUSTAL.*)$'</tt><tt class="py-op">)</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-comment"># (sequence_id) (Sequence) (Optional sequence number)</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"><tt id="link-10" class="py-name" targets="Variable corebio.seq_io.clustal_io.seq_line=corebio.seq_io.clustal_io-module.html#seq_line,Variable corebio.seq_io.msf_io.seq_line=corebio.seq_io.msf_io-module.html#seq_line"><a title="corebio.seq_io.clustal_io.seq_line
corebio.seq_io.msf_io.seq_line" class="py-name" href="#" onclick="return doclink('link-10', 'seq_line', 'link-10');">seq_line</a></tt>   <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'(\s*\S+\s+)(\S+)\s*(\d*)\s*$'</tt><tt class="py-op">)</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"><tt class="py-comment"># Saved group includes variable length leading space.</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"><tt class="py-comment"># Must consult a seq_line to figure out how long the leading space is since</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-comment"># the maximum CLUSTAL ids length (normally 10 characters) can be changed.</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt id="link-11" class="py-name" targets="Variable corebio.seq_io.clustal_io.match_line=corebio.seq_io.clustal_io-module.html#match_line"><a title="corebio.seq_io.clustal_io.match_line" class="py-name" href="#" onclick="return doclink('link-11', 'match_line', 'link-11');">match_line</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'([\s:\.\*]*)$'</tt><tt class="py-op">)</tt>  </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"> </tt>
<a name="iterseq"></a><div id="iterseq-def"><a name="L90"></a><tt class="py-lineno"> 90</tt> <a class="py-toggle" href="#" id="iterseq-toggle" onclick="return toggle('iterseq');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io.clustal_io-module.html#iterseq">iterseq</a><tt class="py-op">(</tt><tt class="py-param">fin</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="iterseq-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="iterseq-expanded"><a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterate over the sequences in the file."""</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">    <tt class="py-comment"># Default implementation</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt id="link-12" class="py-name" targets="Static Method corebio.matrix.SubMatrix.read()=corebio.matrix.SubMatrix-class.html#read,Method corebio.seq_io._nexus.Nexus.read()=corebio.seq_io._nexus.Nexus-class.html#read,Function corebio.seq_io.array_io.read()=corebio.seq_io.array_io-module.html#read,Function corebio.seq_io.clustal_io.read()=corebio.seq_io.clustal_io-module.html#read,Function corebio.seq_io.fasta_io.read()=corebio.seq_io.fasta_io-module.html#read,Function corebio.seq_io.genbank_io.read()=corebio.seq_io.genbank_io-module.html#read,Function corebio.seq_io.intelligenetics_io.read()=corebio.seq_io.intelligenetics_io-module.html#read,Function corebio.seq_io.msf_io.read()=corebio.seq_io.msf_io-module.html#read,Function corebio.seq_io.nbrf_io.read()=corebio.seq_io.nbrf_io-module.html#read,Function corebio.seq_io.nexus_io.read()=corebio.seq_io.nexus_io-module.html#read,Function corebio.seq_io.null_io.read()=corebio.seq_io.null_io-module.html#read,Function corebio.seq_io.phylip_io.read()=corebio.seq_io.phylip_io-module.html#read,Function corebio.seq_io.plain_io.read()=corebio.seq_io.plain_io-module.html#read,Function corebio.seq_io.read()=corebio.seq_io-module.html#read,Function corebio.seq_io.stockholm_io.read()=corebio.seq_io.stockholm_io-module.html#read,Function corebio.seq_io.table_io.read()=corebio.seq_io.table_io-module.html#read"><a title="corebio.matrix.SubMatrix.read
corebio.seq_io._nexus.Nexus.read
corebio.seq_io.array_io.read
corebio.seq_io.clustal_io.read
corebio.seq_io.fasta_io.read
corebio.seq_io.genbank_io.read
corebio.seq_io.intelligenetics_io.read
corebio.seq_io.msf_io.read
corebio.seq_io.nbrf_io.read
corebio.seq_io.nexus_io.read
corebio.seq_io.null_io.read
corebio.seq_io.phylip_io.read
corebio.seq_io.plain_io.read
corebio.seq_io.read
corebio.seq_io.stockholm_io.read
corebio.seq_io.table_io.read" class="py-name" href="#" onclick="return doclink('link-12', 'read', 'link-12');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">fin</tt><tt class="py-op">,</tt> <tt id="link-13" class="py-name" targets="Variable corebio.matrix.Motif.alphabet=corebio.matrix.Motif-class.html#alphabet,Variable corebio.matrix.SubMatrix.alphabet=corebio.matrix.SubMatrix-class.html#alphabet,Variable corebio.seq.Seq.alphabet=corebio.seq.Seq-class.html#alphabet,Variable corebio.seq.SeqList.alphabet=corebio.seq.SeqList-class.html#alphabet"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-13', 'alphabet', 'link-13');">alphabet</a></tt><tt class="py-op">)</tt> <tt class="py-op">)</tt> </tt>
</div><a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"> </tt>
<a name="read"></a><div id="read-def"><a name="L96"></a><tt class="py-lineno"> 96</tt> <a class="py-toggle" href="#" id="read-toggle" onclick="return toggle('read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io.clustal_io-module.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">fin</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt>   </tt>
</div><a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">    <tt id="link-14" class="py-name"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-14', 'alphabet', 'link-13');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-15" class="py-name" targets="Class corebio.seq.Alphabet=corebio.seq.Alphabet-class.html"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-15', 'Alphabet', 'link-15');">Alphabet</a></tt><tt class="py-op">(</tt><tt id="link-16" class="py-name"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-16', 'alphabet', 'link-13');">alphabet</a></tt><tt class="py-op">)</tt>       </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">    <tt class="py-name">seq_ids</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">    <tt class="py-name">seqs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">    <tt class="py-name">block_count</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">    <tt class="py-name">data_len</tt> <tt class="py-op">=</tt><tt class="py-number">0</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">token</tt> <tt class="py-keyword">in</tt> <tt class="py-name">_scan</tt><tt class="py-op">(</tt><tt class="py-name">fin</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">token</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Variable corebio.utils.Token.typeof=corebio.utils.Token-class.html#typeof"><a title="corebio.utils.Token.typeof" class="py-name" href="#" onclick="return doclink('link-17', 'typeof', 'link-17');">typeof</a></tt><tt class="py-op">==</tt> <tt class="py-string">"begin_block"</tt><tt class="py-op">:</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">            <tt class="py-name">block_count</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">token</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="corebio.utils.Token.typeof" class="py-name" href="#" onclick="return doclink('link-18', 'typeof', 'link-17');">typeof</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"seq_id"</tt><tt class="py-op">:</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seqs</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">block_count</tt> <tt class="py-op">:</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">                <tt class="py-name">seq_ids</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name" targets="Method corebio.db.astral.RafSeqMap.append()=corebio.db.astral.RafSeqMap-class.html#append"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-19', 'append', 'link-19');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">token</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Module corebio.data=corebio.data-module.html,Variable corebio.utils.Token.data=corebio.utils.Token-class.html#data"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-20', 'data', 'link-20');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">                <tt class="py-name">seqs</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-21', 'append', 'link-19');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">token</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="corebio.utils.Token.typeof" class="py-name" href="#" onclick="return doclink('link-22', 'typeof', 'link-17');">typeof</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"seq"</tt><tt class="py-op">:</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-23" class="py-name"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-23', 'alphabet', 'link-13');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Method corebio.seq.Alphabet.alphabetic()=corebio.seq.Alphabet-class.html#alphabetic"><a title="corebio.seq.Alphabet.alphabetic" class="py-name" href="#" onclick="return doclink('link-24', 'alphabetic', 'link-24');">alphabetic</a></tt><tt class="py-op">(</tt><tt class="py-name">token</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-25', 'data', 'link-20');">data</a></tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">                    <tt class="py-string">"Character on line: %d not in alphabet: %s : %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">                    <tt class="py-name">token</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Variable corebio.utils.Token.lineno=corebio.utils.Token-class.html#lineno"><a title="corebio.utils.Token.lineno" class="py-name" href="#" onclick="return doclink('link-26', 'lineno', 'link-26');">lineno</a></tt><tt class="py-op">,</tt> <tt id="link-27" class="py-name"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-27', 'alphabet', 'link-13');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">token</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-28', 'data', 'link-20');">data</a></tt><tt class="py-op">)</tt> <tt class="py-op">)</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">            <tt class="py-name">seqs</tt><tt class="py-op">[</tt><tt class="py-name">block_count</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-29', 'append', 'link-19');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">token</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-30', 'data', 'link-20');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">block_count</tt><tt class="py-op">==</tt><tt class="py-number">0</tt> <tt class="py-op">:</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">                <tt class="py-name">data_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">token</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-31', 'data', 'link-20');">data</a></tt><tt class="py-op">)</tt>  </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">data_len</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">token</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-32', 'data', 'link-20');">data</a></tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Inconsistent line lengths"</tt><tt class="py-op">)</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">                 </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">            <tt class="py-name">block_count</tt> <tt class="py-op">+=</tt><tt class="py-number">1</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">           </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">    <tt class="py-name">seqs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> <tt id="link-33" class="py-name" targets="Class corebio.seq.Seq=corebio.seq.Seq-class.html"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-33', 'Seq', 'link-33');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Method corebio.seq.Seq.join()=corebio.seq.Seq-class.html#join"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-34', 'join', 'link-34');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-35" class="py-name"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-35', 'alphabet', 'link-13');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-36" class="py-name" targets="Variable corebio.db.DataSource.name=corebio.db.DataSource-class.html#name,Variable corebio.matrix.SubMatrix.name=corebio.matrix.SubMatrix-class.html#name,Variable corebio.seq.SeqList.name=corebio.seq.SeqList-class.html#name,Variable corebio.transform.Transform.name=corebio.transform.Transform-class.html#name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-36', 'name', 'link-36');">name</a></tt><tt class="py-op">=</tt> <tt class="py-name">i</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">s</tt><tt class="py-op">,</tt><tt class="py-name">i</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">seqs</tt><tt class="py-op">,</tt><tt class="py-name">seq_ids</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-37" class="py-name" targets="Class corebio.seq.SeqList=corebio.seq.SeqList-class.html"><a title="corebio.seq.SeqList" class="py-name" href="#" onclick="return doclink('link-37', 'SeqList', 'link-37');">SeqList</a></tt><tt class="py-op">(</tt><tt class="py-name">seqs</tt><tt class="py-op">)</tt> </tt>
</div><a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-comment"># 1) The word "CLUSTAL" should be the first word on the first line of the file.</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-comment"># (But sometimes isn't.)</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-comment"># 2) The alignment is displayed in blocks of fixed length.</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-comment"># 3) Each line in the block corresponds to one sequence.</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-comment"># 4) Each sequence line starts with a sequence name followed by at least one</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-comment">#     space and then the sequence.</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"> </tt>
<a name="_scan"></a><div id="_scan-def"><a name="L135"></a><tt class="py-lineno">135</tt> <a class="py-toggle" href="#" id="_scan-toggle" onclick="return toggle('_scan');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io.clustal_io-module.html#_scan">_scan</a><tt class="py-op">(</tt> <tt class="py-param">fin</tt> <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_scan-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_scan-expanded"><a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">    <tt class="py-docstring">"""Scan a clustal format MSA file and yield tokens.</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"><tt class="py-docstring">        The basic file structure is</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"><tt class="py-docstring">            begin_document</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt class="py-docstring">                header?     </tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-docstring">               (begin_block</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-docstring">                   (seq_id seq seq_index?)+</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-docstring">                   match_line?</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"><tt class="py-docstring">               end_block)*</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"><tt class="py-docstring">            end_document     </tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"><tt class="py-docstring">        Usage:</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"><tt class="py-docstring">        for token in scan(clustal_file):</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"><tt class="py-docstring">            do_something(token)</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">    <tt class="py-name">header</tt><tt class="py-op">,</tt> <tt class="py-name">body</tt><tt class="py-op">,</tt> <tt class="py-name">block</tt> <tt class="py-op">=</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">)</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">     </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">    <tt class="py-keyword">yield</tt> <tt id="link-38" class="py-name" targets="Class corebio.utils.Token=corebio.utils.Token-class.html"><a title="corebio.utils.Token" class="py-name" href="#" onclick="return doclink('link-38', 'Token', 'link-38');">Token</a></tt><tt class="py-op">(</tt><tt class="py-string">"begin"</tt><tt class="py-op">)</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">    <tt class="py-name">leader_width</tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">    <tt class="py-name">state</tt> <tt class="py-op">=</tt> <tt class="py-name">header</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">L</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">fin</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">state</tt><tt class="py-op">==</tt><tt class="py-name">header</tt> <tt class="py-op">:</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">isspace</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> <tt class="py-keyword">continue</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">            <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt id="link-39" class="py-name"><a title="corebio.seq_io.clustal_io.header_line
corebio.seq_io.stockholm_io.header_line" class="py-name" href="#" onclick="return doclink('link-39', 'header_line', 'link-9');">header_line</a></tt><tt class="py-op">.</tt><tt class="py-name">match</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">            <tt class="py-name">state</tt> <tt class="py-op">=</tt> <tt class="py-name">body</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">m</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-op">:</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt id="link-40" class="py-name"><a title="corebio.utils.Token" class="py-name" href="#" onclick="return doclink('link-40', 'Token', 'link-38');">Token</a></tt><tt class="py-op">(</tt><tt class="py-string">"header"</tt><tt class="py-op">,</tt> <tt class="py-name">m</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">            <tt class="py-comment"># Just keep going and hope for the best.</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">            <tt class="py-comment">#else :</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">                <tt class="py-comment">#raise ValueError("Cannot find required header")</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">                 </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">                 </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">         </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">state</tt> <tt class="py-op">==</tt> <tt class="py-name">body</tt> <tt class="py-op">:</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">isspace</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> <tt class="py-keyword">continue</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt id="link-41" class="py-name"><a title="corebio.utils.Token" class="py-name" href="#" onclick="return doclink('link-41', 'Token', 'link-38');">Token</a></tt><tt class="py-op">(</tt><tt class="py-string">"begin_block"</tt><tt class="py-op">)</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">            <tt class="py-name">state</tt> <tt class="py-op">=</tt> <tt class="py-name">block</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">            <tt class="py-comment"># fall through to block</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">         </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">state</tt> <tt class="py-op">==</tt>  <tt class="py-name">block</tt><tt class="py-op">:</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">isspace</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt id="link-42" class="py-name"><a title="corebio.utils.Token" class="py-name" href="#" onclick="return doclink('link-42', 'Token', 'link-38');">Token</a></tt><tt class="py-op">(</tt><tt class="py-string">"end_block"</tt><tt class="py-op">)</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">                <tt class="py-name">state</tt> <tt class="py-op">=</tt> <tt class="py-name">body</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">             </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">            <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt id="link-43" class="py-name"><a title="corebio.seq_io.clustal_io.match_line" class="py-name" href="#" onclick="return doclink('link-43', 'match_line', 'link-11');">match_line</a></tt><tt class="py-op">.</tt><tt class="py-name">match</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">m</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-op">:</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt id="link-44" class="py-name"><a title="corebio.utils.Token" class="py-name" href="#" onclick="return doclink('link-44', 'Token', 'link-38');">Token</a></tt><tt class="py-op">(</tt><tt class="py-string">"match_line"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">leader_width</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">      </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">            <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt id="link-45" class="py-name"><a title="corebio.seq_io.clustal_io.seq_line
corebio.seq_io.msf_io.seq_line" class="py-name" href="#" onclick="return doclink('link-45', 'seq_line', 'link-10');">seq_line</a></tt><tt class="py-op">.</tt><tt class="py-name">match</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt>  </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">m</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt>  </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Parse error on line: %d (%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">L</tt><tt class="py-op">,</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">            <tt class="py-name">leader_width</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">m</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt id="link-46" class="py-name"><a title="corebio.utils.Token" class="py-name" href="#" onclick="return doclink('link-46', 'Token', 'link-38');">Token</a></tt><tt class="py-op">(</tt><tt class="py-string">"seq_id"</tt><tt class="py-op">,</tt> <tt class="py-name">m</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">strip</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt id="link-47" class="py-name"><a title="corebio.utils.Token" class="py-name" href="#" onclick="return doclink('link-47', 'Token', 'link-38');">Token</a></tt><tt class="py-op">(</tt><tt class="py-string">"seq"</tt><tt class="py-op">,</tt> <tt class="py-name">m</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">strip</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">m</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">)</tt>  <tt class="py-op">:</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt id="link-48" class="py-name"><a title="corebio.utils.Token" class="py-name" href="#" onclick="return doclink('link-48', 'Token', 'link-38');">Token</a></tt><tt class="py-op">(</tt><tt class="py-string">"seq_num"</tt><tt class="py-op">,</tt> <tt class="py-name">m</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>  </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">            <tt class="py-keyword">continue</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line"> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">        <tt class="py-comment"># END state blocks. If I ever get here something has gone terrible wrong</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">     </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">state</tt><tt class="py-op">==</tt><tt class="py-name">block</tt><tt class="py-op">:</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">         <tt class="py-keyword">yield</tt> <tt id="link-49" class="py-name"><a title="corebio.utils.Token" class="py-name" href="#" onclick="return doclink('link-49', 'Token', 'link-38');">Token</a></tt><tt class="py-op">(</tt><tt class="py-string">"end_block"</tt><tt class="py-op">)</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">    <tt class="py-keyword">yield</tt> <tt id="link-50" class="py-name"><a title="corebio.utils.Token" class="py-name" href="#" onclick="return doclink('link-50', 'Token', 'link-38');">Token</a></tt><tt class="py-op">(</tt><tt class="py-string">"end"</tt><tt class="py-op">)</tt>      </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> </tt>
</div><a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"> </tt>
<a name="write"></a><div id="write-def"><a name="L204"></a><tt class="py-lineno">204</tt> <a class="py-toggle" href="#" id="write-toggle" onclick="return toggle('write');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io.clustal_io-module.html#write">write</a><tt class="py-op">(</tt><tt class="py-param">fout</tt><tt class="py-op">,</tt> <tt class="py-param">seqs</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="write-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="write-expanded"><a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">    <tt class="py-docstring">"""Write 'seqs' to 'fout' as text in clustal format"""</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">    <tt class="py-name">header</tt> <tt class="py-op">=</tt> <tt class="py-string">"CLUSTAL W (1.81) multiple sequence alignment"</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">    <tt class="py-name">name_width</tt> <tt class="py-op">=</tt> <tt class="py-number">17</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">    <tt class="py-name">seq_width</tt> <tt class="py-op">=</tt> <tt class="py-number">60</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">     </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt> <tt class="py-op">&gt;&gt;</tt><tt class="py-name">fout</tt><tt class="py-op">,</tt> <tt class="py-name">header</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt> <tt class="py-op">&gt;&gt;</tt><tt class="py-name">fout</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt> <tt class="py-op">&gt;&gt;</tt><tt class="py-name">fout</tt>  </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">     </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">    <tt class="py-name">L</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt class="py-name">seqs</tt><tt class="py-op">:</tt> <tt class="py-name">L</tt> <tt class="py-op">=</tt> <tt class="py-name">max</tt><tt class="py-op">(</tt><tt class="py-name">L</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">     </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">block</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">L</tt><tt class="py-op">,</tt> <tt class="py-name">seq_width</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">s</tt>  <tt class="py-keyword">in</tt> <tt class="py-name">seqs</tt> <tt class="py-op">:</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">            <tt class="py-name">start</tt> <tt class="py-op">=</tt> <tt class="py-name">min</tt><tt class="py-op">(</tt><tt class="py-name">block</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">            <tt class="py-name">end</tt> <tt class="py-op">=</tt> <tt class="py-name">min</tt><tt class="py-op">(</tt> <tt class="py-name">start</tt><tt class="py-op">+</tt><tt class="py-name">seq_width</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">            <tt class="py-keyword">print</tt>  <tt class="py-op">&gt;&gt;</tt><tt class="py-name">fout</tt><tt class="py-op">,</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-51', 'name', 'link-36');">name</a></tt><tt class="py-op">.</tt><tt class="py-name">ljust</tt><tt class="py-op">(</tt><tt class="py-name">name_width</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">            <tt class="py-keyword">print</tt>  <tt class="py-op">&gt;&gt;</tt><tt class="py-name">fout</tt><tt class="py-op">,</tt> <tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-name">start</tt><tt class="py-op">:</tt><tt class="py-name">end</tt><tt class="py-op">]</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt>  <tt class="py-op">&gt;&gt;</tt><tt class="py-name">fout</tt> </tt>
</div><a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Mar 12 21:34:13 2014
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
