<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>corebio</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        Package&nbsp;corebio
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="corebio-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== PACKAGE DESCRIPTION ==================== -->
<h1 class="epydoc">Package corebio</h1><p class="nomargin-top"><span class="codelink"><a href="corebio-pysrc.html">source&nbsp;code</a></span></p>
<pre class="literalblock">
CoreBio 3.3 (2012-01-31) :  A python toolkit for computational biology. 

http://code.google.com/p/corebio/

Attributes:
    __version__ - The corebio version.
    description - The library name, version and release date.
    user_agent  - The http user agent string used by corebio. 

Environment Variables:
    - COREBIOPATH -- An alternative search path for shell commands. Used
        by corebio.utils.find_command
    

</pre>

<hr />
<div class="fields">      <p><strong>Version:</strong>
        3.3
      </p>
</div><!-- ==================== SUBMODULES ==================== -->
<a name="section-Submodules"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td align="left" colspan="2" class="table-header">
    <span class="table-header">Submodules</span></td>
</tr>
  <tr><td class="summary">
  <ul class="nomargin">
    <li class="private"> <strong class="uidlink"><a href="corebio._version-module.html" onclick="show_private();">corebio._version</a></strong>    </li>
    <li> <strong class="uidlink"><a href="corebio.data-module.html">corebio.data</a></strong>: <em class="summary">Standard data used in computational biology.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.db-module.html">corebio.db</a></strong>: <em class="summary">Database access.</em>
    <ul>
    <li> <strong class="uidlink"><a href="corebio.db.astral-module.html">corebio.db.astral</a></strong>: <em class="summary">ASTRAL:  Compendium for Sequence and Structure Analysis.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.db.scop-module.html">corebio.db.scop</a></strong>: <em class="summary">SCOP: Structural Classification of Proteins.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="corebio.matrix-module.html">corebio.matrix</a></strong>: <em class="summary">Arrays indexed by alphabetic strings.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.moremath-module.html">corebio.moremath</a></strong>: <em class="summary">Various bits of useful math not in the standard python library.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.secstruc-module.html">corebio.secstruc</a></strong>: <em class="summary">Protein secondary structure and associated tools and data.</em>
    <ul>
    <li> <strong class="uidlink"><a href="corebio.secstruc.dssp-module.html">corebio.secstruc.dssp</a></strong>: <em class="summary">Command and control of the program DSSP: Dictionary of protein secondary structure.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.secstruc.stride-module.html">corebio.secstruc.stride</a></strong>: <em class="summary">Command and control of the program STRIDE: Protein secondary 
        structure assignment from atomic coordinates.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="corebio.seq-module.html">corebio.seq</a></strong>: <em class="summary">Alphabetic sequences and associated tools and data.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io-module.html">corebio.seq_io</a></strong>: <em class="summary">Sequence file reading and writing.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="corebio.seq_io._nexus-module.html" onclick="show_private();">corebio.seq_io._nexus</a></strong>: <em class="summary">Parse the contents of a nexus file.</em>
    <ul class="private">
    <li> <strong class="uidlink"><a href="corebio.seq_io._nexus.Nodes-module.html">corebio.seq_io._nexus.Nodes</a></strong>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io._nexus.Trees-module.html">corebio.seq_io._nexus.Trees</a></strong>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.array_io-module.html">corebio.seq_io.array_io</a></strong>: <em class="summary">Read and write a rectangular array of sequence data.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.clustal_io-module.html">corebio.seq_io.clustal_io</a></strong>: <em class="summary">Read and write the CLUSTAL sequence file format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.fasta_io-module.html">corebio.seq_io.fasta_io</a></strong>: <em class="summary">Read and write sequence information in FASTA format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.genbank_io-module.html">corebio.seq_io.genbank_io</a></strong>: <em class="summary">Read GenBank flat files.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.intelligenetics_io-module.html">corebio.seq_io.intelligenetics_io</a></strong>: <em class="summary">Read and write sequence information in IntelliGenetics format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.msf_io-module.html">corebio.seq_io.msf_io</a></strong>: <em class="summary">Read sequence information in MSF format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.nbrf_io-module.html">corebio.seq_io.nbrf_io</a></strong>: <em class="summary">Sequence IO for NBRF/PIR format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.nexus_io-module.html">corebio.seq_io.nexus_io</a></strong>: <em class="summary">Read the sequence data from a nexus file.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.null_io-module.html">corebio.seq_io.null_io</a></strong>: <em class="summary">Null sequence IO.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.phylip_io-module.html">corebio.seq_io.phylip_io</a></strong>: <em class="summary">Reads Sequences in interleaved Phylip format (not sequential) and returns a
list of sequences.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.plain_io-module.html">corebio.seq_io.plain_io</a></strong>: <em class="summary">Read and write raw, unformatted sequence data.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.stockholm_io-module.html">corebio.seq_io.stockholm_io</a></strong>: <em class="summary">Read a multiple sequence alignment in STOCKHOLM format.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.seq_io.table_io-module.html">corebio.seq_io.table_io</a></strong>: <em class="summary">Read and write sequence information in tab-delimited format.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="corebio.transform-module.html">corebio.transform</a></strong>: <em class="summary">Transformations of Seqs (alphabetic sequences).</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.utils-module.html">corebio.utils</a></strong>: <em class="summary">Extra utilities and core classes not in standard python.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="corebio.utils._which-module.html">corebio.utils._which</a></strong>: <em class="summary">Find the full path to commands.</em>    </li>
    <li> <strong class="uidlink"><a href="corebio.utils.deoptparse-module.html">corebio.utils.deoptparse</a></strong>: <em class="summary">Custom extensions to OptionParse for parsing command line options.</em>    </li>
    </ul>
    </li>
  </ul></td></tr>
</table>

<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Mar 12 21:34:12 2014
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
