<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>corebio.transform</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="corebio-module.html">Package&nbsp;corebio</a> ::
        Module&nbsp;transform
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="corebio.transform-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="corebio.transform-module.html">Module corebio.transform</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment">#  Copyright (c) 2006 John Gilman</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment">#  This software is distributed under the MIT Open Source License.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment">#  &lt;http://www.opensource.org/licenses/mit-license.html&gt;</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment">#  Permission is hereby granted, free of charge, to any person obtaining a </tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment">#  copy of this software and associated documentation files (the "Software"),</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment">#  to deal in the Software without restriction, including without limitation</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-comment">#  the rights to use, copy, modify, merge, publish, distribute, sublicense,</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-comment">#  and/or sell copies of the Software, and to permit persons to whom the</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-comment">#  Software is furnished to do so, subject to the following conditions:</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-comment">#  The above copyright notice and this permission notice shall be included</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-comment">#  in all copies or substantial portions of the Software.</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-comment">#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR </tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-comment">#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, </tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-comment">#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-comment">#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER </tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-comment">#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-comment">#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN </tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-comment">#  THE SOFTWARE.</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">""" Transformations of Seqs (alphabetic sequences).</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">Classes :</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">- Transform   -- Simple transforms of alphabetic strings.</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">- GeneticCode -- The genetic mapping of DNA to protein.</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">Functions :</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">-  mask_low_complexity -- Implementation of Seg algorithm to remove low complexity  </tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">        regions from protein sequences.</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">Other:</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">-   reduced_protein_alphabets -- A dictionary of transforms that reduce the size of the protein alphabet,</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">        merging various amino acids into classes.</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">        "LiBn" where n is 2 to 19 are from Li et al (2003), table I, 2 to 19 groups.</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">        "LiBn" where n is 2 to 19 are from Li et al (2003), table II (no interlacing), 2 to 19 groups.</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring">        Ref: Li et al Reduction of protein sequence complexity by residue grouping, </tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">            Prot. Eng. 16 323-330 (2003)  </tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package corebio=corebio-module.html"><a title="corebio" class="py-name" href="#" onclick="return doclink('link-0', 'corebio', 'link-0');">corebio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Module corebio.data=corebio.data-module.html,Variable corebio.utils.Token.data=corebio.utils.Token-class.html#data"><a title="corebio.data
corebio.utils.Token.data" class="py-name" href="#" onclick="return doclink('link-1', 'data', 'link-1');">data</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Variable corebio.data.dna_extended_letters=corebio.data-module.html#dna_extended_letters"><a title="corebio.data.dna_extended_letters" class="py-name" href="#" onclick="return doclink('link-2', 'dna_extended_letters', 'link-2');">dna_extended_letters</a></tt><tt class="py-op">,</tt> <tt id="link-3" class="py-name" targets="Variable corebio.data.dna_ambiguity=corebio.data-module.html#dna_ambiguity"><a title="corebio.data.dna_ambiguity" class="py-name" href="#" onclick="return doclink('link-3', 'dna_ambiguity', 'link-3');">dna_ambiguity</a></tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="corebio" class="py-name" href="#" onclick="return doclink('link-4', 'corebio', 'link-0');">corebio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Module corebio.seq=corebio.seq-module.html"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-5', 'seq', 'link-5');">seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-6" class="py-name" targets="Class corebio.seq.Seq=corebio.seq.Seq-class.html"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-6', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-7" class="py-name" targets="Variable corebio.seq.protein_alphabet=corebio.seq-module.html#protein_alphabet,Variable corebio.seq_io.array_io.protein_alphabet=corebio.seq_io.array_io-module.html#protein_alphabet,Variable corebio.seq_io.clustal_io.protein_alphabet=corebio.seq_io.clustal_io-module.html#protein_alphabet,Variable corebio.seq_io.fasta_io.protein_alphabet=corebio.seq_io.fasta_io-module.html#protein_alphabet,Variable corebio.seq_io.genbank_io.protein_alphabet=corebio.seq_io.genbank_io-module.html#protein_alphabet,Variable corebio.seq_io.intelligenetics_io.protein_alphabet=corebio.seq_io.intelligenetics_io-module.html#protein_alphabet,Variable corebio.seq_io.msf_io.protein_alphabet=corebio.seq_io.msf_io-module.html#protein_alphabet,Variable corebio.seq_io.nbrf_io.protein_alphabet=corebio.seq_io.nbrf_io-module.html#protein_alphabet,Variable corebio.seq_io.phylip_io.protein_alphabet=corebio.seq_io.phylip_io-module.html#protein_alphabet,Variable corebio.seq_io.plain_io.protein_alphabet=corebio.seq_io.plain_io-module.html#protein_alphabet,Variable corebio.seq_io.protein_alphabet=corebio.seq_io-module.html#protein_alphabet,Variable corebio.seq_io.stockholm_io.protein_alphabet=corebio.seq_io.stockholm_io-module.html#protein_alphabet,Variable corebio.seq_io.table_io.protein_alphabet=corebio.seq_io.table_io-module.html#protein_alphabet"><a title="corebio.seq.protein_alphabet
corebio.seq_io.array_io.protein_alphabet
corebio.seq_io.clustal_io.protein_alphabet
corebio.seq_io.fasta_io.protein_alphabet
corebio.seq_io.genbank_io.protein_alphabet
corebio.seq_io.intelligenetics_io.protein_alphabet
corebio.seq_io.msf_io.protein_alphabet
corebio.seq_io.nbrf_io.protein_alphabet
corebio.seq_io.phylip_io.protein_alphabet
corebio.seq_io.plain_io.protein_alphabet
corebio.seq_io.protein_alphabet
corebio.seq_io.stockholm_io.protein_alphabet
corebio.seq_io.table_io.protein_alphabet" class="py-name" href="#" onclick="return doclink('link-7', 'protein_alphabet', 'link-7');">protein_alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-8" class="py-name" targets="Variable corebio.seq.nucleic_alphabet=corebio.seq-module.html#nucleic_alphabet,Variable corebio.seq_io.array_io.nucleic_alphabet=corebio.seq_io.array_io-module.html#nucleic_alphabet,Variable corebio.seq_io.clustal_io.nucleic_alphabet=corebio.seq_io.clustal_io-module.html#nucleic_alphabet,Variable corebio.seq_io.fasta_io.nucleic_alphabet=corebio.seq_io.fasta_io-module.html#nucleic_alphabet,Variable corebio.seq_io.genbank_io.nucleic_alphabet=corebio.seq_io.genbank_io-module.html#nucleic_alphabet,Variable corebio.seq_io.intelligenetics_io.nucleic_alphabet=corebio.seq_io.intelligenetics_io-module.html#nucleic_alphabet,Variable corebio.seq_io.msf_io.nucleic_alphabet=corebio.seq_io.msf_io-module.html#nucleic_alphabet,Variable corebio.seq_io.nbrf_io.nucleic_alphabet=corebio.seq_io.nbrf_io-module.html#nucleic_alphabet,Variable corebio.seq_io.nucleic_alphabet=corebio.seq_io-module.html#nucleic_alphabet,Variable corebio.seq_io.phylip_io.nucleic_alphabet=corebio.seq_io.phylip_io-module.html#nucleic_alphabet,Variable corebio.seq_io.plain_io.nucleic_alphabet=corebio.seq_io.plain_io-module.html#nucleic_alphabet,Variable corebio.seq_io.stockholm_io.nucleic_alphabet=corebio.seq_io.stockholm_io-module.html#nucleic_alphabet,Variable corebio.seq_io.table_io.nucleic_alphabet=corebio.seq_io.table_io-module.html#nucleic_alphabet"><a title="corebio.seq.nucleic_alphabet
corebio.seq_io.array_io.nucleic_alphabet
corebio.seq_io.clustal_io.nucleic_alphabet
corebio.seq_io.fasta_io.nucleic_alphabet
corebio.seq_io.genbank_io.nucleic_alphabet
corebio.seq_io.intelligenetics_io.nucleic_alphabet
corebio.seq_io.msf_io.nucleic_alphabet
corebio.seq_io.nbrf_io.nucleic_alphabet
corebio.seq_io.nucleic_alphabet
corebio.seq_io.phylip_io.nucleic_alphabet
corebio.seq_io.plain_io.nucleic_alphabet
corebio.seq_io.stockholm_io.nucleic_alphabet
corebio.seq_io.table_io.nucleic_alphabet" class="py-name" href="#" onclick="return doclink('link-8', 'nucleic_alphabet', 'link-8');">nucleic_alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-9" class="py-name" targets="Variable corebio.seq.dna_alphabet=corebio.seq-module.html#dna_alphabet,Variable corebio.seq_io.array_io.dna_alphabet=corebio.seq_io.array_io-module.html#dna_alphabet,Variable corebio.seq_io.clustal_io.dna_alphabet=corebio.seq_io.clustal_io-module.html#dna_alphabet,Variable corebio.seq_io.dna_alphabet=corebio.seq_io-module.html#dna_alphabet,Variable corebio.seq_io.fasta_io.dna_alphabet=corebio.seq_io.fasta_io-module.html#dna_alphabet,Variable corebio.seq_io.genbank_io.dna_alphabet=corebio.seq_io.genbank_io-module.html#dna_alphabet,Variable corebio.seq_io.intelligenetics_io.dna_alphabet=corebio.seq_io.intelligenetics_io-module.html#dna_alphabet,Variable corebio.seq_io.msf_io.dna_alphabet=corebio.seq_io.msf_io-module.html#dna_alphabet,Variable corebio.seq_io.nbrf_io.dna_alphabet=corebio.seq_io.nbrf_io-module.html#dna_alphabet,Variable corebio.seq_io.phylip_io.dna_alphabet=corebio.seq_io.phylip_io-module.html#dna_alphabet,Variable corebio.seq_io.plain_io.dna_alphabet=corebio.seq_io.plain_io-module.html#dna_alphabet,Variable corebio.seq_io.stockholm_io.dna_alphabet=corebio.seq_io.stockholm_io-module.html#dna_alphabet,Variable corebio.seq_io.table_io.dna_alphabet=corebio.seq_io.table_io-module.html#dna_alphabet"><a title="corebio.seq.dna_alphabet
corebio.seq_io.array_io.dna_alphabet
corebio.seq_io.clustal_io.dna_alphabet
corebio.seq_io.dna_alphabet
corebio.seq_io.fasta_io.dna_alphabet
corebio.seq_io.genbank_io.dna_alphabet
corebio.seq_io.intelligenetics_io.dna_alphabet
corebio.seq_io.msf_io.dna_alphabet
corebio.seq_io.nbrf_io.dna_alphabet
corebio.seq_io.phylip_io.dna_alphabet
corebio.seq_io.plain_io.dna_alphabet
corebio.seq_io.stockholm_io.dna_alphabet
corebio.seq_io.table_io.dna_alphabet" class="py-name" href="#" onclick="return doclink('link-9', 'dna_alphabet', 'link-9');">dna_alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-10" class="py-name" targets="Class corebio.seq.Alphabet=corebio.seq.Alphabet-class.html"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-10', 'Alphabet', 'link-10');">Alphabet</a></tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-11" class="py-name"><a title="corebio" class="py-name" href="#" onclick="return doclink('link-11', 'corebio', 'link-0');">corebio</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-12', 'seq', 'link-5');">seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Variable corebio.seq.reduced_protein_alphabet=corebio.seq-module.html#reduced_protein_alphabet,Variable corebio.seq_io.array_io.reduced_protein_alphabet=corebio.seq_io.array_io-module.html#reduced_protein_alphabet,Variable corebio.seq_io.clustal_io.reduced_protein_alphabet=corebio.seq_io.clustal_io-module.html#reduced_protein_alphabet,Variable corebio.seq_io.fasta_io.reduced_protein_alphabet=corebio.seq_io.fasta_io-module.html#reduced_protein_alphabet,Variable corebio.seq_io.genbank_io.reduced_protein_alphabet=corebio.seq_io.genbank_io-module.html#reduced_protein_alphabet,Variable corebio.seq_io.intelligenetics_io.reduced_protein_alphabet=corebio.seq_io.intelligenetics_io-module.html#reduced_protein_alphabet,Variable corebio.seq_io.msf_io.reduced_protein_alphabet=corebio.seq_io.msf_io-module.html#reduced_protein_alphabet,Variable corebio.seq_io.nbrf_io.reduced_protein_alphabet=corebio.seq_io.nbrf_io-module.html#reduced_protein_alphabet,Variable corebio.seq_io.phylip_io.reduced_protein_alphabet=corebio.seq_io.phylip_io-module.html#reduced_protein_alphabet,Variable corebio.seq_io.plain_io.reduced_protein_alphabet=corebio.seq_io.plain_io-module.html#reduced_protein_alphabet,Variable corebio.seq_io.reduced_protein_alphabet=corebio.seq_io-module.html#reduced_protein_alphabet,Variable corebio.seq_io.stockholm_io.reduced_protein_alphabet=corebio.seq_io.stockholm_io-module.html#reduced_protein_alphabet,Variable corebio.seq_io.table_io.reduced_protein_alphabet=corebio.seq_io.table_io-module.html#reduced_protein_alphabet"><a title="corebio.seq.reduced_protein_alphabet
corebio.seq_io.array_io.reduced_protein_alphabet
corebio.seq_io.clustal_io.reduced_protein_alphabet
corebio.seq_io.fasta_io.reduced_protein_alphabet
corebio.seq_io.genbank_io.reduced_protein_alphabet
corebio.seq_io.intelligenetics_io.reduced_protein_alphabet
corebio.seq_io.msf_io.reduced_protein_alphabet
corebio.seq_io.nbrf_io.reduced_protein_alphabet
corebio.seq_io.phylip_io.reduced_protein_alphabet
corebio.seq_io.plain_io.reduced_protein_alphabet
corebio.seq_io.reduced_protein_alphabet
corebio.seq_io.stockholm_io.reduced_protein_alphabet
corebio.seq_io.table_io.reduced_protein_alphabet" class="py-name" href="#" onclick="return doclink('link-13', 'reduced_protein_alphabet', 'link-13');">reduced_protein_alphabet</a></tt> <tt class="py-keyword">as</tt> <tt class="py-name">std_protein_alphabet</tt>  </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">string</tt> <tt class="py-keyword">import</tt> <tt class="py-name">maketrans</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-14" class="py-name"><a title="corebio" class="py-name" href="#" onclick="return doclink('link-14', 'corebio', 'link-0');">corebio</a></tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Module corebio.moremath=corebio.moremath-module.html"><a title="corebio.moremath" class="py-name" href="#" onclick="return doclink('link-15', 'moremath', 'link-15');">moremath</a></tt> <tt class="py-keyword">import</tt> <tt id="link-16" class="py-name" targets="Function corebio.moremath.log2()=corebio.moremath-module.html#log2"><a title="corebio.moremath.log2" class="py-name" href="#" onclick="return doclink('link-16', 'log2', 'link-16');">log2</a></tt> <tt class="py-op">,</tt> <tt id="link-17" class="py-name" targets="Function corebio.moremath.entropy()=corebio.moremath-module.html#entropy"><a title="corebio.moremath.entropy" class="py-name" href="#" onclick="return doclink('link-17', 'entropy', 'link-17');">entropy</a></tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-name">__all__</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">    <tt class="py-string">'Transform'</tt><tt class="py-op">,</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">    <tt class="py-string">'mask_low_complexity'</tt><tt class="py-op">,</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">    <tt class="py-string">'GeneticCode'</tt><tt class="py-op">,</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">    <tt class="py-string">'reduced_protein_alphabets'</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">    <tt class="py-op">]</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"> </tt>
<a name="Transform"></a><div id="Transform-def"><a name="L64"></a><tt class="py-lineno"> 64</tt> <a class="py-toggle" href="#" id="Transform-toggle" onclick="return toggle('Transform');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="corebio.transform.Transform-class.html">Transform</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="Transform-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Transform-expanded"><a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">    <tt class="py-docstring">"""A translation between alphabetic strings.</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring">    (This class is not called 'Translation' to avoid confusion with the</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">    biological translation of RNA to protein.)</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring">    Example:</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-docstring">    trans = Transform( </tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"><tt class="py-docstring">        Seq("ACGTRYSWKMBDHVN-acgtUuryswkmbdhvnXx?.~", dna_alphabet),                    </tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-docstring">        Seq("ACGTRYSWKMNNNNN-acgtUuryswkmbnnnnXx?.~", reduced_nucleic_alphabet)         </tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-docstring">        )</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt class="py-docstring">    s0 = Seq("AAAAAV", nucleic_alphabet)</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-docstring">    s1 = trans(s0)              </tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-docstring">    assert(s1.alphabet == reduced_nucleic_alphabet)</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring">    assert(s2 == Seq("AAAAAN",  reduced_nucleic_alphabet)</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring">    Status : Beta </tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">     </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">    <tt class="py-name">__slots__</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"table"</tt><tt class="py-op">,</tt> <tt class="py-string">"source"</tt><tt class="py-op">,</tt> <tt class="py-string">"target"</tt><tt class="py-op">,</tt><tt class="py-string">"name"</tt><tt class="py-op">,</tt><tt class="py-string">"description"</tt><tt class="py-op">]</tt> </tt>
<a name="Transform.__init__"></a><div id="Transform.__init__-def"><a name="L83"></a><tt class="py-lineno"> 83</tt> <a class="py-toggle" href="#" id="Transform.__init__-toggle" onclick="return toggle('Transform.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.Transform-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">source</tt><tt class="py-op">,</tt> <tt class="py-param">target</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt><tt class="py-param">description</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="Transform.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Transform.__init__-expanded"><a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name" targets="Variable corebio.transform.GeneticCode.table=corebio.transform.GeneticCode-class.html#table,Variable corebio.transform.Transform.table=corebio.transform.Transform-class.html#table"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-18', 'table', 'link-18');">table</a></tt> <tt class="py-op">=</tt> <tt class="py-name">maketrans</tt><tt class="py-op">(</tt><tt id="link-19" class="py-name" targets="Variable corebio.transform.Transform.source=corebio.transform.Transform-class.html#source"><a title="corebio.transform.Transform.source" class="py-name" href="#" onclick="return doclink('link-19', 'source', 'link-19');">source</a></tt><tt class="py-op">,</tt> <tt id="link-20" class="py-name" targets="Variable corebio.transform.Transform.target=corebio.transform.Transform-class.html#target"><a title="corebio.transform.Transform.target" class="py-name" href="#" onclick="return doclink('link-20', 'target', 'link-20');">target</a></tt><tt class="py-op">)</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="corebio.transform.Transform.source" class="py-name" href="#" onclick="return doclink('link-21', 'source', 'link-19');">source</a></tt> <tt class="py-op">=</tt> <tt id="link-22" class="py-name"><a title="corebio.transform.Transform.source" class="py-name" href="#" onclick="return doclink('link-22', 'source', 'link-19');">source</a></tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="corebio.transform.Transform.target" class="py-name" href="#" onclick="return doclink('link-23', 'target', 'link-20');">target</a></tt> <tt class="py-op">=</tt> <tt id="link-24" class="py-name"><a title="corebio.transform.Transform.target" class="py-name" href="#" onclick="return doclink('link-24', 'target', 'link-20');">target</a></tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Variable corebio.db.DataSource.name=corebio.db.DataSource-class.html#name,Variable corebio.matrix.SubMatrix.name=corebio.matrix.SubMatrix-class.html#name,Variable corebio.seq.SeqList.name=corebio.seq.SeqList-class.html#name,Variable corebio.transform.Transform.name=corebio.transform.Transform-class.html#name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-25', 'name', 'link-25');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-26" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-26', 'name', 'link-25');">name</a></tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name" targets="Variable corebio._version.description=corebio._version-module.html#description,Variable corebio.db.DataSource.description=corebio.db.DataSource-class.html#description,Variable corebio.matrix.SubMatrix.description=corebio.matrix.SubMatrix-class.html#description,Variable corebio.seq.SeqList.description=corebio.seq.SeqList-class.html#description,Variable corebio.transform.Transform.description=corebio.transform.Transform-class.html#description,Variable logoddslogolib.description=logoddslogolib-module.html#description,Variable weblogolib.description=weblogolib-module.html#description"><a title="corebio._version.description
corebio.db.DataSource.description
corebio.matrix.SubMatrix.description
corebio.seq.SeqList.description
corebio.transform.Transform.description
logoddslogolib.description
weblogolib.description" class="py-name" href="#" onclick="return doclink('link-27', 'description', 'link-27');">description</a></tt> <tt class="py-op">=</tt> <tt id="link-28" class="py-name"><a title="corebio._version.description
corebio.db.DataSource.description
corebio.matrix.SubMatrix.description
corebio.seq.SeqList.description
corebio.transform.Transform.description
logoddslogolib.description
weblogolib.description" class="py-name" href="#" onclick="return doclink('link-28', 'description', 'link-27');">description</a></tt> </tt>
</div><a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">      </tt>
<a name="Transform.__call__"></a><div id="Transform.__call__-def"><a name="L91"></a><tt class="py-lineno"> 91</tt> <a class="py-toggle" href="#" id="Transform.__call__-toggle" onclick="return toggle('Transform.__call__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.Transform-class.html#__call__">__call__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seq</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="Transform.__call__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Transform.__call__-expanded"><a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">        <tt class="py-docstring">"""Translate sequence."""</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="corebio.transform.Transform.source" class="py-name" href="#" onclick="return doclink('link-29', 'source', 'link-19');">source</a></tt><tt class="py-op">.</tt><tt id="link-30" class="py-name" targets="Variable corebio.matrix.Motif.alphabet=corebio.matrix.Motif-class.html#alphabet,Variable corebio.matrix.SubMatrix.alphabet=corebio.matrix.SubMatrix-class.html#alphabet,Variable corebio.seq.Seq.alphabet=corebio.seq.Seq-class.html#alphabet,Variable corebio.seq.SeqList.alphabet=corebio.seq.SeqList-class.html#alphabet"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-30', 'alphabet', 'link-30');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-31" class="py-name" targets="Method corebio.seq.Alphabet.alphabetic()=corebio.seq.Alphabet-class.html#alphabetic"><a title="corebio.seq.Alphabet.alphabetic" class="py-name" href="#" onclick="return doclink('link-31', 'alphabetic', 'link-31');">alphabetic</a></tt><tt class="py-op">(</tt><tt id="link-32" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-32', 'seq', 'link-5');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets"</tt><tt class="py-op">)</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">        <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name" targets="Method corebio.seq.Seq.translate()=corebio.seq.Seq-class.html#translate,Method corebio.transform.GeneticCode.translate()=corebio.transform.GeneticCode-class.html#translate"><a title="corebio.seq.Seq.translate
corebio.transform.GeneticCode.translate" class="py-name" href="#" onclick="return doclink('link-33', 'translate', 'link-33');">translate</a></tt><tt class="py-op">(</tt><tt id="link-34" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-34', 'seq', 'link-5');">seq</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-35', 'table', 'link-18');">table</a></tt><tt class="py-op">)</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">        <tt class="py-name">cls</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name"><a title="corebio.transform.Transform.target" class="py-name" href="#" onclick="return doclink('link-36', 'target', 'link-20');">target</a></tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">cls</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="corebio.transform.Transform.target" class="py-name" href="#" onclick="return doclink('link-37', 'target', 'link-20');">target</a></tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-38', 'alphabet', 'link-30');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-39" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-39', 'seq', 'link-5');">seq</a></tt><tt class="py-op">.</tt><tt id="link-40" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-40', 'name', 'link-25');">name</a></tt><tt class="py-op">,</tt> <tt id="link-41" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-41', 'seq', 'link-5');">seq</a></tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="corebio._version.description
corebio.db.DataSource.description
corebio.matrix.SubMatrix.description
corebio.seq.SeqList.description
corebio.transform.Transform.description
logoddslogolib.description
weblogolib.description" class="py-name" href="#" onclick="return doclink('link-42', 'description', 'link-27');">description</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-comment"># End class Translation</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-comment"># FIXME: Test, document, add to seq.</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt id="link-43" class="py-name" targets="Variable corebio.transform.dna_complement=corebio.transform-module.html#dna_complement"><a title="corebio.transform.dna_complement" class="py-name" href="#" onclick="return doclink('link-43', 'dna_complement', 'link-43');">dna_complement</a></tt> <tt class="py-op">=</tt> <tt id="link-44" class="py-name" targets="Class corebio.transform.Transform=corebio.transform.Transform-class.html"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-44', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">        <tt id="link-45" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-45', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"ACGTRYSWKMBDHVN-acgtUuryswkmbdhvnXx?.~"</tt><tt class="py-op">,</tt> <tt id="link-46" class="py-name"><a title="corebio.seq.dna_alphabet
corebio.seq_io.array_io.dna_alphabet
corebio.seq_io.clustal_io.dna_alphabet
corebio.seq_io.dna_alphabet
corebio.seq_io.fasta_io.dna_alphabet
corebio.seq_io.genbank_io.dna_alphabet
corebio.seq_io.intelligenetics_io.dna_alphabet
corebio.seq_io.msf_io.dna_alphabet
corebio.seq_io.nbrf_io.dna_alphabet
corebio.seq_io.phylip_io.dna_alphabet
corebio.seq_io.plain_io.dna_alphabet
corebio.seq_io.stockholm_io.dna_alphabet
corebio.seq_io.table_io.dna_alphabet" class="py-name" href="#" onclick="return doclink('link-46', 'dna_alphabet', 'link-9');">dna_alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">        <tt id="link-47" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-47', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"TGCAYRSWMKVHDBN-tgcaAayrswmkvhdbnXx?.~"</tt><tt class="py-op">,</tt> <tt id="link-48" class="py-name"><a title="corebio.seq.dna_alphabet
corebio.seq_io.array_io.dna_alphabet
corebio.seq_io.clustal_io.dna_alphabet
corebio.seq_io.dna_alphabet
corebio.seq_io.fasta_io.dna_alphabet
corebio.seq_io.genbank_io.dna_alphabet
corebio.seq_io.intelligenetics_io.dna_alphabet
corebio.seq_io.msf_io.dna_alphabet
corebio.seq_io.nbrf_io.dna_alphabet
corebio.seq_io.phylip_io.dna_alphabet
corebio.seq_io.plain_io.dna_alphabet
corebio.seq_io.stockholm_io.dna_alphabet
corebio.seq_io.table_io.dna_alphabet" class="py-name" href="#" onclick="return doclink('link-48', 'dna_alphabet', 'link-9');">dna_alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">    <tt class="py-op">)</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line"> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">      </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"> </tt>
<a name="mask_low_complexity"></a><div id="mask_low_complexity-def"><a name="L108"></a><tt class="py-lineno">108</tt> <a class="py-toggle" href="#" id="mask_low_complexity-toggle" onclick="return toggle('mask_low_complexity');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform-module.html#mask_low_complexity">mask_low_complexity</a><tt class="py-op">(</tt><tt class="py-param">seq</tt><tt class="py-op">,</tt> <tt class="py-param">width</tt> <tt class="py-op">=</tt><tt class="py-number">12</tt><tt class="py-op">,</tt> <tt class="py-param">trigger</tt><tt class="py-op">=</tt><tt class="py-number">1.8</tt><tt class="py-op">,</tt> <tt class="py-param">extension</tt><tt class="py-op">=</tt><tt class="py-number">2.0</tt><tt class="py-op">,</tt> <tt class="py-param">mask</tt><tt class="py-op">=</tt><tt class="py-string">'X'</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="mask_low_complexity-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="mask_low_complexity-expanded"><a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">    <tt class="py-docstring">""" Mask low complexity regions in protein sequences.</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"><tt class="py-docstring">    Uses the method of Seg [1] by Wootton &amp; Federhen [2] to divide a sequence   </tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line"><tt class="py-docstring">    into regions of high and low complexity. The sequence is divided into</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"><tt class="py-docstring">    overlapping windows. Low complexity windows either have a sequence entropy</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line"><tt class="py-docstring">    less than the trigger complexity, or have an entropy less than the extension    </tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line"><tt class="py-docstring">    complexity and neighbor other low-complexity windows. The sequence within   </tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"><tt class="py-docstring">    a low complexity region is replaced with the mask character (default 'X'), </tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"><tt class="py-docstring">    and the masked alphabetic sequence is returned.</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"><tt class="py-docstring">    The default parameters, width=12, trigger=1.8, extension=2.0, mask='X' are</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line"><tt class="py-docstring">    suitable for masking protein sequences before a database search. The </tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"><tt class="py-docstring">    standard default seg parameters are width=12, trigger=2.2, extension=2.5</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line"><tt class="py-docstring">        Seq seq         -- An alphabetic sequence</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"><tt class="py-docstring">        int width       -- Window width</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring">        float trigger   -- Entropy in bits between 0 and 4.3.. ( =log_2(20) )</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">        float extension -- Entropy in bits between 0 and 4.3.. ( =log_2(20) )</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring">        char mask       -- The mask character (default: 'X') </tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">    Returns :</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring">        Seq         -- A masked alphabetic sequence</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-docstring">    Raises :</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-docstring">        ValueError  -- On invalid arguments</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-docstring">    Refs:</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"><tt class="py-docstring">        [1] seg man page: </tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"><tt class="py-docstring">            http://bioportal.weizmann.ac.il/education/materials/gcg/seg.html</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line"><tt class="py-docstring">        [2] Wootton &amp; Federhen (Computers and Chemistry 17; 149-163, (1993)) </tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"><tt class="py-docstring">    Authors:</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"><tt class="py-docstring">        GEC 2005</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt class="py-docstring">    Future :</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-docstring">        - Optional mask character.</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-docstring">        - Option to lower case masked symbols.</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-docstring">        - Remove arbitary restriction to protein.</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">     </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">    <tt class="py-name">lg20</tt> <tt class="py-op">=</tt> <tt id="link-49" class="py-name"><a title="corebio.moremath.log2" class="py-name" href="#" onclick="return doclink('link-49', 'log2', 'link-16');">log2</a></tt><tt class="py-op">(</tt><tt class="py-number">20</tt><tt class="py-op">)</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">trigger</tt><tt class="py-op">&lt;</tt><tt class="py-number">0</tt> <tt class="py-keyword">or</tt> <tt class="py-name">trigger</tt><tt class="py-op">&gt;</tt><tt class="py-name">lg20</tt> <tt class="py-op">:</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid trigger complexity: %f"</tt><tt class="py-op">%</tt> <tt class="py-name">trigger</tt><tt class="py-op">)</tt>  </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-50" class="py-name" targets="Variable weblogolib._cgi.extension=weblogolib._cgi-module.html#extension"><a title="weblogolib._cgi.extension" class="py-name" href="#" onclick="return doclink('link-50', 'extension', 'link-50');">extension</a></tt><tt class="py-op">&lt;</tt><tt class="py-number">0</tt> <tt class="py-keyword">or</tt> <tt id="link-51" class="py-name"><a title="weblogolib._cgi.extension" class="py-name" href="#" onclick="return doclink('link-51', 'extension', 'link-50');">extension</a></tt><tt class="py-op">&gt;</tt><tt class="py-name">lg20</tt> <tt class="py-keyword">or</tt> <tt id="link-52" class="py-name"><a title="weblogolib._cgi.extension" class="py-name" href="#" onclick="return doclink('link-52', 'extension', 'link-50');">extension</a></tt><tt class="py-op">&lt;</tt><tt class="py-name">trigger</tt><tt class="py-op">:</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid extension complexity: %f"</tt><tt class="py-op">%</tt> <tt id="link-53" class="py-name"><a title="weblogolib._cgi.extension" class="py-name" href="#" onclick="return doclink('link-53', 'extension', 'link-50');">extension</a></tt><tt class="py-op">)</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">width</tt><tt class="py-op">&lt;</tt><tt class="py-number">0</tt> <tt class="py-op">:</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid width: %d"</tt><tt class="py-op">%</tt> <tt class="py-name">width</tt><tt class="py-op">)</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line"> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">width</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-54" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-54', 'seq', 'link-5');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> <tt class="py-keyword">return</tt> <tt id="link-55" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-55', 'seq', 'link-5');">seq</a></tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">     </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">    <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt id="link-56" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-56', 'seq', 'link-5');">seq</a></tt><tt class="py-op">.</tt><tt id="link-57" class="py-name" targets="Method corebio.seq.Alphabet.ords()=corebio.seq.Alphabet-class.html#ords,Method corebio.seq.Seq.ords()=corebio.seq.Seq-class.html#ords,Method corebio.seq.SeqList.ords()=corebio.seq.SeqList-class.html#ords"><a title="corebio.seq.Alphabet.ords
corebio.seq.Seq.ords
corebio.seq.SeqList.ords" class="py-name" href="#" onclick="return doclink('link-57', 'ords', 'link-57');">ords</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">    <tt class="py-name">X</tt> <tt class="py-op">=</tt> <tt id="link-58" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-58', 'seq', 'link-5');">seq</a></tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-59', 'alphabet', 'link-30');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-60" class="py-name" targets="Method corebio.seq.Alphabet.ord()=corebio.seq.Alphabet-class.html#ord"><a title="corebio.seq.Alphabet.ord" class="py-name" href="#" onclick="return doclink('link-60', 'ord', 'link-60');">ord</a></tt><tt class="py-op">(</tt><tt id="link-61" class="py-name" targets="Method corebio.seq.Seq.mask()=corebio.seq.Seq-class.html#mask"><a title="corebio.seq.Seq.mask" class="py-name" href="#" onclick="return doclink('link-61', 'mask', 'link-61');">mask</a></tt><tt class="py-op">)</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line"> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">     </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">    <tt class="py-name">nwindows</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-62" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-62', 'seq', 'link-5');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">-</tt> <tt class="py-name">width</tt> <tt class="py-op">+</tt><tt class="py-number">1</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">    <tt class="py-name">ent</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> <tt class="py-number">0</tt> <tt class="py-keyword">for</tt> <tt class="py-name">x</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">nwindows</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">    <tt class="py-name">count</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> <tt class="py-number">0</tt> <tt class="py-keyword">for</tt> <tt class="py-name">x</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-63" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-63', 'seq', 'link-5');">seq</a></tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-64', 'alphabet', 'link-30');">alphabet</a></tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">     </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">:</tt><tt class="py-name">width</tt><tt class="py-op">]</tt> <tt class="py-op">:</tt> <tt class="py-name">count</tt><tt class="py-op">[</tt><tt class="py-name">c</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt><tt class="py-number">1</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">    <tt class="py-name">ent</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-65" class="py-name"><a title="corebio.moremath.entropy" class="py-name" href="#" onclick="return doclink('link-65', 'entropy', 'link-17');">entropy</a></tt><tt class="py-op">(</tt><tt class="py-name">count</tt><tt class="py-op">,</tt><tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">     </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">i</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">nwindows</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">        <tt class="py-name">count</tt><tt class="py-op">[</tt> <tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">]</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-name">count</tt><tt class="py-op">[</tt> <tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">+</tt><tt class="py-name">width</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">]</tt> <tt class="py-op">+=</tt><tt class="py-number">1</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">        <tt class="py-name">ent</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-66" class="py-name"><a title="corebio.moremath.entropy" class="py-name" href="#" onclick="return doclink('link-66', 'entropy', 'link-17');">entropy</a></tt><tt class="py-op">(</tt><tt class="py-name">count</tt><tt class="py-op">,</tt><tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">     </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">    <tt class="py-name">prev_segged</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt>  </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">i</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">nwindows</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">prev_segged</tt> <tt class="py-keyword">and</tt> <tt class="py-name">ent</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt><tt class="py-op">&lt;</tt> <tt id="link-67" class="py-name"><a title="weblogolib._cgi.extension" class="py-name" href="#" onclick="return doclink('link-67', 'extension', 'link-50');">extension</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt>  </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">            <tt class="py-name">ent</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt><tt class="py-op">&lt;</tt> <tt class="py-name">trigger</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">j</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">width</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> <tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">+</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">=</tt><tt class="py-name">X</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">            <tt class="py-name">prev_segged</tt><tt class="py-op">=</tt><tt class="py-name">True</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt> <tt class="py-op">:</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">            <tt class="py-name">prev_segged</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line"> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line"> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">    <tt class="py-comment"># Redo, only backwards</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">    <tt class="py-name">prev_segged</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt>  </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">i</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-name">nwindows</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">prev_segged</tt> <tt class="py-keyword">and</tt> <tt class="py-name">ent</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt><tt class="py-op">&lt;</tt> <tt id="link-68" class="py-name"><a title="weblogolib._cgi.extension" class="py-name" href="#" onclick="return doclink('link-68', 'extension', 'link-50');">extension</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt>  </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">            <tt class="py-name">ent</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt><tt class="py-op">&lt;</tt> <tt class="py-name">trigger</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">j</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">width</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> <tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">+</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">=</tt><tt class="py-name">X</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">            <tt class="py-name">prev_segged</tt><tt class="py-op">=</tt><tt class="py-name">True</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt> <tt class="py-op">:</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">            <tt class="py-name">prev_segged</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line"> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">    <tt class="py-name">segged</tt> <tt class="py-op">=</tt> <tt id="link-69" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-69', 'seq', 'link-5');">seq</a></tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-70', 'alphabet', 'link-30');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-71" class="py-name" targets="Method corebio.seq.Alphabet.chrs()=corebio.seq.Alphabet-class.html#chrs"><a title="corebio.seq.Alphabet.chrs" class="py-name" href="#" onclick="return doclink('link-71', 'chrs', 'link-71');">chrs</a></tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">    <tt class="py-name">segged</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-72', 'name', 'link-25');">name</a></tt> <tt class="py-op">=</tt><tt id="link-73" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-73', 'seq', 'link-5');">seq</a></tt><tt class="py-op">.</tt><tt id="link-74" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-74', 'name', 'link-25');">name</a></tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">    <tt class="py-name">segged</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="corebio._version.description
corebio.db.DataSource.description
corebio.matrix.SubMatrix.description
corebio.seq.SeqList.description
corebio.transform.Transform.description
logoddslogolib.description
weblogolib.description" class="py-name" href="#" onclick="return doclink('link-75', 'description', 'link-27');">description</a></tt> <tt class="py-op">=</tt> <tt id="link-76" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-76', 'seq', 'link-5');">seq</a></tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="corebio._version.description
corebio.db.DataSource.description
corebio.matrix.SubMatrix.description
corebio.seq.SeqList.description
corebio.transform.Transform.description
logoddslogolib.description
weblogolib.description" class="py-name" href="#" onclick="return doclink('link-77', 'description', 'link-27');">description</a></tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">segged</tt> </tt>
</div><a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">     </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">     </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">     </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">     </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line"><tt class="py-comment"># end mask_low_complexity()    </tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">      </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"> </tt>
<a name="GeneticCode"></a><div id="GeneticCode-def"><a name="L203"></a><tt class="py-lineno">203</tt> <a class="py-toggle" href="#" id="GeneticCode-toggle" onclick="return toggle('GeneticCode');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html">GeneticCode</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="GeneticCode-expanded"><a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">    <tt class="py-docstring">"""An encoding of amino acids by DNA triplets.</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"><tt class="py-docstring"> </tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line"><tt class="py-docstring">    Example : </tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"><tt class="py-docstring">    Genetic Code [1]: Standard   </tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"><tt class="py-docstring">          T         C         A         G      </tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line"><tt class="py-docstring">       +---------+---------+---------+---------+</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"><tt class="py-docstring">     T | TTT F   | TCT S   | TAT Y   | TGT C   | T</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line"><tt class="py-docstring">     T | TTC F   | TCC S   | TAC Y   | TGC C   | C</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line"><tt class="py-docstring">     T | TTA L   | TCA S   | TAA Stop| TGA Stop| A</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"><tt class="py-docstring">     T | TTG L(s)| TCG S   | TAG Stop| TGG W   | G</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line"><tt class="py-docstring">       +---------+---------+---------+---------+</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"><tt class="py-docstring">     C | CTT L   | CCT P   | CAT H   | CGT R   | T</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line"><tt class="py-docstring">     C | CTC L   | CCC P   | CAC H   | CGC R   | C</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"><tt class="py-docstring">     C | CTA L   | CCA P   | CAA Q   | CGA R   | A</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line"><tt class="py-docstring">     C | CTG L(s)| CCG P   | CAG Q   | CGG R   | G</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line"><tt class="py-docstring">       +---------+---------+---------+---------+</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"><tt class="py-docstring">     A | ATT I   | ACT T   | AAT N   | AGT S   | T</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring">     A | ATC I   | ACC T   | AAC N   | AGC S   | C</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">     A | ATA I   | ACA T   | AAA K   | AGA R   | A</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-docstring">     A | ATG M(s)| ACG T   | AAG K   | AGG R   | G</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"><tt class="py-docstring">       +---------+---------+---------+---------+</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line"><tt class="py-docstring">     G | GTT V   | GCT A   | GAT D   | GGT G   | T</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line"><tt class="py-docstring">     G | GTC V   | GCC A   | GAC D   | GGC G   | C</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line"><tt class="py-docstring">     G | GTA V   | GCA A   | GAA E   | GGA G   | A</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line"><tt class="py-docstring">     G | GTG V   | GCG A   | GAG E   | GGG G   | G</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line"><tt class="py-docstring">       +---------+---------+---------+---------+</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line"><tt class="py-docstring">    See Also :</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line"><tt class="py-docstring">    -- http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"><tt class="py-docstring">    -- http://www.ncbi.nlm.nih.gov/projects/collab/FT/index.html#7.5</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"><tt class="py-docstring">    Authors:</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line"><tt class="py-docstring">        JXG, GEC</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">    <tt class="py-comment"># TODO: Explain use of '?' in translated sequence.</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">    <tt class="py-comment"># TODO: Does translate fails with aproriate execption when fed gaps?</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">    <tt class="py-comment"># TODO: Can back_translate handle gaps?</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">     </tt>
<a name="GeneticCode.__init__"></a><div id="GeneticCode.__init__-def"><a name="L243"></a><tt class="py-lineno">243</tt> <a class="py-toggle" href="#" id="GeneticCode.__init__-toggle" onclick="return toggle('GeneticCode.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">ident</tt><tt class="py-op">,</tt> <tt class="py-param">description</tt><tt class="py-op">,</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">        <tt class="py-param">amino_acid</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">base1</tt><tt class="py-op">,</tt> <tt class="py-param">base2</tt><tt class="py-op">,</tt> <tt class="py-param">base3</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticCode.__init__-expanded"><a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a new GeneticCode.</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line"><tt class="py-docstring">        Args:</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line"><tt class="py-docstring">        -- ident - Standard identifier (or zero). An integer</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line"><tt class="py-docstring">        -- description </tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line"><tt class="py-docstring">        -- amino acid - A sequence of amino acids and stop codons. e.g.</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line"><tt class="py-docstring">            "FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line"><tt class="py-docstring">        -- start - A sequence indicating start codons, e.g.,</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line"><tt class="py-docstring">            "---M---------------M---------------M----------------------------"</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line"><tt class="py-docstring">        -- base1 - The first base of each codon. e.g., </tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line"><tt class="py-docstring">            "TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line"><tt class="py-docstring">        -- base2 - The second base of each codon. e.g.,</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line"><tt class="py-docstring">            "TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line"><tt class="py-docstring">        -- base3 - The last base of each codon. e.g., </tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line"><tt class="py-docstring">            "TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"            </tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ident</tt> <tt class="py-op">=</tt> <tt class="py-name">ident</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="corebio._version.description
corebio.db.DataSource.description
corebio.matrix.SubMatrix.description
corebio.seq.SeqList.description
corebio.transform.Transform.description
logoddslogolib.description
weblogolib.description" class="py-name" href="#" onclick="return doclink('link-78', 'description', 'link-27');">description</a></tt> <tt class="py-op">=</tt> <tt id="link-79" class="py-name"><a title="corebio._version.description
corebio.db.DataSource.description
corebio.matrix.SubMatrix.description
corebio.seq.SeqList.description
corebio.transform.Transform.description
logoddslogolib.description
weblogolib.description" class="py-name" href="#" onclick="return doclink('link-79', 'description', 'link-27');">description</a></tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">    </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">amino_acid</tt> <tt class="py-op">=</tt> <tt class="py-name">amino_acid</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">start</tt> <tt class="py-op">=</tt> <tt class="py-name">start</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base1</tt> <tt class="py-op">=</tt> <tt class="py-name">base1</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base2</tt> <tt class="py-op">=</tt> <tt class="py-name">base2</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base3</tt> <tt class="py-op">=</tt> <tt class="py-name">base3</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">     </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">        <tt class="py-name">stop_codons</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt class="py-name">start_codons</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">i</tt><tt class="py-op">,</tt> <tt class="py-name">a</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">amino_acid</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">            <tt class="py-name">codon</tt> <tt class="py-op">=</tt> <tt class="py-name">base1</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">base2</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">base3</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">a</tt><tt class="py-op">==</tt><tt class="py-string">'*'</tt> <tt class="py-op">:</tt>  <tt class="py-name">stop_codons</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name" targets="Method corebio.db.astral.RafSeqMap.append()=corebio.db.astral.RafSeqMap-class.html#append"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-80', 'append', 'link-80');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">codon</tt><tt class="py-op">)</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">start</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'M'</tt><tt class="py-op">:</tt> <tt class="py-name">start_codons</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-81', 'append', 'link-80');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">codon</tt><tt class="py-op">)</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">             </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">stop_codons</tt> <tt class="py-op">=</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">stop_codons</tt><tt class="py-op">)</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">start_codons</tt> <tt class="py-op">=</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">start_codons</tt><tt class="py-op">)</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">         </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">        <tt class="py-comment"># Building the full translation table is expensive,</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">        <tt class="py-comment"># so we avoid doing so until necessary.</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_table</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_back_table</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line"> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">    <tt class="py-comment">#@staticmethod</tt> </tt>
<a name="GeneticCode.std_list"></a><div id="GeneticCode.std_list-def"><a name="L286"></a><tt class="py-lineno">286</tt> <a class="py-toggle" href="#" id="GeneticCode.std_list-toggle" onclick="return toggle('GeneticCode.std_list');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html#std_list">std_list</a><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode.std_list-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticCode.std_list-expanded"><a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt class="py-docstring">"Return a list of standard genetic codes."</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-82" class="py-name" targets="Variable corebio.transform._codon_tables=corebio.transform-module.html#_codon_tables"><a title="corebio.transform._codon_tables" class="py-name" href="#" onclick="return doclink('link-82', '_codon_tables', 'link-82');">_codon_tables</a></tt> </tt>
</div><a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">    <tt id="link-83" class="py-name" targets="Static Method corebio.transform.GeneticCode.std_list()=corebio.transform.GeneticCode-class.html#std_list"><a title="corebio.transform.GeneticCode.std_list" class="py-name" href="#" onclick="return doclink('link-83', 'std_list', 'link-83');">std_list</a></tt> <tt class="py-op">=</tt> <tt class="py-name">staticmethod</tt><tt class="py-op">(</tt><tt id="link-84" class="py-name"><a title="corebio.transform.GeneticCode.std_list" class="py-name" href="#" onclick="return doclink('link-84', 'std_list', 'link-83');">std_list</a></tt><tt class="py-op">)</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line"> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">    <tt class="py-comment">#@staticmethod</tt> </tt>
<a name="GeneticCode.std"></a><div id="GeneticCode.std-def"><a name="L292"></a><tt class="py-lineno">292</tt> <a class="py-toggle" href="#" id="GeneticCode.std-toggle" onclick="return toggle('GeneticCode.std');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html#std">std</a><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode.std-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticCode.std-expanded"><a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">        <tt class="py-docstring">"The standard 'universal' genetic code."</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-85" class="py-name"><a title="corebio.transform._codon_tables" class="py-name" href="#" onclick="return doclink('link-85', '_codon_tables', 'link-82');">_codon_tables</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div><a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">    <tt id="link-86" class="py-name" targets="Static Method corebio.transform.GeneticCode.std()=corebio.transform.GeneticCode-class.html#std"><a title="corebio.transform.GeneticCode.std" class="py-name" href="#" onclick="return doclink('link-86', 'std', 'link-86');">std</a></tt> <tt class="py-op">=</tt> <tt class="py-name">staticmethod</tt><tt class="py-op">(</tt><tt id="link-87" class="py-name"><a title="corebio.transform.GeneticCode.std" class="py-name" href="#" onclick="return doclink('link-87', 'std', 'link-86');">std</a></tt><tt class="py-op">)</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line"> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">     </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">    <tt class="py-comment">#@staticmethod</tt> </tt>
<a name="GeneticCode.by_name"></a><div id="GeneticCode.by_name-def"><a name="L299"></a><tt class="py-lineno">299</tt> <a class="py-toggle" href="#" id="GeneticCode.by_name-toggle" onclick="return toggle('GeneticCode.by_name');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html#by_name">by_name</a><tt class="py-op">(</tt><tt class="py-param">name</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode.by_name-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticCode.by_name-expanded"><a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">        <tt class="py-docstring">"""Find a genetic code in the code list by name or identifier.</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">t</tt> <tt class="py-keyword">in</tt> <tt id="link-88" class="py-name"><a title="corebio.transform._codon_tables" class="py-name" href="#" onclick="return doclink('link-88', '_codon_tables', 'link-82');">_codon_tables</a></tt> <tt class="py-op">:</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">t</tt><tt class="py-op">.</tt><tt class="py-name">ident</tt> <tt class="py-op">==</tt> <tt id="link-89" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-89', 'name', 'link-25');">name</a></tt> <tt class="py-keyword">or</tt> <tt class="py-name">t</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="corebio._version.description
corebio.db.DataSource.description
corebio.matrix.SubMatrix.description
corebio.seq.SeqList.description
corebio.transform.Transform.description
logoddslogolib.description
weblogolib.description" class="py-name" href="#" onclick="return doclink('link-90', 'description', 'link-27');">description</a></tt> <tt class="py-op">==</tt> <tt id="link-91" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-91', 'name', 'link-25');">name</a></tt> <tt class="py-op">:</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">t</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No such translation table: %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-92" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-92', 'name', 'link-25');">name</a></tt><tt class="py-op">)</tt> <tt class="py-op">)</tt> </tt>
</div><a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">    <tt id="link-93" class="py-name" targets="Static Method corebio.transform.GeneticCode.by_name()=corebio.transform.GeneticCode-class.html#by_name,Static Method weblogolib.color.Color.by_name()=weblogolib.color.Color-class.html#by_name"><a title="corebio.transform.GeneticCode.by_name
weblogolib.color.Color.by_name" class="py-name" href="#" onclick="return doclink('link-93', 'by_name', 'link-93');">by_name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">staticmethod</tt><tt class="py-op">(</tt><tt id="link-94" class="py-name"><a title="corebio.transform.GeneticCode.by_name
weblogolib.color.Color.by_name" class="py-name" href="#" onclick="return doclink('link-94', 'by_name', 'link-93');">by_name</a></tt><tt class="py-op">)</tt>         </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">     </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">   </tt>
<a name="GeneticCode._get_table"></a><div id="GeneticCode._get_table-def"><a name="L309"></a><tt class="py-lineno">309</tt> <a class="py-toggle" href="#" id="GeneticCode._get_table-toggle" onclick="return toggle('GeneticCode._get_table');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html#_get_table">_get_table</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode._get_table-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticCode._get_table-expanded"><a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_table</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_create_table</tt><tt class="py-op">(</tt><tt class="py-op">)</tt>  </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_table</tt>   </tt>
</div><a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">    <tt id="link-95" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-95', 'table', 'link-18');">table</a></tt> <tt class="py-op">=</tt> <tt class="py-name">property</tt><tt class="py-op">(</tt><tt id="link-96" class="py-name" targets="Method corebio.transform.GeneticCode._get_table()=corebio.transform.GeneticCode-class.html#_get_table"><a title="corebio.transform.GeneticCode._get_table" class="py-name" href="#" onclick="return doclink('link-96', '_get_table', 'link-96');">_get_table</a></tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-string">"A map between codons and amino acids"</tt><tt class="py-op">)</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line"> </tt>
<a name="GeneticCode._get_back_table"></a><div id="GeneticCode._get_back_table-def"><a name="L314"></a><tt class="py-lineno">314</tt> <a class="py-toggle" href="#" id="GeneticCode._get_back_table-toggle" onclick="return toggle('GeneticCode._get_back_table');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html#_get_back_table">_get_back_table</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode._get_back_table-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticCode._get_back_table-expanded"><a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_back_table</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-op">:</tt>  </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_create_table</tt><tt class="py-op">(</tt><tt class="py-op">)</tt>  </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_back_table</tt>   </tt>
</div><a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">    <tt id="link-97" class="py-name" targets="Variable corebio.transform.GeneticCode.back_table=corebio.transform.GeneticCode-class.html#back_table"><a title="corebio.transform.GeneticCode.back_table" class="py-name" href="#" onclick="return doclink('link-97', 'back_table', 'link-97');">back_table</a></tt> <tt class="py-op">=</tt> <tt class="py-name">property</tt><tt class="py-op">(</tt><tt id="link-98" class="py-name" targets="Method corebio.transform.GeneticCode._get_back_table()=corebio.transform.GeneticCode-class.html#_get_back_table"><a title="corebio.transform.GeneticCode._get_back_table" class="py-name" href="#" onclick="return doclink('link-98', '_get_back_table', 'link-98');">_get_back_table</a></tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-string">"A map between amino acids and codons"</tt><tt class="py-op">)</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line"> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line"> </tt>
<a name="GeneticCode._create_table"></a><div id="GeneticCode._create_table-def"><a name="L321"></a><tt class="py-lineno">321</tt> <a class="py-toggle" href="#" id="GeneticCode._create_table-toggle" onclick="return toggle('GeneticCode._create_table');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html#_create_table">_create_table</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode._create_table-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticCode._create_table-expanded"><a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt id="link-99" class="py-name" targets="Variable corebio.secstruc.dssp.DsspResidue.aa=corebio.secstruc.dssp.DsspResidue-class.html#aa,Variable corebio.secstruc.stride.StrideResidue.aa=corebio.secstruc.stride.StrideResidue-class.html#aa"><a title="corebio.secstruc.dssp.DsspResidue.aa
corebio.secstruc.stride.StrideResidue.aa" class="py-name" href="#" onclick="return doclink('link-99', 'aa', 'link-99');">aa</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">amino_acid</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt class="py-name">base1</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base1</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt class="py-name">base2</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base2</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">        <tt class="py-name">base3</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base3</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">         </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">        <tt class="py-comment"># Construct a table of unambiguous codon translations</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt id="link-100" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-100', 'table', 'link-18');">table</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">i</tt><tt class="py-op">,</tt> <tt class="py-name">a</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-101" class="py-name"><a title="corebio.secstruc.dssp.DsspResidue.aa
corebio.secstruc.stride.StrideResidue.aa" class="py-name" href="#" onclick="return doclink('link-101', 'aa', 'link-99');">aa</a></tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">            <tt class="py-name">codon</tt> <tt class="py-op">=</tt> <tt class="py-name">base1</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">base2</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">base3</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">            <tt id="link-102" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-102', 'table', 'link-18');">table</a></tt><tt class="py-op">[</tt><tt class="py-name">codon</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">a</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">         </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">        <tt class="py-comment"># Build the back table.</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">        <tt id="link-103" class="py-name"><a title="corebio.transform.GeneticCode.back_table" class="py-name" href="#" onclick="return doclink('link-103', 'back_table', 'link-97');">back_table</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">        <tt class="py-name">items</tt> <tt class="py-op">=</tt> <tt id="link-104" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-104', 'table', 'link-18');">table</a></tt><tt class="py-op">.</tt><tt class="py-name">items</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">        <tt class="py-name">items</tt><tt class="py-op">.</tt><tt class="py-name">sort</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">codon</tt><tt class="py-op">,</tt> <tt id="link-105" class="py-name"><a title="corebio.secstruc.dssp.DsspResidue.aa
corebio.secstruc.stride.StrideResidue.aa" class="py-name" href="#" onclick="return doclink('link-105', 'aa', 'link-99');">aa</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">items</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">:</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">            <tt id="link-106" class="py-name"><a title="corebio.transform.GeneticCode.back_table" class="py-name" href="#" onclick="return doclink('link-106', 'back_table', 'link-97');">back_table</a></tt><tt class="py-op">[</tt><tt id="link-107" class="py-name"><a title="corebio.secstruc.dssp.DsspResidue.aa
corebio.secstruc.stride.StrideResidue.aa" class="py-name" href="#" onclick="return doclink('link-107', 'aa', 'link-99');">aa</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">codon</tt>   <tt class="py-comment"># Use first codon, alphabetically.</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">        <tt id="link-108" class="py-name"><a title="corebio.transform.GeneticCode.back_table" class="py-name" href="#" onclick="return doclink('link-108', 'back_table', 'link-97');">back_table</a></tt><tt class="py-op">[</tt><tt class="py-string">'X'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'NNN'</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">        <tt id="link-109" class="py-name"><a title="corebio.transform.GeneticCode.back_table" class="py-name" href="#" onclick="return doclink('link-109', 'back_table', 'link-97');">back_table</a></tt><tt class="py-op">[</tt><tt class="py-string">'B'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'NNN'</tt>          </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">        <tt id="link-110" class="py-name"><a title="corebio.transform.GeneticCode.back_table" class="py-name" href="#" onclick="return doclink('link-110', 'back_table', 'link-97');">back_table</a></tt><tt class="py-op">[</tt><tt class="py-string">'Z'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'NNN'</tt>          </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">        <tt id="link-111" class="py-name"><a title="corebio.transform.GeneticCode.back_table" class="py-name" href="#" onclick="return doclink('link-111', 'back_table', 'link-97');">back_table</a></tt><tt class="py-op">[</tt><tt class="py-string">'J'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'NNN'</tt>          </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_back_table</tt> <tt class="py-op">=</tt> <tt id="link-112" class="py-name"><a title="corebio.transform.GeneticCode.back_table" class="py-name" href="#" onclick="return doclink('link-112', 'back_table', 'link-97');">back_table</a></tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">                 </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">        <tt class="py-name">ltable</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">        <tt id="link-113" class="py-name" targets="Method corebio.seq.Alphabet.letters()=corebio.seq.Alphabet-class.html#letters"><a title="corebio.seq.Alphabet.letters" class="py-name" href="#" onclick="return doclink('link-113', 'letters', 'link-113');">letters</a></tt> <tt class="py-op">=</tt> <tt id="link-114" class="py-name"><a title="corebio.data.dna_extended_letters" class="py-name" href="#" onclick="return doclink('link-114', 'dna_extended_letters', 'link-2');">dna_extended_letters</a></tt><tt class="py-op">+</tt><tt class="py-string">'U'</tt> <tt class="py-comment"># include RNA in table</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line"> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-comment"># Create a list of all possble codons</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">        <tt class="py-name">codons</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">c1</tt> <tt class="py-keyword">in</tt> <tt id="link-115" class="py-name"><a title="corebio.seq.Alphabet.letters" class="py-name" href="#" onclick="return doclink('link-115', 'letters', 'link-113');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">c2</tt> <tt class="py-keyword">in</tt> <tt id="link-116" class="py-name"><a title="corebio.seq.Alphabet.letters" class="py-name" href="#" onclick="return doclink('link-116', 'letters', 'link-113');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">c3</tt> <tt class="py-keyword">in</tt> <tt id="link-117" class="py-name"><a title="corebio.seq.Alphabet.letters" class="py-name" href="#" onclick="return doclink('link-117', 'letters', 'link-113');">letters</a></tt> <tt class="py-op">:</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">                    <tt class="py-name">codons</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-118', 'append', 'link-80');">append</a></tt><tt class="py-op">(</tt> <tt class="py-name">c1</tt><tt class="py-op">+</tt><tt class="py-name">c2</tt><tt class="py-op">+</tt><tt class="py-name">c3</tt><tt class="py-op">)</tt>         </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line"> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-comment"># For each ambiguous codon, construct all compatible unambiguous codons.</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">        <tt class="py-comment"># Translate and collect a set of all possible translated amino acids.</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">        <tt class="py-comment"># If more than one translation look for possible amino acid ambiguity       </tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">        <tt class="py-comment"># codes. </tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-119" class="py-name" targets="Variable corebio.seq_io._nexus.C=corebio.seq_io._nexus-module.html#C"><a title="corebio.seq_io._nexus.C" class="py-name" href="#" onclick="return doclink('link-119', 'C', 'link-119');">C</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">codons</tt> <tt class="py-op">:</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">            <tt class="py-name">translated</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-comment"># Use dict, because no set in py2.3</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">            <tt class="py-name">c</tt> <tt class="py-op">=</tt> <tt id="link-120" class="py-name"><a title="corebio.seq_io._nexus.C" class="py-name" href="#" onclick="return doclink('link-120', 'C', 'link-119');">C</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'U'</tt><tt class="py-op">,</tt> <tt class="py-string">'T'</tt><tt class="py-op">)</tt> <tt class="py-comment"># Convert RNA codon to DNA</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">c1</tt> <tt class="py-keyword">in</tt> <tt id="link-121" class="py-name"><a title="corebio.data.dna_ambiguity" class="py-name" href="#" onclick="return doclink('link-121', 'dna_ambiguity', 'link-3');">dna_ambiguity</a></tt><tt class="py-op">[</tt><tt class="py-name">c</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">c2</tt> <tt class="py-keyword">in</tt> <tt id="link-122" class="py-name"><a title="corebio.data.dna_ambiguity" class="py-name" href="#" onclick="return doclink('link-122', 'dna_ambiguity', 'link-3');">dna_ambiguity</a></tt><tt class="py-op">[</tt><tt class="py-name">c</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">c3</tt> <tt class="py-keyword">in</tt> <tt id="link-123" class="py-name"><a title="corebio.data.dna_ambiguity" class="py-name" href="#" onclick="return doclink('link-123', 'dna_ambiguity', 'link-3');">dna_ambiguity</a></tt><tt class="py-op">[</tt><tt class="py-name">c</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">                        <tt id="link-124" class="py-name"><a title="corebio.secstruc.dssp.DsspResidue.aa
corebio.secstruc.stride.StrideResidue.aa" class="py-name" href="#" onclick="return doclink('link-124', 'aa', 'link-99');">aa</a></tt> <tt class="py-op">=</tt> <tt id="link-125" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-125', 'table', 'link-18');">table</a></tt><tt class="py-op">[</tt> <tt class="py-name">c1</tt><tt class="py-op">+</tt><tt class="py-name">c2</tt><tt class="py-op">+</tt><tt class="py-name">c3</tt> <tt class="py-op">]</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">                        <tt class="py-name">translated</tt><tt class="py-op">[</tt><tt id="link-126" class="py-name"><a title="corebio.secstruc.dssp.DsspResidue.aa
corebio.secstruc.stride.StrideResidue.aa" class="py-name" href="#" onclick="return doclink('link-126', 'aa', 'link-99');">aa</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">            <tt class="py-name">translated</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">translated</tt><tt class="py-op">.</tt><tt class="py-name">keys</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">            <tt class="py-name">translated</tt><tt class="py-op">.</tt><tt class="py-name">sort</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">translated</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt><tt class="py-number">1</tt> <tt class="py-op">:</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">                <tt class="py-name">trans</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">translated</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">translated</tt> <tt class="py-op">==</tt> <tt class="py-op">[</tt><tt class="py-string">'D'</tt><tt class="py-op">,</tt><tt class="py-string">'N'</tt><tt class="py-op">]</tt> <tt class="py-op">:</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">                <tt class="py-name">trans</tt> <tt class="py-op">=</tt> <tt class="py-string">'B'</tt>       </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">translated</tt> <tt class="py-op">==</tt> <tt class="py-op">[</tt><tt class="py-string">'E'</tt><tt class="py-op">,</tt><tt class="py-string">'Q'</tt><tt class="py-op">]</tt> <tt class="py-op">:</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">                <tt class="py-name">trans</tt> <tt class="py-op">=</tt> <tt class="py-string">'Z'</tt>  </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">translated</tt> <tt class="py-op">==</tt> <tt class="py-op">[</tt><tt class="py-string">'I'</tt><tt class="py-op">,</tt><tt class="py-string">'L'</tt><tt class="py-op">]</tt> <tt class="py-op">:</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">                <tt class="py-name">trans</tt> <tt class="py-op">=</tt> <tt class="py-string">'J'</tt>              </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-string">'*'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">translated</tt><tt class="py-op">:</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">                <tt class="py-name">trans</tt> <tt class="py-op">=</tt> <tt class="py-string">'?'</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt> <tt class="py-op">:</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">                <tt class="py-name">trans</tt> <tt class="py-op">=</tt> <tt class="py-string">'X'</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">            <tt class="py-name">ltable</tt><tt class="py-op">[</tt><tt id="link-127" class="py-name"><a title="corebio.seq_io._nexus.C" class="py-name" href="#" onclick="return doclink('link-127', 'C', 'link-119');">C</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">trans</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line"> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_table</tt> <tt class="py-op">=</tt> <tt class="py-name">ltable</tt> </tt>
</div><a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">    <tt class="py-comment"># End create tables</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line"> </tt>
<a name="GeneticCode.translate"></a><div id="GeneticCode.translate-def"><a name="L386"></a><tt class="py-lineno">386</tt> <a class="py-toggle" href="#" id="GeneticCode.translate-toggle" onclick="return toggle('GeneticCode.translate');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html#translate">translate</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seq</tt><tt class="py-op">,</tt> <tt class="py-param">frame</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode.translate-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticCode.translate-expanded"><a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">        <tt class="py-docstring">"""Translate a DNA sequence to a polypeptide using full</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line"><tt class="py-docstring">        IUPAC ambiguities in DNA/RNA and amino acid codes.</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line"><tt class="py-docstring">        Returns : </tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line"><tt class="py-docstring">        -- Seq - A polypeptide sequence </tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">        <tt class="py-comment"># TODO: Optimize.</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">        <tt class="py-comment"># TODO: Insanity check alphabet.</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">        <tt id="link-128" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-128', 'seq', 'link-5');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-129" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-129', 'seq', 'link-5');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">        <tt id="link-130" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-130', 'table', 'link-18');">table</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-131', 'table', 'link-18');">table</a></tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">        <tt class="py-name">trans</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">        <tt class="py-name">L</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-132" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-132', 'seq', 'link-5');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">i</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-name">frame</tt><tt class="py-op">,</tt> <tt class="py-name">L</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt>    </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">            <tt class="py-name">codon</tt> <tt class="py-op">=</tt> <tt id="link-133" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-133', 'seq', 'link-5');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">:</tt><tt class="py-name">i</tt><tt class="py-op">+</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-134" class="py-name" targets="Method corebio.seq.Seq.upper()=corebio.seq.Seq-class.html#upper"><a title="corebio.seq.Seq.upper" class="py-name" href="#" onclick="return doclink('link-134', 'upper', 'link-134');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">            <tt class="py-name">trans</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="corebio.db.astral.RafSeqMap.append" class="py-name" href="#" onclick="return doclink('link-135', 'append', 'link-80');">append</a></tt><tt class="py-op">(</tt> <tt id="link-136" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-136', 'table', 'link-18');">table</a></tt><tt class="py-op">[</tt><tt class="py-name">codon</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-137" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-137', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name" targets="Method corebio.seq.Seq.join()=corebio.seq.Seq-class.html#join"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-138', 'join', 'link-138');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">trans</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-139" class="py-name"><a title="corebio.seq.protein_alphabet
corebio.seq_io.array_io.protein_alphabet
corebio.seq_io.clustal_io.protein_alphabet
corebio.seq_io.fasta_io.protein_alphabet
corebio.seq_io.genbank_io.protein_alphabet
corebio.seq_io.intelligenetics_io.protein_alphabet
corebio.seq_io.msf_io.protein_alphabet
corebio.seq_io.nbrf_io.protein_alphabet
corebio.seq_io.phylip_io.protein_alphabet
corebio.seq_io.plain_io.protein_alphabet
corebio.seq_io.protein_alphabet
corebio.seq_io.stockholm_io.protein_alphabet
corebio.seq_io.table_io.protein_alphabet" class="py-name" href="#" onclick="return doclink('link-139', 'protein_alphabet', 'link-7');">protein_alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line"> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">              </tt>
<a name="GeneticCode.back_translate"></a><div id="GeneticCode.back_translate-def"><a name="L405"></a><tt class="py-lineno">405</tt> <a class="py-toggle" href="#" id="GeneticCode.back_translate-toggle" onclick="return toggle('GeneticCode.back_translate');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html#back_translate">back_translate</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seq</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode.back_translate-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticCode.back_translate-expanded"><a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">        <tt class="py-docstring">"""Convert protein back into coding DNA.</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line"><tt class="py-docstring">        Args:</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line"><tt class="py-docstring">        -- seq - A polypeptide sequence.</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line"><tt class="py-docstring">        Returns :</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line"><tt class="py-docstring">        -- Seq - A DNA sequence</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">        <tt class="py-comment"># TODO: Optimzie</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">        <tt class="py-comment"># TODO: Insanity check alphabet.</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">        <tt id="link-140" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-140', 'table', 'link-18');">table</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="corebio.transform.GeneticCode.back_table" class="py-name" href="#" onclick="return doclink('link-141', 'back_table', 'link-97');">back_table</a></tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">        <tt id="link-142" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-142', 'seq', 'link-5');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-143" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-143', 'seq', 'link-5');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">        <tt class="py-name">trans</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> <tt id="link-144" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-144', 'table', 'link-18');">table</a></tt><tt class="py-op">[</tt><tt class="py-name">a</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">a</tt> <tt class="py-keyword">in</tt> <tt id="link-145" class="py-name"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-145', 'seq', 'link-5');">seq</a></tt><tt class="py-op">]</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-146" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-146', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">.</tt><tt id="link-147" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-147', 'join', 'link-138');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">trans</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-148" class="py-name"><a title="corebio.seq.dna_alphabet
corebio.seq_io.array_io.dna_alphabet
corebio.seq_io.clustal_io.dna_alphabet
corebio.seq_io.dna_alphabet
corebio.seq_io.fasta_io.dna_alphabet
corebio.seq_io.genbank_io.dna_alphabet
corebio.seq_io.intelligenetics_io.dna_alphabet
corebio.seq_io.msf_io.dna_alphabet
corebio.seq_io.nbrf_io.dna_alphabet
corebio.seq_io.phylip_io.dna_alphabet
corebio.seq_io.plain_io.dna_alphabet
corebio.seq_io.stockholm_io.dna_alphabet
corebio.seq_io.table_io.dna_alphabet" class="py-name" href="#" onclick="return doclink('link-148', 'dna_alphabet', 'link-9');">dna_alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">         </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line"> <tt class="py-comment">#TODO: translate_orf(self, seq, start) ?</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line"> <tt class="py-comment">#TODO: translate_to_stop(self, seq, frame) ?       </tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line"> <tt class="py-comment">#TODO: translate_all_frames(self,seq) -&gt; 6 translations.</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">  </tt>
<a name="GeneticCode.__repr__"></a><div id="GeneticCode.__repr__-def"><a name="L425"></a><tt class="py-lineno">425</tt> <a class="py-toggle" href="#" id="GeneticCode.__repr__-toggle" onclick="return toggle('GeneticCode.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticCode.__repr__-expanded"><a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'GeneticCode( %d, "'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ident</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-149" class="py-name"><a title="corebio._version.description
corebio.db.DataSource.description
corebio.matrix.SubMatrix.description
corebio.seq.SeqList.description
corebio.transform.Transform.description
logoddslogolib.description
weblogolib.description" class="py-name" href="#" onclick="return doclink('link-149', 'description', 'link-27');">description</a></tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'", \n'</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'    amino_acid = "'</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">amino_acid</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'",\n'</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'    start =      "'</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">start</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'",\n'</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'    base1 =      "'</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base1</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'",\n'</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'    base2 =      "'</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base2</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'",\n'</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'    base3 =      "'</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base3</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'" )'</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-150', 'join', 'link-138');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">string</tt><tt class="py-op">)</tt> </tt>
</div><a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">         </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">         </tt>
<a name="GeneticCode.__str__"></a><div id="GeneticCode.__str__-def"><a name="L448"></a><tt class="py-lineno">448</tt> <a class="py-toggle" href="#" id="GeneticCode.__str__-toggle" onclick="return toggle('GeneticCode.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.transform.GeneticCode-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="GeneticCode.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticCode.__str__-expanded"><a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns a text representation of this genetic code."""</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">        <tt class="py-comment"># Inspired by http://bugzilla.open-bio.org/show_bug.cgi?id=1963</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">        <tt id="link-151" class="py-name"><a title="corebio.seq.Alphabet.letters" class="py-name" href="#" onclick="return doclink('link-151', 'letters', 'link-113');">letters</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"TCAG"</tt> <tt class="py-comment"># Conventional ordering for codon tables.</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">         </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ident</tt> <tt class="py-op">:</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">            <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'Genetic Code [%d]: '</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ident</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt> <tt class="py-op">:</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">            <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'Genetic Code: '</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt>  <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name"><a title="corebio._version.description
corebio.db.DataSource.description
corebio.matrix.SubMatrix.description
corebio.seq.SeqList.description
corebio.transform.Transform.description
logoddslogolib.description
weblogolib.description" class="py-name" href="#" onclick="return doclink('link-152', 'description', 'link-27');">description</a></tt> <tt class="py-keyword">or</tt> <tt class="py-string">''</tt>         </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line"> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n    "</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-153', 'join', 'link-138');">join</a></tt><tt class="py-op">(</tt> <tt class="py-op">[</tt><tt class="py-string">"  %s      "</tt> <tt class="py-op">%</tt> <tt class="py-name">c2</tt> <tt class="py-keyword">for</tt> <tt class="py-name">c2</tt> <tt class="py-keyword">in</tt> <tt id="link-154" class="py-name"><a title="corebio.seq.Alphabet.letters" class="py-name" href="#" onclick="return doclink('link-154', 'letters', 'link-113');">letters</a></tt><tt class="py-op">]</tt> <tt class="py-op">)</tt>  </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">         </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n   +"</tt>  </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt>  <tt class="py-string">"+"</tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-155', 'join', 'link-138');">join</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">"---------"</tt> <tt class="py-keyword">for</tt> <tt class="py-name">c2</tt> <tt class="py-keyword">in</tt> <tt id="link-156" class="py-name"><a title="corebio.seq.Alphabet.letters" class="py-name" href="#" onclick="return doclink('link-156', 'letters', 'link-113');">letters</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">"+  "</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">         </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">        <tt id="link-157" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-157', 'table', 'link-18');">table</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-158" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-158', 'table', 'link-18');">table</a></tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">         </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">c1</tt> <tt class="py-keyword">in</tt> <tt id="link-159" class="py-name"><a title="corebio.seq.Alphabet.letters" class="py-name" href="#" onclick="return doclink('link-159', 'letters', 'link-113');">letters</a></tt> <tt class="py-op">:</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">c3</tt> <tt class="py-keyword">in</tt> <tt id="link-160" class="py-name"><a title="corebio.seq.Alphabet.letters" class="py-name" href="#" onclick="return doclink('link-160', 'letters', 'link-113');">letters</a></tt> <tt class="py-op">:</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">                <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'\n '</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">                <tt class="py-name">string</tt>  <tt class="py-op">+=</tt> <tt class="py-name">c1</tt>  </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">                <tt class="py-name">string</tt>  <tt class="py-op">+=</tt> <tt class="py-string">" |"</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">c2</tt> <tt class="py-keyword">in</tt> <tt id="link-161" class="py-name"><a title="corebio.seq.Alphabet.letters" class="py-name" href="#" onclick="return doclink('link-161', 'letters', 'link-113');">letters</a></tt> <tt class="py-op">:</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">                    <tt class="py-name">codon</tt> <tt class="py-op">=</tt> <tt class="py-name">c1</tt><tt class="py-op">+</tt><tt class="py-name">c2</tt><tt class="py-op">+</tt><tt class="py-name">c3</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">                    <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt class="py-name">codon</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">codon</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">stop_codons</tt> <tt class="py-op">:</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">                        <tt class="py-name">string</tt>  <tt class="py-op">+=</tt> <tt class="py-string">" Stop|"</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt> <tt class="py-op">:</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">                        <tt class="py-name">amino</tt> <tt class="py-op">=</tt> <tt id="link-162" class="py-name"><a title="corebio.transform.GeneticCode.table
corebio.transform.Transform.table" class="py-name" href="#" onclick="return doclink('link-162', 'table', 'link-18');">table</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-name">codon</tt><tt class="py-op">,</tt> <tt class="py-string">'?'</tt><tt class="py-op">)</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">codon</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">start_codons</tt> <tt class="py-op">:</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">                            <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">" %s(s)|"</tt> <tt class="py-op">%</tt> <tt class="py-name">amino</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt> <tt class="py-op">:</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">                            <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">" %s   |"</tt> <tt class="py-op">%</tt> <tt class="py-name">amino</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">                <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt class="py-name">c3</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">                 </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">            <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n   +"</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">            <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">"+"</tt><tt class="py-op">.</tt><tt id="link-163" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-163', 'join', 'link-138');">join</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">"---------"</tt> <tt class="py-keyword">for</tt> <tt class="py-name">c2</tt> <tt class="py-keyword">in</tt> <tt id="link-164" class="py-name"><a title="corebio.seq.Alphabet.letters" class="py-name" href="#" onclick="return doclink('link-164', 'letters', 'link-113');">letters</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt>  </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">            <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">"+  "</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">        <tt class="py-name">string</tt> <tt class="py-op">+=</tt> <tt class="py-string">'\n'</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-165', 'join', 'link-138');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">string</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line"><tt class="py-comment"># end class GeneticCode</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line"> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">  </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line"><tt class="py-comment"># Data from http://www.ncbi.nlm.nih.gov/projects/collab/FT/index.html#7.5</tt> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line"><tt class="py-comment"># Aug. 2006</tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line"><tt class="py-comment"># Genetic Code Tables</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line"><tt class="py-comment"># </tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line"><tt class="py-comment"># Authority      International Sequence Databank Collaboration</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line"><tt class="py-comment"># Contact        NCBI</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line"><tt class="py-comment"># Scope          /transl_table qualifier</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line"><tt class="py-comment"># URL            http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c   </tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line"><tt id="link-166" class="py-name"><a title="corebio.transform._codon_tables" class="py-name" href="#" onclick="return doclink('link-166', '_codon_tables', 'link-82');">_codon_tables</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt>  </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">    <tt id="link-167" class="py-name" targets="Class corebio.transform.GeneticCode=corebio.transform.GeneticCode-class.html"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-167', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-string">"Standard"</tt><tt class="py-op">,</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">        <tt class="py-string">"---M---------------M---------------M----------------------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">         </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">    <tt id="link-168" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-168', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-string">"Vertebrate Mitochondrial"</tt><tt class="py-op">,</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSS**VVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line">        <tt class="py-string">"--------------------------------MMMM---------------M------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">  </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">    <tt id="link-169" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-169', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">,</tt> <tt class="py-string">"Yeast Mitochondrial"</tt><tt class="py-op">,</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY**CCWWTTTTPPPPHHQQRRRRIIMMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">        <tt class="py-string">"----------------------------------MM----------------------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line"> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line">    <tt id="link-170" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-170', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-string">"Mold, Protozoan, Coelenterate Mitochondrial &amp; Mycoplasma/Spiroplasma"</tt><tt class="py-op">,</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">        <tt class="py-string">"--MM---------------M------------MMMM---------------M------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">         </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">    <tt id="link-171" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-171', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-string">"Invertebrate Mitochondrial"</tt><tt class="py-op">,</tt>  </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSSSSVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">        <tt class="py-string">"---M----------------------------MMMM---------------M------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line"> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">    <tt id="link-172" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-172', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">6</tt><tt class="py-op">,</tt> <tt class="py-string">"Ciliate, Dasycladacean and Hexamita Nuclear"</tt><tt class="py-op">,</tt>     </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYYQQCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">        <tt class="py-string">"-----------------------------------M----------------------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">  </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">    <tt id="link-173" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-173', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">9</tt><tt class="py-op">,</tt> <tt class="py-string">"Echinoderm and Flatworm Mitochondrial"</tt><tt class="py-op">,</tt>  </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNNKSSSSVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">        <tt class="py-string">"-----------------------------------M---------------M------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line"> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">    <tt id="link-174" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-174', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">10</tt><tt class="py-op">,</tt> <tt class="py-string">"Euplotid Nuclear"</tt><tt class="py-op">,</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY**CCCWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">        <tt class="py-string">"-----------------------------------M----------------------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">   </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">    <tt id="link-175" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-175', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">11</tt><tt class="py-op">,</tt> <tt class="py-string">"Bacterial and Plant Plastid"</tt><tt class="py-op">,</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">        <tt class="py-string">"---M---------------M------------MMMM---------------M------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line"> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">    <tt id="link-176" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-176', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">12</tt><tt class="py-op">,</tt> <tt class="py-string">"Alternative Yeast Nuclear"</tt><tt class="py-op">,</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY**CC*WLLLSPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">        <tt class="py-string">"-------------------M---------------M----------------------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">                                                                    </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">    <tt id="link-177" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-177', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">13</tt><tt class="py-op">,</tt><tt class="py-string">"Ascidian Mitochondrial"</tt><tt class="py-op">,</tt>  </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSSGGVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">        <tt class="py-string">"-----------------------------------M----------------------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line"> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line">    <tt id="link-178" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-178', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">14</tt><tt class="py-op">,</tt> <tt class="py-string">"Alternative Flatworm Mitochondrial"</tt><tt class="py-op">,</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYYY*CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNNKSSSSVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">        <tt class="py-string">"-----------------------------------M----------------------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line"> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">    <tt id="link-179" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-179', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">15</tt><tt class="py-op">,</tt> <tt class="py-string">"Blepharisma Nuclear"</tt><tt class="py-op">,</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY*QCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">        <tt class="py-string">"-----------------------------------M----------------------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line"> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">    <tt id="link-180" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-180', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">16</tt><tt class="py-op">,</tt> <tt class="py-string">"Chlorophycean Mitochondrial"</tt><tt class="py-op">,</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY*LCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line">        <tt class="py-string">"-----------------------------------M----------------------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line"> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">    <tt id="link-181" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-181', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">21</tt><tt class="py-op">,</tt> <tt class="py-string">"Trematode Mitochondrial"</tt><tt class="py-op">,</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNNKSSSSVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">        <tt class="py-string">"-----------------------------------M---------------M------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line"> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line">    <tt id="link-182" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-182', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">22</tt><tt class="py-op">,</tt> <tt class="py-string">"Scenedesmus obliquus Mitochondrial"</tt><tt class="py-op">,</tt> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">        <tt class="py-string">"FFLLSS*SYY*LCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">        <tt class="py-string">"-----------------------------------M----------------------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line"> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">    <tt id="link-183" class="py-name"><a title="corebio.transform.GeneticCode" class="py-name" href="#" onclick="return doclink('link-183', 'GeneticCode', 'link-167');">GeneticCode</a></tt><tt class="py-op">(</tt><tt class="py-number">23</tt><tt class="py-op">,</tt><tt class="py-string">"Thraustochytrium Mitochondrial"</tt><tt class="py-op">,</tt> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line">        <tt class="py-string">"FF*LSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line">        <tt class="py-string">"--------------------------------M--M---------------M------------"</tt><tt class="py-op">,</tt> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line">        <tt class="py-string">"TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line">        <tt class="py-string">"TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG"</tt><tt class="py-op">,</tt> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">        <tt class="py-string">"TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line">    <tt class="py-op">)</tt> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line">     </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line"> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line"> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line"><tt id="link-184" class="py-name" targets="Variable corebio.transform.reduced_protein_alphabets=corebio.transform-module.html#reduced_protein_alphabets"><a title="corebio.transform.reduced_protein_alphabets" class="py-name" href="#" onclick="return doclink('link-184', 'reduced_protein_alphabets', 'link-184');">reduced_protein_alphabets</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line">    <tt class="py-string">"LiB2"</tt><tt class="py-op">:</tt> <tt id="link-185" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-185', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">        <tt id="link-186" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-186', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"CFYWMLIV-GPATSNHQEDRKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line">        <tt id="link-187" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-187', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"IIIIIIII-SSSSSSSSSSSSX*-"</tt><tt class="py-op">,</tt> <tt id="link-188" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-188', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"ISX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 2"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">    <tt class="py-string">"LiB3"</tt><tt class="py-op">:</tt> <tt id="link-189" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-189', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">        <tt id="link-190" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-190', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"CFYWMLIV-GPATS-NHQEDRKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">        <tt id="link-191" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-191', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"IIIIIIII-SSSSS-EEEEEEEX*-"</tt><tt class="py-op">,</tt> <tt id="link-192" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-192', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"ISEX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 3"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">    <tt class="py-string">"LiB4"</tt><tt class="py-op">:</tt> <tt id="link-193" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-193', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">        <tt id="link-194" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-194', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"CFYW-MLIV-GPATS-NHQEDRKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">        <tt id="link-195" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-195', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYY-IIII-SSSSS-EEEEEEEX*-"</tt><tt class="py-op">,</tt> <tt id="link-196" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-196', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YISEX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 4"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line">    <tt class="py-string">"LiB5"</tt><tt class="py-op">:</tt> <tt id="link-197" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-197', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line">        <tt id="link-198" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-198', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"CFYW-MLIV-G-PATS-NHQEDRKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt> <tt class="py-op">)</tt> <tt class="py-op">,</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">        <tt id="link-199" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-199', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYY-IIII-G-SSSS-EEEEEEEX*-"</tt><tt class="py-op">,</tt> <tt id="link-200" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-200', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YIGSEX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 5"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">    <tt class="py-string">"LiB6"</tt><tt class="py-op">:</tt> <tt id="link-201" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-201', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">        <tt id="link-202" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-202', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"CFYW-MLIV-G-P-ATS-NHQEDRKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt> <tt class="py-op">)</tt> <tt class="py-op">,</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line">        <tt id="link-203" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-203', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYY-IIII-G-P-SSS-EEEEEEEX*-"</tt><tt class="py-op">,</tt> <tt id="link-204" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-204', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YIGPSEX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 6"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line">    <tt class="py-string">"LiB7"</tt><tt class="py-op">:</tt> <tt id="link-205" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-205', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">        <tt id="link-206" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-206', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"CFYW-MLIV-G-P-ATS-NHQED-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">        <tt id="link-207" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-207', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYY-IIII-G-P-SSS-EEEEE-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-208" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-208', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YIGPSEKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 7"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">    <tt class="py-string">"LiB8"</tt><tt class="py-op">:</tt> <tt id="link-209" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-209', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">        <tt id="link-210" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-210', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"CFYW-MLIV-G-P-ATS-NH-QED-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">        <tt id="link-211" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-211', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYY-IIII-G-P-SSS-NN-EEE-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-212" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-212', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YIGPSNEKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 8"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line">    <tt class="py-string">"LiB9"</tt><tt class="py-op">:</tt> <tt id="link-213" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-213', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">        <tt id="link-214" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-214', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"CFYW-ML-IV-G-P-ATS-NH-QED-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">        <tt id="link-215" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-215', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYY-LL-II-G-P-SSS-NN-EEE-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-216" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-216', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YLIGPSNEKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 9"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">    <tt class="py-string">"LiB10"</tt><tt class="py-op">:</tt> <tt id="link-217" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-217', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line">        <tt id="link-218" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-218', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-FYW-ML-IV-G-P-ATS-NH-QED-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">        <tt id="link-219" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-219', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-YYY-LL-II-G-P-SSS-NN-EEE-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-220" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-220', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"CYLIGPSNEKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 10"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">    <tt class="py-string">"LiB11"</tt><tt class="py-op">:</tt> <tt id="link-221" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-221', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line">        <tt id="link-222" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-222', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-FYW-ML-IV-G-P-A-TS-NH-QED-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line">        <tt id="link-223" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-223', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-YYY-LL-II-G-P-A-SS-NN-EEE-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-224" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-224', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"CYLIGPASNEKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 11"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">    <tt class="py-string">"LiB12"</tt><tt class="py-op">:</tt> <tt id="link-225" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-225', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">        <tt id="link-226" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-226', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-FYW-ML-IV-G-P-A-TS-NH-QE-D-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">        <tt id="link-227" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-227', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-YYY-LL-II-G-P-A-SS-NN-EE-D-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-228" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-228', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"CYLIGPASNEDKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 12"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">    <tt class="py-string">"LiB13"</tt><tt class="py-op">:</tt> <tt id="link-229" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-229', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line">        <tt id="link-230" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-230', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-FYW-ML-IV-G-P-A-T-S-NH-QE-D-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">        <tt id="link-231" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-231', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-YYY-LL-II-G-P-A-T-S-NN-EE-D-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-232" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-232', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"CYLIGPATSNEDKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 13"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">    <tt class="py-string">"LiB14"</tt><tt class="py-op">:</tt> <tt id="link-233" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-233', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line">        <tt id="link-234" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-234', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-FYW-ML-IV-G-P-A-T-S-N-H-QE-D-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">        <tt id="link-235" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-235', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-YYY-LL-II-G-P-A-T-S-N-H-EE-D-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-236" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-236', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"CYLIGPATSNHEDKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 14"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line">    <tt class="py-string">"LiB15"</tt><tt class="py-op">:</tt> <tt id="link-237" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-237', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line">        <tt id="link-238" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-238', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-FYW-ML-IV-G-P-A-T-S-N-H-QE-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line">        <tt id="link-239" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-239', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-YYY-LL-II-G-P-A-T-S-N-H-EE-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-240" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-240', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"CYLIGPATSNHEDRKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 15"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">    <tt class="py-string">"LiB16"</tt><tt class="py-op">:</tt> <tt id="link-241" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-241', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line">        <tt id="link-242" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-242', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-FY-W-ML-IV-G-P-A-T-S-N-H-QE-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line">        <tt id="link-243" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-243', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-YY-W-LL-II-G-P-A-T-S-N-H-EE-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-244" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-244', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"CYWLIGPATSNHEDRKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 16"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line">    <tt class="py-string">"LiB17"</tt><tt class="py-op">:</tt> <tt id="link-245" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-245', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line">        <tt id="link-246" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-246', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-FY-W-ML-IV-G-P-A-T-S-N-H-Q-E-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line">        <tt id="link-247" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-247', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-YY-W-LL-II-G-P-A-T-S-N-H-Q-E-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-248" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-248', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"CYWLIGPATSNHQEDRKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 17"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line">    <tt class="py-string">"LiB18"</tt><tt class="py-op">:</tt> <tt id="link-249" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-249', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line">        <tt id="link-250" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-250', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-FY-W-M-L-IV-G-P-A-T-S-N-H-Q-E-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line">        <tt id="link-251" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-251', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-YY-W-M-L-II-G-P-A-T-S-N-H-Q-E-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-252" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-252', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"CYWMLIGPATSNHQEDRKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 18"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line">    <tt class="py-string">"LiB19"</tt><tt class="py-op">:</tt> <tt id="link-253" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-253', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line">        <tt id="link-254" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-254', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-F-Y-W-M-L-IV-G-P-A-T-S-N-H-Q-E-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line">        <tt id="link-255" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-255', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-F-Y-W-M-L-II-G-P-A-T-S-N-H-Q-E-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-256" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-256', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"CFYWMLIGPATSNHQEDRKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table II, group 19"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L717"></a><tt class="py-lineno">717</tt>  <tt class="py-line">    <tt class="py-string">"LiB19"</tt><tt class="py-op">:</tt> <tt id="link-257" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-257', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">        <tt id="link-258" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-258', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-F-Y-W-M-L-I-V-G-P-A-T-S-N-H-Q-E-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line">        <tt id="link-259" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-259', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"C-F-Y-W-M-L-I-V-G-P-A-T-S-N-H-Q-E-D-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-260" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-260', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"CFYWMLIVGPATSNHQEDRKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L720"></a><tt class="py-lineno">720</tt>  <tt class="py-line">            <tt class="py-string">"Li et al (2003), table II, group 20"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>      </tt>
<a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line">         </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L723"></a><tt class="py-lineno">723</tt>  <tt class="py-line">    <tt class="py-string">"LiA2"</tt><tt class="py-op">:</tt> <tt id="link-261" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-261', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line">        <tt id="link-262" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-262', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"CMFILVWY-AGTSNQDEHRKPX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line">        <tt id="link-263" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-263', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"IIIIIIII-SSSSSSSSSSSSX*-"</tt><tt class="py-op">,</tt> <tt id="link-264" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-264', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"ISX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 2"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line">    <tt class="py-string">"LiA3"</tt><tt class="py-op">:</tt> <tt id="link-265" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-265', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L729"></a><tt class="py-lineno">729</tt>  <tt class="py-line">        <tt id="link-266" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-266', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"CMFILVWY-AGTSP-NQDEHRKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L730"></a><tt class="py-lineno">730</tt>  <tt class="py-line">        <tt id="link-267" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-267', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"IIIIIIII-SSSSS-EEEEEEEX*-"</tt><tt class="py-op">,</tt> <tt id="link-268" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-268', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"ISEX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 3"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L732"></a><tt class="py-lineno">732</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line">    <tt class="py-string">"LiA4"</tt><tt class="py-op">:</tt> <tt id="link-269" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-269', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line">        <tt id="link-270" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-270', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"CMFWY-ILV-AGTS-NQDEHRKPX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L735"></a><tt class="py-lineno">735</tt>  <tt class="py-line">        <tt id="link-271" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-271', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYYY-III-SSSS-EEEEEEEEX*-"</tt><tt class="py-op">,</tt> <tt id="link-272" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-272', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YISEX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 4"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>    </tt>
<a name="L737"></a><tt class="py-lineno">737</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L738"></a><tt class="py-lineno">738</tt>  <tt class="py-line">    <tt class="py-string">"LiA5"</tt><tt class="py-op">:</tt> <tt id="link-273" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-273', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line">        <tt id="link-274" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-274', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"FWYH-MILV-CATSP-G-NQDERKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt> <tt class="py-op">)</tt> <tt class="py-op">,</tt> </tt>
<a name="L740"></a><tt class="py-lineno">740</tt>  <tt class="py-line">        <tt id="link-275" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-275', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYY-IIII-SSSSS-G-EEEEEEX*-"</tt><tt class="py-op">,</tt> <tt id="link-276" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-276', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YISGEX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L741"></a><tt class="py-lineno">741</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 5"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>    </tt>
<a name="L742"></a><tt class="py-lineno">742</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L743"></a><tt class="py-lineno">743</tt>  <tt class="py-line">    <tt class="py-string">"LiA6"</tt><tt class="py-op">:</tt> <tt id="link-277" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-277', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L744"></a><tt class="py-lineno">744</tt>  <tt class="py-line">        <tt id="link-278" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-278', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"FWYH-MILV-CATS-P-G-NQDERKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt> <tt class="py-op">)</tt> <tt class="py-op">,</tt> </tt>
<a name="L745"></a><tt class="py-lineno">745</tt>  <tt class="py-line">        <tt id="link-279" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-279', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYY-IIII-SSSS-P-G-EEEEEEX*-"</tt><tt class="py-op">,</tt> <tt id="link-280" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-280', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YISPGEX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L746"></a><tt class="py-lineno">746</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 6"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>    </tt>
<a name="L747"></a><tt class="py-lineno">747</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L748"></a><tt class="py-lineno">748</tt>  <tt class="py-line">    <tt class="py-string">"LiA7"</tt><tt class="py-op">:</tt> <tt id="link-281" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-281', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L749"></a><tt class="py-lineno">749</tt>  <tt class="py-line">        <tt id="link-282" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-282', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"FWYH-MILV-CATS-P-G-NQDE-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L750"></a><tt class="py-lineno">750</tt>  <tt class="py-line">        <tt id="link-283" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-283', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYY-IIII-SSSS-P-G-EEEE-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-284" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-284', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YISPGEKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L751"></a><tt class="py-lineno">751</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 7"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>    </tt>
<a name="L752"></a><tt class="py-lineno">752</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L753"></a><tt class="py-lineno">753</tt>  <tt class="py-line">    <tt class="py-string">"LiA8"</tt><tt class="py-op">:</tt> <tt id="link-285" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-285', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L754"></a><tt class="py-lineno">754</tt>  <tt class="py-line">        <tt id="link-286" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-286', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"FWYH-MILV-CA-NTS-P-G-DE-QRKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L755"></a><tt class="py-lineno">755</tt>  <tt class="py-line">        <tt id="link-287" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-287', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYY-IIII-AA-SSS-P-G-NN-KKKX*-"</tt><tt class="py-op">,</tt> <tt id="link-288" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-288', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YIASPGNKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L756"></a><tt class="py-lineno">756</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 8"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>    </tt>
<a name="L757"></a><tt class="py-lineno">757</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L758"></a><tt class="py-lineno">758</tt>  <tt class="py-line">    <tt class="py-string">"LiA9"</tt><tt class="py-op">:</tt> <tt id="link-289" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-289', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L759"></a><tt class="py-lineno">759</tt>  <tt class="py-line">        <tt id="link-290" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-290', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"FWYH-ML-IV-CA-NTS-P-G-DE-QRKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L760"></a><tt class="py-lineno">760</tt>  <tt class="py-line">        <tt id="link-291" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-291', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYYY-LL-VV-AA-SSS-P-G-NN-KKKX*-"</tt><tt class="py-op">,</tt> <tt id="link-292" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-292', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YLVASPGNKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L761"></a><tt class="py-lineno">761</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 9"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>    </tt>
<a name="L762"></a><tt class="py-lineno">762</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L763"></a><tt class="py-lineno">763</tt>  <tt class="py-line">    <tt class="py-string">"LiA10"</tt><tt class="py-op">:</tt> <tt id="link-293" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-293', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L764"></a><tt class="py-lineno">764</tt>  <tt class="py-line">        <tt id="link-294" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-294', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"FWY-ML-IV-CA-TS-NH-P-G-DE-QRKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L765"></a><tt class="py-lineno">765</tt>  <tt class="py-line">        <tt id="link-295" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-295', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYY-LL-VV-AA-TT-NN-P-G-DD-KKKX*-"</tt><tt class="py-op">,</tt> <tt id="link-296" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-296', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YLVATNPGDKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L766"></a><tt class="py-lineno">766</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 10"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L767"></a><tt class="py-lineno">767</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L768"></a><tt class="py-lineno">768</tt>  <tt class="py-line">    <tt class="py-string">"LiA11"</tt><tt class="py-op">:</tt> <tt id="link-297" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-297', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L769"></a><tt class="py-lineno">769</tt>  <tt class="py-line">        <tt id="link-298" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-298', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"FWY-ML-IV-CA-TS-NH-P-G-D-QE-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L770"></a><tt class="py-lineno">770</tt>  <tt class="py-line">        <tt id="link-299" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-299', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYY-LL-VV-AA-TT-NN-P-G-D-EE-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-300" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-300', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YLVATNPGDEKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L771"></a><tt class="py-lineno">771</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 11"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L772"></a><tt class="py-lineno">772</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L773"></a><tt class="py-lineno">773</tt>  <tt class="py-line">    <tt class="py-string">"LiA12"</tt><tt class="py-op">:</tt> <tt id="link-301" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-301', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L774"></a><tt class="py-lineno">774</tt>  <tt class="py-line">        <tt id="link-302" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-302', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"FWY-ML-IV-C-A-TS-NH-P-G-D-QE-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L775"></a><tt class="py-lineno">775</tt>  <tt class="py-line">        <tt id="link-303" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-303', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYY-LL-VV-C-A-TT-NN-P-G-D-EE-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-304" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-304', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YLVCATNPGDEKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L776"></a><tt class="py-lineno">776</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 12"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L777"></a><tt class="py-lineno">777</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L778"></a><tt class="py-lineno">778</tt>  <tt class="py-line">    <tt class="py-string">"LiA13"</tt><tt class="py-op">:</tt> <tt id="link-305" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-305', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L779"></a><tt class="py-lineno">779</tt>  <tt class="py-line">        <tt id="link-306" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-306', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"FWY-ML-IV-C-A-T-S-NH-P-G-D-QE-RKX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L780"></a><tt class="py-lineno">780</tt>  <tt class="py-line">        <tt id="link-307" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-307', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYY-LL-VV-C-A-T-S-NN-P-G-D-EE-KKX*-"</tt><tt class="py-op">,</tt> <tt id="link-308" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-308', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YLVCATSNPGDEKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L781"></a><tt class="py-lineno">781</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 13"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L782"></a><tt class="py-lineno">782</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L783"></a><tt class="py-lineno">783</tt>  <tt class="py-line">    <tt class="py-string">"LiA14"</tt><tt class="py-op">:</tt> <tt id="link-309" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-309', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L784"></a><tt class="py-lineno">784</tt>  <tt class="py-line">        <tt id="link-310" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-310', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"FWY-ML-IV-C-A-T-S-NH-P-G-D-QE-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L785"></a><tt class="py-lineno">785</tt>  <tt class="py-line">        <tt id="link-311" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-311', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYY-LL-VV-C-A-T-S-NN-P-G-D-EE-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-312" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-312', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YLVCATSNPGDERKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L786"></a><tt class="py-lineno">786</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 14"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L787"></a><tt class="py-lineno">787</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L788"></a><tt class="py-lineno">788</tt>  <tt class="py-line">    <tt class="py-string">"LiA15"</tt><tt class="py-op">:</tt> <tt id="link-313" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-313', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L789"></a><tt class="py-lineno">789</tt>  <tt class="py-line">        <tt id="link-314" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-314', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"FWY-ML-IV-C-A-T-S-N-H-P-G-D-QE-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L790"></a><tt class="py-lineno">790</tt>  <tt class="py-line">        <tt id="link-315" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-315', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"YYY-LL-VV-C-A-T-S-N-H-P-G-D-EE-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-316" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-316', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"YLVCATSNHPGDERKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L791"></a><tt class="py-lineno">791</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 15"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L792"></a><tt class="py-lineno">792</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L793"></a><tt class="py-lineno">793</tt>  <tt class="py-line">    <tt class="py-string">"LiA16"</tt><tt class="py-op">:</tt> <tt id="link-317" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-317', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L794"></a><tt class="py-lineno">794</tt>  <tt class="py-line">        <tt id="link-318" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-318', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"W-FY-ML-IV-C-A-T-S-N-H-P-G-D-QE-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L795"></a><tt class="py-lineno">795</tt>  <tt class="py-line">        <tt id="link-319" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-319', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"W-YY-LL-VV-C-A-T-S-N-H-P-G-D-EE-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-320" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-320', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"WYLVCATSNHPGDERKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L796"></a><tt class="py-lineno">796</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 16"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L797"></a><tt class="py-lineno">797</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L798"></a><tt class="py-lineno">798</tt>  <tt class="py-line">    <tt class="py-string">"LiA17"</tt><tt class="py-op">:</tt> <tt id="link-321" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-321', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L799"></a><tt class="py-lineno">799</tt>  <tt class="py-line">        <tt id="link-322" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-322', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"W-FY-ML-IV-C-A-T-S-N-H-P-G-D-Q-E-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L800"></a><tt class="py-lineno">800</tt>  <tt class="py-line">        <tt id="link-323" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-323', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"W-YY-LL-VV-C-A-T-S-N-H-P-G-D-Q-E-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-324" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-324', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"WYLVCATSNHPGDQERKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L801"></a><tt class="py-lineno">801</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 17"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L802"></a><tt class="py-lineno">802</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L803"></a><tt class="py-lineno">803</tt>  <tt class="py-line">    <tt class="py-string">"LiA18"</tt><tt class="py-op">:</tt> <tt id="link-325" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-325', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L804"></a><tt class="py-lineno">804</tt>  <tt class="py-line">        <tt id="link-326" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-326', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"W-FY-M-L-IV-C-A-T-S-N-H-P-G-D-Q-E-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L805"></a><tt class="py-lineno">805</tt>  <tt class="py-line">        <tt id="link-327" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-327', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"W-YY-M-L-VV-C-A-T-S-N-H-P-G-D-Q-E-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-328" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-328', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"WYMLVCATSNHPGDQERKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L806"></a><tt class="py-lineno">806</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 18"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   </tt>
<a name="L807"></a><tt class="py-lineno">807</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L808"></a><tt class="py-lineno">808</tt>  <tt class="py-line">    <tt class="py-string">"LiA19"</tt><tt class="py-op">:</tt> <tt id="link-329" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-329', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L809"></a><tt class="py-lineno">809</tt>  <tt class="py-line">        <tt id="link-330" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-330', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"W-F-Y-M-L-IV-C-A-T-S-N-H-P-G-D-Q-E-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L810"></a><tt class="py-lineno">810</tt>  <tt class="py-line">        <tt id="link-331" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-331', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"W-F-Y-M-L-VV-C-A-T-S-N-H-P-G-D-Q-E-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-332" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-332', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"WFYMLVCATSNHPGDQERKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L811"></a><tt class="py-lineno">811</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 19"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L812"></a><tt class="py-lineno">812</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L813"></a><tt class="py-lineno">813</tt>  <tt class="py-line">    <tt class="py-string">"LiA20"</tt><tt class="py-op">:</tt> <tt id="link-333" class="py-name"><a title="corebio.transform.Transform" class="py-name" href="#" onclick="return doclink('link-333', 'Transform', 'link-44');">Transform</a></tt><tt class="py-op">(</tt> </tt>
<a name="L814"></a><tt class="py-lineno">814</tt>  <tt class="py-line">        <tt id="link-334" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-334', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"W-F-Y-M-L-I-V-C-A-T-S-N-H-P-G-D-Q-E-R-KX*-"</tt><tt class="py-op">,</tt> <tt class="py-name">std_protein_alphabet</tt>  <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L815"></a><tt class="py-lineno">815</tt>  <tt class="py-line">        <tt id="link-335" class="py-name"><a title="corebio.seq.Seq" class="py-name" href="#" onclick="return doclink('link-335', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"W-F-Y-M-L-I-V-C-A-T-S-N-H-P-G-D-Q-E-R-KX*-"</tt><tt class="py-op">,</tt> <tt id="link-336" class="py-name"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-336', 'Alphabet', 'link-10');">Alphabet</a></tt><tt class="py-op">(</tt><tt class="py-string">"WFYMLIVCATSNHPGDQERKX*-"</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L816"></a><tt class="py-lineno">816</tt>  <tt class="py-line">        <tt class="py-string">"Li et al (2003), table I, group 20"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L817"></a><tt class="py-lineno">817</tt>  <tt class="py-line"> </tt>
<a name="L818"></a><tt class="py-lineno">818</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L819"></a><tt class="py-lineno">819</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Mar 12 21:34:16 2014
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
