<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>corebio.seq_io</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="corebio-module.html">Package&nbsp;corebio</a> ::
        Package&nbsp;seq_io
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="corebio.seq_io-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="corebio.seq_io-module.html">Package corebio.seq_io</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment">#  Copyright (c) 2005 Gavin E. Crooks &lt;gec@threeplusone.com&gt;</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment">#  Copyright (c) 2006, The Regents of the University of California, through </tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment">#  Lawrence Berkeley National Laboratory (subject to receipt of any required</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment">#  approvals from the U.S. Dept. of Energy).  All rights reserved.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment">#  This software is distributed under the new BSD Open Source License.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment">#  &lt;http://www.opensource.org/licenses/bsd-license.html&gt;</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-comment">#  Redistribution and use in source and binary forms, with or without </tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-comment">#  modification, are permitted provided that the following conditions are met: </tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-comment">#  (1) Redistributions of source code must retain the above copyright notice, </tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-comment">#  this list of conditions and the following disclaimer. </tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-comment">#  (2) Redistributions in binary form must reproduce the above copyright </tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-comment">#  notice, this list of conditions and the following disclaimer in the </tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-comment">#  documentation and or other materials provided with the distribution. </tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-comment">#  (3) Neither the name of the University of California, Lawrence Berkeley </tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-comment">#  National Laboratory, U.S. Dept. of Energy nor the names of its contributors </tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-comment">#  may be used to endorse or promote products derived from this software </tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-comment">#  without specific prior written permission. </tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-comment">#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" </tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-comment">#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE </tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-comment">#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE </tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-comment">#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE </tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-comment">#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR </tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-comment">#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF </tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-comment">#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS </tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-comment">#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN </tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-comment">#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) </tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-comment">#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE </tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-comment">#  POSSIBILITY OF SUCH DAMAGE. </tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">""" Sequence file reading and writing.</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring">Biological sequence data is stored and transmitted using a wide variety of</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">different file formats. This package provides convenient methods to read and</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring">write several of these file fomats.</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">CoreBio is often capable of guessing the correct file type, either from the</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring">file extension or the structure of the file:</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; import corebio.seq_io</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; afile = open("test_corebio/data/cap.fa")</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; seqs = corebio.seq_io.read(afile)</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring">Alternatively, each sequence file type has a separate module named FILETYPE_io</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">(e.g. fasta_io, clustal_io).</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; import corebio.seq_io.fasta_io</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; afile = open("test_corebio/data/cap.fa")</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; seqs = corebio.seq_io.fasta_io.read( afile )</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">Sequence data can also be written back to files:</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; fout = open("out.fa", "w")</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; corebio.seq_io.fasta_io.write( fout, seqs )</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">Supported File Formats</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring">----------------------</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">Module              Name            Extension  read write features   </tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring">---------------------------------------------------------------------------</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring">array_io            array, flatfile             yes  yes    none</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-docstring">clustal_io          clustalw        aln         yes  yes</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"><tt class="py-docstring">fasta_io            fasta, Pearson  fa          yes  yes    none</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-docstring">genbank_io          genbank         gb          yes         </tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-docstring">intelligenetics_io  intelligenetics ig          yes  yes</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt class="py-docstring">msf_io              msf             msf         yes</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-docstring">nbrf_io             nbrf, pir       pir         yes</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-docstring">nexus_io            nexus           nexus       yes</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring">phylip_io           phylip          phy         yes</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt class="py-docstring">plain_io            plain, raw                  yes  yes    none</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring">table_io            table           tbl         yes  yes    none</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-docstring">Each IO module defines one or more of the following functions and variables:</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"><tt class="py-docstring">read(afile, alphabet=None) </tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"><tt class="py-docstring">    Read a file of sequence data and return a SeqList, a collection</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"><tt class="py-docstring">    of Seq's (Alphabetic strings) and features.</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring">read_seq(afile, alphabet=None)</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring">    Read a single sequence from a file.</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"><tt class="py-docstring">iter_seq(afile, alphabet =None) </tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"><tt class="py-docstring">    Iterate over the sequences in a file. </tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring">index(afile, alphabet = None)</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"><tt class="py-docstring">    Instead of loading all of the sequences into memory, scan the file and</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring">    return an index map that will load sequences on demand. Typically not</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"><tt class="py-docstring">    implemented for formats with interleaved sequences.</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring">write(afile, seqlist)</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"><tt class="py-docstring">    Write a collection of sequences to the specifed file.</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt class="py-docstring">write_seq(afile, seq)</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"><tt class="py-docstring">    Write one sequence to the file. Only implemented for non-interleaved, </tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"><tt class="py-docstring">    headerless formats, such as fasta and plain.</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line"><tt class="py-docstring">example</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"><tt class="py-docstring">    A string containing a short example of the file format</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"><tt class="py-docstring">names</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line"><tt class="py-docstring">    A list of synonyms for the file format. E.g. for fasta_io, ( 'fasta',    </tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"><tt class="py-docstring">    'pearson', 'fa'). The first entry is the preferred format name.</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line"><tt class="py-docstring">extensions</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"><tt class="py-docstring">    A list of file name extensions used for this file format. e.g. </tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line"><tt class="py-docstring">    fasta_io.extensions is ('fa', 'fasta', 'fast', 'seq', 'fsa', 'fst', 'nt',</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line"><tt class="py-docstring">    'aa','fna','mpfa').  The preferred or standard extension is first in the </tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"><tt class="py-docstring">    list.</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"><tt class="py-docstring">Attributes :</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line"><tt class="py-docstring">- formats -- Available seq_io format parsers</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"><tt class="py-docstring">- format_names -- A map between format names and format parsers.</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"><tt class="py-docstring">- format_extensions -- A map between filename extensions and parsers.</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-comment"># Dev. References :</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-comment">#    - http://iubio.bio.indiana.edu/soft/molbio/readseq/java/Readseq2-help.html</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-comment">#    - http://www.ebi.ac.uk/help/formats_frame.html</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-comment">#    - http://www.cmbi.kun.nl/bioinf/tools/crab_pir.html</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-comment">#    - http://bioperl.org/HOWTOs/html/SeqIO.html</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-comment">#    - http://emboss.sourceforge.net/docs/themes/SequenceFormats.html</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-comment">#    - http://www.cse.ucsc.edu/research/compbio/a2m-desc.html (a2m)</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"><tt class="py-comment">#    - http://www.genomatix.de/online_help/help/sequence_formats.html</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package corebio=corebio-module.html"><a title="corebio" class="py-name" href="#" onclick="return doclink('link-0', 'corebio', 'link-0');">corebio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Module corebio.seq=corebio.seq-module.html"><a title="corebio.seq" class="py-name" href="#" onclick="return doclink('link-1', 'seq', 'link-1');">seq</a></tt> <tt class="py-keyword">import</tt> <tt class="py-op">*</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Module corebio.seq_io.clustal_io=corebio.seq_io.clustal_io-module.html"><a title="corebio.seq_io.clustal_io" class="py-name" href="#" onclick="return doclink('link-2', 'clustal_io', 'link-2');">clustal_io</a></tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Module corebio.seq_io.fasta_io=corebio.seq_io.fasta_io-module.html"><a title="corebio.seq_io.fasta_io" class="py-name" href="#" onclick="return doclink('link-3', 'fasta_io', 'link-3');">fasta_io</a></tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-4" class="py-name" targets="Module corebio.seq_io.msf_io=corebio.seq_io.msf_io-module.html"><a title="corebio.seq_io.msf_io" class="py-name" href="#" onclick="return doclink('link-4', 'msf_io', 'link-4');">msf_io</a></tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Module corebio.seq_io.nbrf_io=corebio.seq_io.nbrf_io-module.html"><a title="corebio.seq_io.nbrf_io" class="py-name" href="#" onclick="return doclink('link-5', 'nbrf_io', 'link-5');">nbrf_io</a></tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-6" class="py-name" targets="Module corebio.seq_io.nexus_io=corebio.seq_io.nexus_io-module.html"><a title="corebio.seq_io.nexus_io" class="py-name" href="#" onclick="return doclink('link-6', 'nexus_io', 'link-6');">nexus_io</a></tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-7" class="py-name" targets="Module corebio.seq_io.plain_io=corebio.seq_io.plain_io-module.html"><a title="corebio.seq_io.plain_io" class="py-name" href="#" onclick="return doclink('link-7', 'plain_io', 'link-7');">plain_io</a></tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-8" class="py-name" targets="Module corebio.seq_io.phylip_io=corebio.seq_io.phylip_io-module.html"><a title="corebio.seq_io.phylip_io" class="py-name" href="#" onclick="return doclink('link-8', 'phylip_io', 'link-8');">phylip_io</a></tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"><tt class="py-comment">#import null_io</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-9" class="py-name" targets="Module corebio.seq_io.stockholm_io=corebio.seq_io.stockholm_io-module.html"><a title="corebio.seq_io.stockholm_io" class="py-name" href="#" onclick="return doclink('link-9', 'stockholm_io', 'link-9');">stockholm_io</a></tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-10" class="py-name" targets="Module corebio.seq_io.intelligenetics_io=corebio.seq_io.intelligenetics_io-module.html"><a title="corebio.seq_io.intelligenetics_io" class="py-name" href="#" onclick="return doclink('link-10', 'intelligenetics_io', 'link-10');">intelligenetics_io</a></tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-11" class="py-name" targets="Module corebio.seq_io.table_io=corebio.seq_io.table_io-module.html"><a title="corebio.seq_io.table_io" class="py-name" href="#" onclick="return doclink('link-11', 'table_io', 'link-11');">table_io</a></tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-12" class="py-name" targets="Module corebio.seq_io.array_io=corebio.seq_io.array_io-module.html"><a title="corebio.seq_io.array_io" class="py-name" href="#" onclick="return doclink('link-12', 'array_io', 'link-12');">array_io</a></tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Module corebio.seq_io.genbank_io=corebio.seq_io.genbank_io-module.html"><a title="corebio.seq_io.genbank_io" class="py-name" href="#" onclick="return doclink('link-13', 'genbank_io', 'link-13');">genbank_io</a></tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line"> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line"><tt class="py-name">__all__</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">    <tt class="py-string">'clustal_io'</tt><tt class="py-op">,</tt>  </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">    <tt class="py-string">'fasta_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">    <tt class="py-string">'msf_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">    <tt class="py-string">'nbrf_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">    <tt class="py-string">'nexus_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">    <tt class="py-string">'plain_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">    <tt class="py-string">'phylip_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">    <tt class="py-string">'null_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">    <tt class="py-string">'stockholm_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">    <tt class="py-string">'intelligenetics_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">    <tt class="py-string">'table_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">    <tt class="py-string">'array_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">    <tt class="py-string">'genbank_io'</tt><tt class="py-op">,</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">    <tt class="py-string">'read'</tt><tt class="py-op">,</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">    <tt class="py-string">'formats'</tt><tt class="py-op">,</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">    <tt class="py-string">'format_names'</tt><tt class="py-op">,</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">    <tt class="py-string">'format_extensions'</tt><tt class="py-op">,</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">    <tt class="py-op">]</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"><tt id="link-14" class="py-name" targets="Variable corebio.seq_io.formats=corebio.seq_io-module.html#formats,Variable weblogolib.GhostscriptAPI.formats=weblogolib.GhostscriptAPI-class.html#formats"><a title="corebio.seq_io.formats
weblogolib.GhostscriptAPI.formats" class="py-name" href="#" onclick="return doclink('link-14', 'formats', 'link-14');">formats</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt> <tt id="link-15" class="py-name"><a title="corebio.seq_io.clustal_io" class="py-name" href="#" onclick="return doclink('link-15', 'clustal_io', 'link-2');">clustal_io</a></tt><tt class="py-op">,</tt> <tt id="link-16" class="py-name"><a title="corebio.seq_io.fasta_io" class="py-name" href="#" onclick="return doclink('link-16', 'fasta_io', 'link-3');">fasta_io</a></tt><tt class="py-op">,</tt> <tt id="link-17" class="py-name"><a title="corebio.seq_io.plain_io" class="py-name" href="#" onclick="return doclink('link-17', 'plain_io', 'link-7');">plain_io</a></tt><tt class="py-op">,</tt> <tt id="link-18" class="py-name"><a title="corebio.seq_io.msf_io" class="py-name" href="#" onclick="return doclink('link-18', 'msf_io', 'link-4');">msf_io</a></tt><tt class="py-op">,</tt> <tt id="link-19" class="py-name"><a title="corebio.seq_io.genbank_io" class="py-name" href="#" onclick="return doclink('link-19', 'genbank_io', 'link-13');">genbank_io</a></tt><tt class="py-op">,</tt><tt id="link-20" class="py-name"><a title="corebio.seq_io.nbrf_io" class="py-name" href="#" onclick="return doclink('link-20', 'nbrf_io', 'link-5');">nbrf_io</a></tt><tt class="py-op">,</tt> <tt id="link-21" class="py-name"><a title="corebio.seq_io.nexus_io" class="py-name" href="#" onclick="return doclink('link-21', 'nexus_io', 'link-6');">nexus_io</a></tt><tt class="py-op">,</tt> <tt id="link-22" class="py-name"><a title="corebio.seq_io.phylip_io" class="py-name" href="#" onclick="return doclink('link-22', 'phylip_io', 'link-8');">phylip_io</a></tt><tt class="py-op">,</tt> <tt id="link-23" class="py-name"><a title="corebio.seq_io.stockholm_io" class="py-name" href="#" onclick="return doclink('link-23', 'stockholm_io', 'link-9');">stockholm_io</a></tt><tt class="py-op">,</tt> <tt id="link-24" class="py-name"><a title="corebio.seq_io.intelligenetics_io" class="py-name" href="#" onclick="return doclink('link-24', 'intelligenetics_io', 'link-10');">intelligenetics_io</a></tt><tt class="py-op">,</tt> <tt id="link-25" class="py-name"><a title="corebio.seq_io.table_io" class="py-name" href="#" onclick="return doclink('link-25', 'table_io', 'link-11');">table_io</a></tt><tt class="py-op">,</tt> <tt id="link-26" class="py-name"><a title="corebio.seq_io.array_io" class="py-name" href="#" onclick="return doclink('link-26', 'array_io', 'link-12');">array_io</a></tt><tt class="py-op">)</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line"><tt class="py-string">"""Available seq_io formats"""</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"> </tt>
<a name="format_names"></a><div id="format_names-def"><a name="L176"></a><tt class="py-lineno">176</tt> <a class="py-toggle" href="#" id="format_names-toggle" onclick="return toggle('format_names');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io-module.html#format_names">format_names</a><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt>    </tt>
</div><a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return a map between format names and format modules"""</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">    <tt class="py-keyword">global</tt> <tt id="link-27" class="py-name"><a title="corebio.seq_io.formats
weblogolib.GhostscriptAPI.formats" class="py-name" href="#" onclick="return doclink('link-27', 'formats', 'link-14');">formats</a></tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">    <tt class="py-name">fnames</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt id="link-28" class="py-name"><a title="corebio.seq_io.formats
weblogolib.GhostscriptAPI.formats" class="py-name" href="#" onclick="return doclink('link-28', 'formats', 'link-14');">formats</a></tt> <tt class="py-op">:</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-29" class="py-name" targets="Variable corebio.db.DataSource.name=corebio.db.DataSource-class.html#name,Variable corebio.matrix.SubMatrix.name=corebio.matrix.SubMatrix-class.html#name,Variable corebio.seq.SeqList.name=corebio.seq.SeqList-class.html#name,Variable corebio.transform.Transform.name=corebio.transform.Transform-class.html#name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-29', 'name', 'link-29');">name</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name" targets="Variable corebio.seq_io.array_io.names=corebio.seq_io.array_io-module.html#names,Variable corebio.seq_io.clustal_io.names=corebio.seq_io.clustal_io-module.html#names,Variable corebio.seq_io.fasta_io.names=corebio.seq_io.fasta_io-module.html#names,Variable corebio.seq_io.genbank_io.names=corebio.seq_io.genbank_io-module.html#names,Variable corebio.seq_io.intelligenetics_io.names=corebio.seq_io.intelligenetics_io-module.html#names,Variable corebio.seq_io.msf_io.names=corebio.seq_io.msf_io-module.html#names,Variable corebio.seq_io.nbrf_io.names=corebio.seq_io.nbrf_io-module.html#names,Variable corebio.seq_io.nexus_io.names=corebio.seq_io.nexus_io-module.html#names,Variable corebio.seq_io.null_io.names=corebio.seq_io.null_io-module.html#names,Variable corebio.seq_io.phylip_io.names=corebio.seq_io.phylip_io-module.html#names,Variable corebio.seq_io.plain_io.names=corebio.seq_io.plain_io-module.html#names,Variable corebio.seq_io.stockholm_io.names=corebio.seq_io.stockholm_io-module.html#names,Variable corebio.seq_io.table_io.names=corebio.seq_io.table_io-module.html#names,Static Method weblogolib.color.Color.names()=weblogolib.color.Color-class.html#names"><a title="corebio.seq_io.array_io.names
corebio.seq_io.clustal_io.names
corebio.seq_io.fasta_io.names
corebio.seq_io.genbank_io.names
corebio.seq_io.intelligenetics_io.names
corebio.seq_io.msf_io.names
corebio.seq_io.nbrf_io.names
corebio.seq_io.nexus_io.names
corebio.seq_io.null_io.names
corebio.seq_io.phylip_io.names
corebio.seq_io.plain_io.names
corebio.seq_io.stockholm_io.names
corebio.seq_io.table_io.names
weblogolib.color.Color.names" class="py-name" href="#" onclick="return doclink('link-30', 'names', 'link-30');">names</a></tt> <tt class="py-op">:</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt id="link-31" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-31', 'name', 'link-29');">name</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fnames</tt> <tt class="py-comment"># Insanity check</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">            <tt class="py-name">fnames</tt><tt class="py-op">[</tt><tt id="link-32" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-32', 'name', 'link-29');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">f</tt>   </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">fnames</tt> </tt>
</div><a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line"> </tt>
<a name="format_extensions"></a><div id="format_extensions-def"><a name="L186"></a><tt class="py-lineno">186</tt> <a class="py-toggle" href="#" id="format_extensions-toggle" onclick="return toggle('format_extensions');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io-module.html#format_extensions">format_extensions</a><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt>    </tt>
</div><a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return a map between filename extensions and sequence file types"""</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">    <tt class="py-keyword">global</tt> <tt id="link-33" class="py-name"><a title="corebio.seq_io.formats
weblogolib.GhostscriptAPI.formats" class="py-name" href="#" onclick="return doclink('link-33', 'formats', 'link-14');">formats</a></tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">    <tt class="py-name">fext</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt id="link-34" class="py-name"><a title="corebio.seq_io.formats
weblogolib.GhostscriptAPI.formats" class="py-name" href="#" onclick="return doclink('link-34', 'formats', 'link-14');">formats</a></tt> <tt class="py-op">:</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">ext</tt> <tt class="py-keyword">in</tt> <tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name" targets="Variable corebio.seq_io.array_io.extensions=corebio.seq_io.array_io-module.html#extensions,Variable corebio.seq_io.clustal_io.extensions=corebio.seq_io.clustal_io-module.html#extensions,Variable corebio.seq_io.fasta_io.extensions=corebio.seq_io.fasta_io-module.html#extensions,Variable corebio.seq_io.genbank_io.extensions=corebio.seq_io.genbank_io-module.html#extensions,Variable corebio.seq_io.intelligenetics_io.extensions=corebio.seq_io.intelligenetics_io-module.html#extensions,Variable corebio.seq_io.msf_io.extensions=corebio.seq_io.msf_io-module.html#extensions,Variable corebio.seq_io.nbrf_io.extensions=corebio.seq_io.nbrf_io-module.html#extensions,Variable corebio.seq_io.nexus_io.extensions=corebio.seq_io.nexus_io-module.html#extensions,Variable corebio.seq_io.null_io.extensions=corebio.seq_io.null_io-module.html#extensions,Variable corebio.seq_io.phylip_io.extensions=corebio.seq_io.phylip_io-module.html#extensions,Variable corebio.seq_io.plain_io.extensions=corebio.seq_io.plain_io-module.html#extensions,Variable corebio.seq_io.stockholm_io.extensions=corebio.seq_io.stockholm_io-module.html#extensions,Variable corebio.seq_io.table_io.extensions=corebio.seq_io.table_io-module.html#extensions"><a title="corebio.seq_io.array_io.extensions
corebio.seq_io.clustal_io.extensions
corebio.seq_io.fasta_io.extensions
corebio.seq_io.genbank_io.extensions
corebio.seq_io.intelligenetics_io.extensions
corebio.seq_io.msf_io.extensions
corebio.seq_io.nbrf_io.extensions
corebio.seq_io.nexus_io.extensions
corebio.seq_io.null_io.extensions
corebio.seq_io.phylip_io.extensions
corebio.seq_io.plain_io.extensions
corebio.seq_io.stockholm_io.extensions
corebio.seq_io.table_io.extensions" class="py-name" href="#" onclick="return doclink('link-35', 'extensions', 'link-35');">extensions</a></tt> <tt class="py-op">:</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">ext</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fext</tt> <tt class="py-comment"># Insanity check</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">            <tt class="py-name">fext</tt><tt class="py-op">[</tt><tt class="py-name">ext</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">f</tt>   </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">fext</tt> </tt>
</div><a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line"> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">     </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line"><tt class="py-comment"># seq_io._parsers is an ordered list of sequence parsers that are tried, in </tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"><tt class="py-comment"># turn, on files of unknown format. Each parser must raise an exception when</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"><tt class="py-comment"># fed a format further down the list.</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"><tt class="py-comment"># The general trend is most common to least common file format. However, </tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"><tt class="py-comment"># 'nbrf_io' is before 'fasta_io' because nbrf looks like fasta with extras, and</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"><tt class="py-comment"># 'array_io' is last, since it is very general.</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line"><tt id="link-36" class="py-name" targets="Variable corebio.seq_io._parsers=corebio.seq_io-module.html#_parsers"><a title="corebio.seq_io._parsers" class="py-name" href="#" onclick="return doclink('link-36', '_parsers', 'link-36');">_parsers</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt id="link-37" class="py-name"><a title="corebio.seq_io.nbrf_io" class="py-name" href="#" onclick="return doclink('link-37', 'nbrf_io', 'link-5');">nbrf_io</a></tt><tt class="py-op">,</tt> <tt id="link-38" class="py-name"><a title="corebio.seq_io.fasta_io" class="py-name" href="#" onclick="return doclink('link-38', 'fasta_io', 'link-3');">fasta_io</a></tt><tt class="py-op">,</tt> <tt id="link-39" class="py-name"><a title="corebio.seq_io.clustal_io" class="py-name" href="#" onclick="return doclink('link-39', 'clustal_io', 'link-2');">clustal_io</a></tt><tt class="py-op">,</tt> <tt id="link-40" class="py-name"><a title="corebio.seq_io.phylip_io" class="py-name" href="#" onclick="return doclink('link-40', 'phylip_io', 'link-8');">phylip_io</a></tt><tt class="py-op">,</tt> <tt id="link-41" class="py-name"><a title="corebio.seq_io.genbank_io" class="py-name" href="#" onclick="return doclink('link-41', 'genbank_io', 'link-13');">genbank_io</a></tt><tt class="py-op">,</tt> <tt id="link-42" class="py-name"><a title="corebio.seq_io.stockholm_io" class="py-name" href="#" onclick="return doclink('link-42', 'stockholm_io', 'link-9');">stockholm_io</a></tt><tt class="py-op">,</tt> <tt id="link-43" class="py-name"><a title="corebio.seq_io.msf_io" class="py-name" href="#" onclick="return doclink('link-43', 'msf_io', 'link-4');">msf_io</a></tt><tt class="py-op">,</tt> <tt id="link-44" class="py-name"><a title="corebio.seq_io.nexus_io" class="py-name" href="#" onclick="return doclink('link-44', 'nexus_io', 'link-6');">nexus_io</a></tt><tt class="py-op">,</tt> <tt id="link-45" class="py-name"><a title="corebio.seq_io.table_io" class="py-name" href="#" onclick="return doclink('link-45', 'table_io', 'link-11');">table_io</a></tt><tt class="py-op">,</tt> <tt id="link-46" class="py-name"><a title="corebio.seq_io.array_io" class="py-name" href="#" onclick="return doclink('link-46', 'array_io', 'link-12');">array_io</a></tt><tt class="py-op">)</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">  </tt>
<a name="_get_parsers"></a><div id="_get_parsers-def"><a name="L207"></a><tt class="py-lineno">207</tt> <a class="py-toggle" href="#" id="_get_parsers-toggle" onclick="return toggle('_get_parsers');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io-module.html#_get_parsers">_get_parsers</a><tt class="py-op">(</tt><tt class="py-param">fin</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="_get_parsers-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_parsers-expanded"><a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">    <tt class="py-keyword">global</tt> <tt id="link-47" class="py-name"><a title="corebio.seq_io._parsers" class="py-name" href="#" onclick="return doclink('link-47', '_parsers', 'link-36');">_parsers</a></tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">     </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">    <tt class="py-name">fnames</tt> <tt class="py-op">=</tt> <tt id="link-48" class="py-name" targets="Function corebio.seq_io.format_names()=corebio.seq_io-module.html#format_names"><a title="corebio.seq_io.format_names" class="py-name" href="#" onclick="return doclink('link-48', 'format_names', 'link-48');">format_names</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">    <tt class="py-name">fext</tt> <tt class="py-op">=</tt> <tt id="link-49" class="py-name" targets="Function corebio.seq_io.format_extensions()=corebio.seq_io-module.html#format_extensions"><a title="corebio.seq_io.format_extensions" class="py-name" href="#" onclick="return doclink('link-49', 'format_extensions', 'link-49');">format_extensions</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">    <tt class="py-name">parsers</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-50" class="py-name"><a title="corebio.seq_io._parsers" class="py-name" href="#" onclick="return doclink('link-50', '_parsers', 'link-36');">_parsers</a></tt><tt class="py-op">)</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">    <tt class="py-name">best_guess</tt> <tt class="py-op">=</tt> <tt class="py-name">parsers</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">     </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">    <tt class="py-comment"># If a filename is supplied use the extension to guess the format.</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">fin</tt><tt class="py-op">,</tt> <tt class="py-string">"name"</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-string">'.'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fin</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-51', 'name', 'link-29');">name</a></tt> <tt class="py-op">:</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">        <tt id="link-52" class="py-name" targets="Variable weblogolib._cgi.extension=weblogolib._cgi-module.html#extension"><a title="weblogolib._cgi.extension" class="py-name" href="#" onclick="return doclink('link-52', 'extension', 'link-52');">extension</a></tt> <tt class="py-op">=</tt> <tt class="py-name">fin</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name"><a title="corebio.db.DataSource.name
corebio.matrix.SubMatrix.name
corebio.seq.SeqList.name
corebio.transform.Transform.name" class="py-name" href="#" onclick="return doclink('link-53', 'name', 'link-29');">name</a></tt><tt class="py-op">.</tt><tt id="link-54" class="py-name" targets="Method corebio.seq_io._nexus.Trees.Tree.split()=corebio.seq_io._nexus.Trees.Tree-class.html#split"><a title="corebio.seq_io._nexus.Trees.Tree.split" class="py-name" href="#" onclick="return doclink('link-54', 'split', 'link-54');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'.'</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-55" class="py-name"><a title="weblogolib._cgi.extension" class="py-name" href="#" onclick="return doclink('link-55', 'extension', 'link-52');">extension</a></tt> <tt class="py-keyword">in</tt>  <tt class="py-name">fnames</tt><tt class="py-op">:</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">            <tt class="py-name">best_guess</tt> <tt class="py-op">=</tt> <tt class="py-name">fnames</tt><tt class="py-op">[</tt><tt id="link-56" class="py-name"><a title="weblogolib._cgi.extension" class="py-name" href="#" onclick="return doclink('link-56', 'extension', 'link-52');">extension</a></tt><tt class="py-op">]</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-57" class="py-name"><a title="weblogolib._cgi.extension" class="py-name" href="#" onclick="return doclink('link-57', 'extension', 'link-52');">extension</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">fext</tt> <tt class="py-op">:</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">            <tt class="py-name">best_guess</tt> <tt class="py-op">=</tt> <tt class="py-name">fext</tt><tt class="py-op">[</tt><tt id="link-58" class="py-name"><a title="weblogolib._cgi.extension" class="py-name" href="#" onclick="return doclink('link-58', 'extension', 'link-52');">extension</a></tt><tt class="py-op">]</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">         </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">best_guess</tt> <tt class="py-keyword">in</tt> <tt class="py-name">parsers</tt> <tt class="py-op">:</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">        <tt class="py-name">parsers</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name" targets="Method corebio.seq.Seq.remove()=corebio.seq.Seq-class.html#remove"><a title="corebio.seq.Seq.remove" class="py-name" href="#" onclick="return doclink('link-59', 'remove', 'link-59');">remove</a></tt><tt class="py-op">(</tt><tt class="py-name">best_guess</tt><tt class="py-op">)</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">    <tt class="py-name">parsers</tt><tt class="py-op">.</tt><tt class="py-name">insert</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt><tt class="py-name">best_guess</tt><tt class="py-op">)</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line"> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">parsers</tt> </tt>
</div><a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line"> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">  </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">     </tt>
<a name="read"></a><div id="read-def"><a name="L231"></a><tt class="py-lineno">231</tt> <a class="py-toggle" href="#" id="read-toggle" onclick="return toggle('read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="corebio.seq_io-module.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">fin</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
</div><div id="read-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="read-expanded"><a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">    <tt class="py-docstring">""" Read a sequence file and attempt to guess its format. </tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line"><tt class="py-docstring">    First the filename extension (if available) is used to infer the format.</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line"><tt class="py-docstring">    If that fails, then we attempt to parse the file using several common   </tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"><tt class="py-docstring">    formats.</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line"><tt class="py-docstring">    Note, fin cannot be unseekable stream such as sys.stdin</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"><tt class="py-docstring">    returns :</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"><tt class="py-docstring">        SeqList</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"><tt class="py-docstring">    raises :</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line"><tt class="py-docstring">        ValueError - If the file cannot be parsed.</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"><tt class="py-docstring">        ValueError - Sequence do not conform to the alphabet.</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line"> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">    <tt id="link-60" class="py-name" targets="Variable corebio.matrix.Motif.alphabet=corebio.matrix.Motif-class.html#alphabet,Variable corebio.matrix.SubMatrix.alphabet=corebio.matrix.SubMatrix-class.html#alphabet,Variable corebio.seq.Seq.alphabet=corebio.seq.Seq-class.html#alphabet,Variable corebio.seq.SeqList.alphabet=corebio.seq.SeqList-class.html#alphabet"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-60', 'alphabet', 'link-60');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-61" class="py-name" targets="Class corebio.seq.Alphabet=corebio.seq.Alphabet-class.html"><a title="corebio.seq.Alphabet" class="py-name" href="#" onclick="return doclink('link-61', 'Alphabet', 'link-61');">Alphabet</a></tt><tt class="py-op">(</tt><tt id="link-62" class="py-name"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-62', 'alphabet', 'link-60');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">    <tt class="py-name">parsers</tt> <tt class="py-op">=</tt>  <tt class="py-name">_get_parsers</tt><tt class="py-op">(</tt><tt class="py-name">fin</tt><tt class="py-op">)</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">     </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">p</tt> <tt class="py-keyword">in</tt> <tt class="py-name">_get_parsers</tt><tt class="py-op">(</tt><tt class="py-name">fin</tt><tt class="py-op">)</tt> <tt class="py-op">:</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">        <tt class="py-name">fin</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name" targets="Method corebio.utils.FileIndex.seek()=corebio.utils.FileIndex-class.html#seek"><a title="corebio.utils.FileIndex.seek" class="py-name" href="#" onclick="return doclink('link-63', 'seek', 'link-63');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt>     </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">p</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name" targets="Static Method corebio.matrix.SubMatrix.read()=corebio.matrix.SubMatrix-class.html#read,Method corebio.seq_io._nexus.Nexus.read()=corebio.seq_io._nexus.Nexus-class.html#read,Function corebio.seq_io.array_io.read()=corebio.seq_io.array_io-module.html#read,Function corebio.seq_io.clustal_io.read()=corebio.seq_io.clustal_io-module.html#read,Function corebio.seq_io.fasta_io.read()=corebio.seq_io.fasta_io-module.html#read,Function corebio.seq_io.genbank_io.read()=corebio.seq_io.genbank_io-module.html#read,Function corebio.seq_io.intelligenetics_io.read()=corebio.seq_io.intelligenetics_io-module.html#read,Function corebio.seq_io.msf_io.read()=corebio.seq_io.msf_io-module.html#read,Function corebio.seq_io.nbrf_io.read()=corebio.seq_io.nbrf_io-module.html#read,Function corebio.seq_io.nexus_io.read()=corebio.seq_io.nexus_io-module.html#read,Function corebio.seq_io.null_io.read()=corebio.seq_io.null_io-module.html#read,Function corebio.seq_io.phylip_io.read()=corebio.seq_io.phylip_io-module.html#read,Function corebio.seq_io.plain_io.read()=corebio.seq_io.plain_io-module.html#read,Function corebio.seq_io.read()=corebio.seq_io-module.html#read,Function corebio.seq_io.stockholm_io.read()=corebio.seq_io.stockholm_io-module.html#read,Function corebio.seq_io.table_io.read()=corebio.seq_io.table_io-module.html#read"><a title="corebio.matrix.SubMatrix.read
corebio.seq_io._nexus.Nexus.read
corebio.seq_io.array_io.read
corebio.seq_io.clustal_io.read
corebio.seq_io.fasta_io.read
corebio.seq_io.genbank_io.read
corebio.seq_io.intelligenetics_io.read
corebio.seq_io.msf_io.read
corebio.seq_io.nbrf_io.read
corebio.seq_io.nexus_io.read
corebio.seq_io.null_io.read
corebio.seq_io.phylip_io.read
corebio.seq_io.plain_io.read
corebio.seq_io.read
corebio.seq_io.stockholm_io.read
corebio.seq_io.table_io.read" class="py-name" href="#" onclick="return doclink('link-64', 'read', 'link-64');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">fin</tt><tt class="py-op">,</tt> <tt id="link-65" class="py-name"><a title="corebio.matrix.Motif.alphabet
corebio.matrix.SubMatrix.alphabet
corebio.seq.Seq.alphabet
corebio.seq.SeqList.alphabet" class="py-name" href="#" onclick="return doclink('link-65', 'alphabet', 'link-60');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">             </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">    <tt id="link-66" class="py-name"><a title="corebio.seq_io.array_io.names
corebio.seq_io.clustal_io.names
corebio.seq_io.fasta_io.names
corebio.seq_io.genbank_io.names
corebio.seq_io.intelligenetics_io.names
corebio.seq_io.msf_io.names
corebio.seq_io.nbrf_io.names
corebio.seq_io.nexus_io.names
corebio.seq_io.null_io.names
corebio.seq_io.phylip_io.names
corebio.seq_io.plain_io.names
corebio.seq_io.stockholm_io.names
corebio.seq_io.table_io.names
weblogolib.color.Color.names" class="py-name" href="#" onclick="return doclink('link-66', 'names', 'link-30');">names</a></tt> <tt class="py-op">=</tt> <tt class="py-string">", "</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name" targets="Method corebio.seq.Seq.join()=corebio.seq.Seq-class.html#join"><a title="corebio.seq.Seq.join" class="py-name" href="#" onclick="return doclink('link-67', 'join', 'link-67');">join</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt> <tt class="py-name">p</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="corebio.seq_io.array_io.names
corebio.seq_io.clustal_io.names
corebio.seq_io.fasta_io.names
corebio.seq_io.genbank_io.names
corebio.seq_io.intelligenetics_io.names
corebio.seq_io.msf_io.names
corebio.seq_io.nbrf_io.names
corebio.seq_io.nexus_io.names
corebio.seq_io.null_io.names
corebio.seq_io.phylip_io.names
corebio.seq_io.plain_io.names
corebio.seq_io.stockholm_io.names
corebio.seq_io.table_io.names
weblogolib.color.Color.names" class="py-name" href="#" onclick="return doclink('link-68', 'names', 'link-30');">names</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">p</tt> <tt class="py-keyword">in</tt> <tt class="py-name">parsers</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Cannot parse sequence file: Tried %s "</tt> <tt class="py-op">%</tt> <tt id="link-69" class="py-name"><a title="corebio.seq_io.array_io.names
corebio.seq_io.clustal_io.names
corebio.seq_io.fasta_io.names
corebio.seq_io.genbank_io.names
corebio.seq_io.intelligenetics_io.names
corebio.seq_io.msf_io.names
corebio.seq_io.nbrf_io.names
corebio.seq_io.nexus_io.names
corebio.seq_io.null_io.names
corebio.seq_io.phylip_io.names
corebio.seq_io.plain_io.names
corebio.seq_io.stockholm_io.names
corebio.seq_io.table_io.names
weblogolib.color.Color.names" class="py-name" href="#" onclick="return doclink('link-69', 'names', 'link-30');">names</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Mar 12 21:34:13 2014
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
